// @inkeep/inkeep-analytics/src/models/errors/forbidden.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { ClosedEnum } from "../../types/enums.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { InkeepAnalyticsError } from "./inkeepanalyticserror.js";
import { SDKValidationError } from "./sdkvalidationerror.js";

/**
 * A short code indicating the error code returned.
 */
export const ForbiddenCode = {
  Forbidden: "forbidden",
} as const;
/**
 * A short code indicating the error code returned.
 */
export type ForbiddenCode = ClosedEnum<typeof ForbiddenCode>;

/**
 * A short code indicating the error code returned.
 */
export const ForbiddenErrorCode = {
  Forbidden: "forbidden",
} as const;
/**
 * A short code indicating the error code returned.
 */
export type ForbiddenErrorCode = ClosedEnum<typeof ForbiddenErrorCode>;

/**
 * Legacy error format for backward compatibility.
 */
export type ForbiddenError = {
  /**
   * A short code indicating the error code returned.
   */
  code: ForbiddenErrorCode;
  /**
   * A concise error message suitable for display to end users. May be truncated if the full detail is long.
   */
  message: string;
};

export type ForbiddenData = {
  /**
   * A short, human-readable summary of the problem type.
   */
  title: string;
  /**
   * The HTTP status code.
   */
  status: number;
  /**
   * A detailed explanation specific to this occurrence of the problem, providing context and specifics about what went wrong.
   */
  detail: string;
  /**
   * A URI reference that identifies the specific occurrence of the problem.
   */
  instance?: string | undefined;
  /**
   * A unique identifier for the request, useful for troubleshooting.
   */
  requestId?: string | undefined;
  /**
   * A short code indicating the error code returned.
   */
  code: ForbiddenCode;
  /**
   * Legacy error format for backward compatibility.
   */
  error: ForbiddenError;
};

export class Forbidden extends InkeepAnalyticsError {
  /**
   * A short, human-readable summary of the problem type.
   */
  title: string;
  /**
   * The HTTP status code.
   */
  status: number;
  /**
   * A detailed explanation specific to this occurrence of the problem, providing context and specifics about what went wrong.
   */
  detail: string;
  /**
   * A URI reference that identifies the specific occurrence of the problem.
   */
  instance?: string | undefined;
  /**
   * A unique identifier for the request, useful for troubleshooting.
   */
  requestId?: string | undefined;
  /**
   * A short code indicating the error code returned.
   */
  code: ForbiddenCode;
  /**
   * Legacy error format for backward compatibility.
   */
  error: ForbiddenError;

  /** The original data that was passed to this error instance. */
  data$: ForbiddenData;

  constructor(
    err: ForbiddenData,
    httpMeta: { response: Response; request: Request; body: string },
  ) {
    const message = "message" in err && typeof err.message === "string"
      ? err.message
      : `API error occurred: ${JSON.stringify(err)}`;
    super(message, httpMeta);
    this.data$ = err;
    this.title = err.title;
    this.status = err.status;
    this.detail = err.detail;
    if (err.instance != null) this.instance = err.instance;
    if (err.requestId != null) this.requestId = err.requestId;
    this.code = err.code;
    this.error = err.error;

    this.name = "Forbidden";
  }
}

/** @internal */
export const ForbiddenCode$inboundSchema: z.ZodNativeEnum<
  typeof ForbiddenCode
> = z.nativeEnum(ForbiddenCode);

/** @internal */
export const ForbiddenCode$outboundSchema: z.ZodNativeEnum<
  typeof ForbiddenCode
> = ForbiddenCode$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForbiddenCode$ {
  /** @deprecated use `ForbiddenCode$inboundSchema` instead. */
  export const inboundSchema = ForbiddenCode$inboundSchema;
  /** @deprecated use `ForbiddenCode$outboundSchema` instead. */
  export const outboundSchema = ForbiddenCode$outboundSchema;
}

/** @internal */
export const ForbiddenErrorCode$inboundSchema: z.ZodNativeEnum<
  typeof ForbiddenErrorCode
> = z.nativeEnum(ForbiddenErrorCode);

/** @internal */
export const ForbiddenErrorCode$outboundSchema: z.ZodNativeEnum<
  typeof ForbiddenErrorCode
> = ForbiddenErrorCode$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForbiddenErrorCode$ {
  /** @deprecated use `ForbiddenErrorCode$inboundSchema` instead. */
  export const inboundSchema = ForbiddenErrorCode$inboundSchema;
  /** @deprecated use `ForbiddenErrorCode$outboundSchema` instead. */
  export const outboundSchema = ForbiddenErrorCode$outboundSchema;
}

/** @internal */
export const ForbiddenError$inboundSchema: z.ZodType<
  ForbiddenError,
  z.ZodTypeDef,
  unknown
> = z.object({
  code: ForbiddenErrorCode$inboundSchema,
  message: z.string(),
});

/** @internal */
export type ForbiddenError$Outbound = {
  code: string;
  message: string;
};

/** @internal */
export const ForbiddenError$outboundSchema: z.ZodType<
  ForbiddenError$Outbound,
  z.ZodTypeDef,
  ForbiddenError
> = z.object({
  code: ForbiddenErrorCode$outboundSchema,
  message: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForbiddenError$ {
  /** @deprecated use `ForbiddenError$inboundSchema` instead. */
  export const inboundSchema = ForbiddenError$inboundSchema;
  /** @deprecated use `ForbiddenError$outboundSchema` instead. */
  export const outboundSchema = ForbiddenError$outboundSchema;
  /** @deprecated use `ForbiddenError$Outbound` instead. */
  export type Outbound = ForbiddenError$Outbound;
}

export function forbiddenErrorToJSON(forbiddenError: ForbiddenError): string {
  return JSON.stringify(ForbiddenError$outboundSchema.parse(forbiddenError));
}

export function forbiddenErrorFromJSON(
  jsonString: string,
): SafeParseResult<ForbiddenError, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ForbiddenError$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ForbiddenError' from JSON`,
  );
}

/** @internal */
export const Forbidden$inboundSchema: z.ZodType<
  Forbidden,
  z.ZodTypeDef,
  unknown
> = z.object({
  title: z.string(),
  status: z.number(),
  detail: z.string(),
  instance: z.string().optional(),
  requestId: z.string().optional(),
  code: ForbiddenCode$inboundSchema,
  error: z.lazy(() => ForbiddenError$inboundSchema),
  request$: z.instanceof(Request),
  response$: z.instanceof(Response),
  body$: z.string(),
})
  .transform((v) => {
    return new Forbidden(v, {
      request: v.request$,
      response: v.response$,
      body: v.body$,
    });
  });

/** @internal */
export type Forbidden$Outbound = {
  title: string;
  status: number;
  detail: string;
  instance?: string | undefined;
  requestId?: string | undefined;
  code: string;
  error: ForbiddenError$Outbound;
};

/** @internal */
export const Forbidden$outboundSchema: z.ZodType<
  Forbidden$Outbound,
  z.ZodTypeDef,
  Forbidden
> = z.instanceof(Forbidden)
  .transform(v => v.data$)
  .pipe(z.object({
    title: z.string(),
    status: z.number(),
    detail: z.string(),
    instance: z.string().optional(),
    requestId: z.string().optional(),
    code: ForbiddenCode$outboundSchema,
    error: z.lazy(() => ForbiddenError$outboundSchema),
  }));

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace Forbidden$ {
  /** @deprecated use `Forbidden$inboundSchema` instead. */
  export const inboundSchema = Forbidden$inboundSchema;
  /** @deprecated use `Forbidden$outboundSchema` instead. */
  export const outboundSchema = Forbidden$outboundSchema;
  /** @deprecated use `Forbidden$Outbound` instead. */
  export type Outbound = Forbidden$Outbound;
}
