// @inkeep/inkeep-analytics/dist/commonjs/react-query/conversationsLog.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useConversationsLogMutation = useConversationsLogMutation;
exports.mutationKeyConversationsLog = mutationKeyConversationsLog;
exports.buildConversationsLogMutation = buildConversationsLogMutation;
const react_query_1 = require("@tanstack/react-query");
const conversationsLog_js_1 = require("../funcs/conversationsLog.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Log Conversation
 *
 * @remarks
 * Logs a new conversation or updates an existing one with new messages. Always include all messages.
 *
 * **API Key Types:** `WEB`, `API`
 */
function useConversationsLogMutation(options) {
    const client = (0, _context_js_1.useInkeepAnalyticsContext)();
    return (0, react_query_1.useMutation)({
        ...buildConversationsLogMutation(client, options),
        ...options,
    });
}
function mutationKeyConversationsLog() {
    return ["@inkeep/inkeep-analytics", "conversations", "log"];
}
function buildConversationsLogMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyConversationsLog(),
        mutationFn: function conversationsLogMutationFn({ security, request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, conversationsLog_js_1.conversationsLog)(client$, security, request, mergedOptions));
        },
    };
}
