// @inkeep/inkeep-analytics/dist/commonjs/react-query/conversationsDelete.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useConversationsDeleteMutation = useConversationsDeleteMutation;
exports.mutationKeyConversationsDelete = mutationKeyConversationsDelete;
exports.buildConversationsDeleteMutation = buildConversationsDeleteMutation;
const react_query_1 = require("@tanstack/react-query");
const conversationsDelete_js_1 = require("../funcs/conversationsDelete.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Delete Conversation
 */
function useConversationsDeleteMutation(options) {
    const client = (0, _context_js_1.useInkeepAnalyticsContext)();
    return (0, react_query_1.useMutation)({
        ...buildConversationsDeleteMutation(client, options),
        ...options,
    });
}
function mutationKeyConversationsDelete() {
    return ["@inkeep/inkeep-analytics", "conversations", "delete"];
}
function buildConversationsDeleteMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyConversationsDelete(),
        mutationFn: function conversationsDeleteMutationFn({ security, request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, conversationsDelete_js_1.conversationsDelete)(client$, security, request, mergedOptions));
        },
    };
}
