// @inkeep/inkeep-analytics/dist/commonjs/react-query/eventsLog.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useEventsLogMutation = useEventsLogMutation;
exports.mutationKeyEventsLog = mutationKeyEventsLog;
exports.buildEventsLogMutation = buildEventsLogMutation;
const react_query_1 = require("@tanstack/react-query");
const eventsLog_js_1 = require("../funcs/eventsLog.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Log Event
 */
function useEventsLogMutation(options) {
    const client = (0, _context_js_1.useInkeepAnalyticsContext)();
    return (0, react_query_1.useMutation)({
        ...buildEventsLogMutation(client, options),
        ...options,
    });
}
function mutationKeyEventsLog() {
    return ["@inkeep/inkeep-analytics", "events", "log"];
}
function buildEventsLogMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyEventsLog(),
        mutationFn: function eventsLogMutationFn({ security, request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, eventsLog_js_1.eventsLog)(client$, security, request, mergedOptions));
        },
    };
}
