// @inkeep/inkeep-analytics/dist/commonjs/react-query/queryExportSemanticThreadsQueryResults.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useQueryExportSemanticThreadsQueryResultsMutation = useQueryExportSemanticThreadsQueryResultsMutation;
exports.mutationKeyQueryExportSemanticThreadsQueryResults = mutationKeyQueryExportSemanticThreadsQueryResults;
exports.buildQueryExportSemanticThreadsQueryResultsMutation = buildQueryExportSemanticThreadsQueryResultsMutation;
const react_query_1 = require("@tanstack/react-query");
const queryExportSemanticThreadsQueryResults_js_1 = require("../funcs/queryExportSemanticThreadsQueryResults.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Export Semantic Threads Query Results
 */
function useQueryExportSemanticThreadsQueryResultsMutation(options) {
    const client = (0, _context_js_1.useInkeepAnalyticsContext)();
    return (0, react_query_1.useMutation)({
        ...buildQueryExportSemanticThreadsQueryResultsMutation(client, options),
        ...options,
    });
}
function mutationKeyQueryExportSemanticThreadsQueryResults() {
    return [
        "@inkeep/inkeep-analytics",
        "Query",
        "exportSemanticThreadsQueryResults",
    ];
}
function buildQueryExportSemanticThreadsQueryResultsMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyQueryExportSemanticThreadsQueryResults(),
        mutationFn: function queryExportSemanticThreadsQueryResultsMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, queryExportSemanticThreadsQueryResults_js_1.queryExportSemanticThreadsQueryResults)(client$, request, mergedOptions));
        },
    };
}
