// @inkeep/inkeep-analytics/dist/commonjs/react-query/queryQueryPropertyValues.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useQueryQueryPropertyValuesMutation = useQueryQueryPropertyValuesMutation;
exports.mutationKeyQueryQueryPropertyValues = mutationKeyQueryQueryPropertyValues;
exports.buildQueryQueryPropertyValuesMutation = buildQueryQueryPropertyValuesMutation;
const react_query_1 = require("@tanstack/react-query");
const queryQueryPropertyValues_js_1 = require("../funcs/queryQueryPropertyValues.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Query Property Values
 */
function useQueryQueryPropertyValuesMutation(options) {
    const client = (0, _context_js_1.useInkeepAnalyticsContext)();
    return (0, react_query_1.useMutation)({
        ...buildQueryQueryPropertyValuesMutation(client, options),
        ...options,
    });
}
function mutationKeyQueryQueryPropertyValues() {
    return ["@inkeep/inkeep-analytics", "Query", "queryPropertyValues"];
}
function buildQueryQueryPropertyValuesMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyQueryQueryPropertyValues(),
        mutationFn: function queryQueryPropertyValuesMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, queryQueryPropertyValues_js_1.queryQueryPropertyValues)(client$, request, mergedOptions));
        },
    };
}
