// @inkeep/inkeep-analytics/dist/commonjs/react-query/queryQueryPropertyKeys.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.useQueryQueryPropertyKeysMutation = useQueryQueryPropertyKeysMutation;
exports.mutationKeyQueryQueryPropertyKeys = mutationKeyQueryQueryPropertyKeys;
exports.buildQueryQueryPropertyKeysMutation = buildQueryQueryPropertyKeysMutation;
const react_query_1 = require("@tanstack/react-query");
const queryQueryPropertyKeys_js_1 = require("../funcs/queryQueryPropertyKeys.js");
const primitives_js_1 = require("../lib/primitives.js");
const fp_js_1 = require("../types/fp.js");
const _context_js_1 = require("./_context.js");
/**
 * Query Property Keys
 */
function useQueryQueryPropertyKeysMutation(options) {
    const client = (0, _context_js_1.useInkeepAnalyticsContext)();
    return (0, react_query_1.useMutation)({
        ...buildQueryQueryPropertyKeysMutation(client, options),
        ...options,
    });
}
function mutationKeyQueryQueryPropertyKeys() {
    return ["@inkeep/inkeep-analytics", "Query", "queryPropertyKeys"];
}
function buildQueryQueryPropertyKeysMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyQueryQueryPropertyKeys(),
        mutationFn: function queryQueryPropertyKeysMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: (0, primitives_js_1.combineSignals)(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return (0, fp_js_1.unwrapAsync)((0, queryQueryPropertyKeys_js_1.queryQueryPropertyKeys)(client$, request, mergedOptions));
        },
    };
}
