// @inkeep/inkeep-analytics/src/react-query/conversationsLog.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import {
  MutationKey,
  useMutation,
  UseMutationResult,
} from "@tanstack/react-query";
import { InkeepAnalyticsCore } from "../core.js";
import { conversationsLog } from "../funcs/conversationsLog.js";
import { combineSignals } from "../lib/primitives.js";
import { RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
import { MutationHookOptions } from "./_types.js";

export type ConversationsLogMutationVariables = {
  security: operations.LogConversationSecurity;
  request: components.CreateConversation;
  options?: RequestOptions;
};

export type ConversationsLogMutationData = components.Conversation;

/**
 * Log Conversation
 *
 * @remarks
 * Logs a new conversation or updates an existing one with new messages. Always include all messages.
 *
 * **API Key Types:** `WEB`, `API`
 */
export function useConversationsLogMutation(
  options?: MutationHookOptions<
    ConversationsLogMutationData,
    Error,
    ConversationsLogMutationVariables
  >,
): UseMutationResult<
  ConversationsLogMutationData,
  Error,
  ConversationsLogMutationVariables
> {
  const client = useInkeepAnalyticsContext();
  return useMutation({
    ...buildConversationsLogMutation(client, options),
    ...options,
  });
}

export function mutationKeyConversationsLog(): MutationKey {
  return ["@inkeep/inkeep-analytics", "conversations", "log"];
}

export function buildConversationsLogMutation(
  client$: InkeepAnalyticsCore,
  hookOptions?: RequestOptions,
): {
  mutationKey: MutationKey;
  mutationFn: (
    variables: ConversationsLogMutationVariables,
  ) => Promise<ConversationsLogMutationData>;
} {
  return {
    mutationKey: mutationKeyConversationsLog(),
    mutationFn: function conversationsLogMutationFn({
      security,
      request,
      options,
    }): Promise<ConversationsLogMutationData> {
      const mergedOptions = {
        ...hookOptions,
        ...options,
        fetchOptions: {
          ...hookOptions?.fetchOptions,
          ...options?.fetchOptions,
          signal: combineSignals(
            hookOptions?.fetchOptions?.signal,
            options?.fetchOptions?.signal,
          ),
        },
      };
      return unwrapAsync(conversationsLog(
        client$,
        security,
        request,
        mergedOptions,
      ));
    },
  };
}
