// @inkeep/inkeep-analytics/src/react-query/conversationGetConversationByExternalId.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import {
  InvalidateQueryFilters,
  QueryClient,
  QueryFunctionContext,
  QueryKey,
  useQuery,
  UseQueryResult,
  useSuspenseQuery,
  UseSuspenseQueryResult,
} from "@tanstack/react-query";
import { InkeepAnalyticsCore } from "../core.js";
import { conversationGetConversationByExternalId } from "../funcs/conversationGetConversationByExternalId.js";
import { combineSignals } from "../lib/primitives.js";
import { RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
import {
  QueryHookOptions,
  SuspenseQueryHookOptions,
  TupleToPrefixes,
} from "./_types.js";

export type ConversationGetConversationByExternalIdQueryData =
  components.Conversation;

/**
 * Get Conversation by External ID
 */
export function useConversationGetConversationByExternalId(
  request: operations.GetConversationByExternalIdRequest,
  options?: QueryHookOptions<ConversationGetConversationByExternalIdQueryData>,
): UseQueryResult<ConversationGetConversationByExternalIdQueryData, Error> {
  const client = useInkeepAnalyticsContext();
  return useQuery({
    ...buildConversationGetConversationByExternalIdQuery(
      client,
      request,
      options,
    ),
    ...options,
  });
}

/**
 * Get Conversation by External ID
 */
export function useConversationGetConversationByExternalIdSuspense(
  request: operations.GetConversationByExternalIdRequest,
  options?: SuspenseQueryHookOptions<
    ConversationGetConversationByExternalIdQueryData
  >,
): UseSuspenseQueryResult<
  ConversationGetConversationByExternalIdQueryData,
  Error
> {
  const client = useInkeepAnalyticsContext();
  return useSuspenseQuery({
    ...buildConversationGetConversationByExternalIdQuery(
      client,
      request,
      options,
    ),
    ...options,
  });
}

export function prefetchConversationGetConversationByExternalId(
  queryClient: QueryClient,
  client$: InkeepAnalyticsCore,
  request: operations.GetConversationByExternalIdRequest,
): Promise<void> {
  return queryClient.prefetchQuery({
    ...buildConversationGetConversationByExternalIdQuery(
      client$,
      request,
    ),
  });
}

export function setConversationGetConversationByExternalIdData(
  client: QueryClient,
  queryKeyBase: [externalId: string],
  data: ConversationGetConversationByExternalIdQueryData,
): ConversationGetConversationByExternalIdQueryData | undefined {
  const key = queryKeyConversationGetConversationByExternalId(...queryKeyBase);

  return client.setQueryData<ConversationGetConversationByExternalIdQueryData>(
    key,
    data,
  );
}

export function invalidateConversationGetConversationByExternalId(
  client: QueryClient,
  queryKeyBase: TupleToPrefixes<[externalId: string]>,
  filters?: Omit<InvalidateQueryFilters, "queryKey" | "predicate" | "exact">,
): Promise<void> {
  return client.invalidateQueries({
    ...filters,
    queryKey: [
      "@inkeep/inkeep-analytics",
      "conversation",
      "getConversationByExternalId",
      ...queryKeyBase,
    ],
  });
}

export function invalidateAllConversationGetConversationByExternalId(
  client: QueryClient,
  filters?: Omit<InvalidateQueryFilters, "queryKey" | "predicate" | "exact">,
): Promise<void> {
  return client.invalidateQueries({
    ...filters,
    queryKey: [
      "@inkeep/inkeep-analytics",
      "conversation",
      "getConversationByExternalId",
    ],
  });
}

export function buildConversationGetConversationByExternalIdQuery(
  client$: InkeepAnalyticsCore,
  request: operations.GetConversationByExternalIdRequest,
  options?: RequestOptions,
): {
  queryKey: QueryKey;
  queryFn: (
    context: QueryFunctionContext,
  ) => Promise<ConversationGetConversationByExternalIdQueryData>;
} {
  return {
    queryKey: queryKeyConversationGetConversationByExternalId(
      request.externalId,
    ),
    queryFn: async function conversationGetConversationByExternalIdQueryFn(
      ctx,
    ): Promise<ConversationGetConversationByExternalIdQueryData> {
      const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
      const mergedOptions = {
        ...options,
        fetchOptions: { ...options?.fetchOptions, signal: sig },
      };

      return unwrapAsync(conversationGetConversationByExternalId(
        client$,
        request,
        mergedOptions,
      ));
    },
  };
}

export function queryKeyConversationGetConversationByExternalId(
  externalId: string,
): QueryKey {
  return [
    "@inkeep/inkeep-analytics",
    "conversation",
    "getConversationByExternalId",
    externalId,
  ];
}
