// @inkeep/inkeep-analytics/src/react-query/conversationsDelete.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import {
  MutationKey,
  useMutation,
  UseMutationResult,
} from "@tanstack/react-query";
import { InkeepAnalyticsCore } from "../core.js";
import { conversationsDelete } from "../funcs/conversationsDelete.js";
import { combineSignals } from "../lib/primitives.js";
import { RequestOptions } from "../lib/sdks.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
import { MutationHookOptions } from "./_types.js";

export type ConversationsDeleteMutationVariables = {
  security: operations.DeleteConversationSecurity;
  request: operations.DeleteConversationRequest;
  options?: RequestOptions;
};

export type ConversationsDeleteMutationData =
  operations.DeleteConversationResponseBody;

/**
 * Delete Conversation
 */
export function useConversationsDeleteMutation(
  options?: MutationHookOptions<
    ConversationsDeleteMutationData,
    Error,
    ConversationsDeleteMutationVariables
  >,
): UseMutationResult<
  ConversationsDeleteMutationData,
  Error,
  ConversationsDeleteMutationVariables
> {
  const client = useInkeepAnalyticsContext();
  return useMutation({
    ...buildConversationsDeleteMutation(client, options),
    ...options,
  });
}

export function mutationKeyConversationsDelete(): MutationKey {
  return ["@inkeep/inkeep-analytics", "conversations", "delete"];
}

export function buildConversationsDeleteMutation(
  client$: InkeepAnalyticsCore,
  hookOptions?: RequestOptions,
): {
  mutationKey: MutationKey;
  mutationFn: (
    variables: ConversationsDeleteMutationVariables,
  ) => Promise<ConversationsDeleteMutationData>;
} {
  return {
    mutationKey: mutationKeyConversationsDelete(),
    mutationFn: function conversationsDeleteMutationFn({
      security,
      request,
      options,
    }): Promise<ConversationsDeleteMutationData> {
      const mergedOptions = {
        ...hookOptions,
        ...options,
        fetchOptions: {
          ...hookOptions?.fetchOptions,
          ...options?.fetchOptions,
          signal: combineSignals(
            hookOptions?.fetchOptions?.signal,
            options?.fetchOptions?.signal,
          ),
        },
      };
      return unwrapAsync(conversationsDelete(
        client$,
        security,
        request,
        mergedOptions,
      ));
    },
  };
}
