// @inkeep/inkeep-analytics/src/react-query/eventsLog.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import {
  MutationKey,
  useMutation,
  UseMutationResult,
} from "@tanstack/react-query";
import { InkeepAnalyticsCore } from "../core.js";
import { eventsLog } from "../funcs/eventsLog.js";
import { combineSignals } from "../lib/primitives.js";
import { RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
import { MutationHookOptions } from "./_types.js";

export type EventsLogMutationVariables = {
  security: operations.LogEventSecurity;
  request: components.InsertEvent;
  options?: RequestOptions;
};

export type EventsLogMutationData = components.SelectEvent;

/**
 * Log Event
 */
export function useEventsLogMutation(
  options?: MutationHookOptions<
    EventsLogMutationData,
    Error,
    EventsLogMutationVariables
  >,
): UseMutationResult<EventsLogMutationData, Error, EventsLogMutationVariables> {
  const client = useInkeepAnalyticsContext();
  return useMutation({
    ...buildEventsLogMutation(client, options),
    ...options,
  });
}

export function mutationKeyEventsLog(): MutationKey {
  return ["@inkeep/inkeep-analytics", "events", "log"];
}

export function buildEventsLogMutation(
  client$: InkeepAnalyticsCore,
  hookOptions?: RequestOptions,
): {
  mutationKey: MutationKey;
  mutationFn: (
    variables: EventsLogMutationVariables,
  ) => Promise<EventsLogMutationData>;
} {
  return {
    mutationKey: mutationKeyEventsLog(),
    mutationFn: function eventsLogMutationFn({
      security,
      request,
      options,
    }): Promise<EventsLogMutationData> {
      const mergedOptions = {
        ...hookOptions,
        ...options,
        fetchOptions: {
          ...hookOptions?.fetchOptions,
          ...options?.fetchOptions,
          signal: combineSignals(
            hookOptions?.fetchOptions?.signal,
            options?.fetchOptions?.signal,
          ),
        },
      };
      return unwrapAsync(eventsLog(
        client$,
        security,
        request,
        mergedOptions,
      ));
    },
  };
}
