// @inkeep/inkeep-analytics/src/react-query/queryQueryPropertyValues.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import {
  MutationKey,
  useMutation,
  UseMutationResult,
} from "@tanstack/react-query";
import { InkeepAnalyticsCore } from "../core.js";
import { queryQueryPropertyValues } from "../funcs/queryQueryPropertyValues.js";
import { combineSignals } from "../lib/primitives.js";
import { RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
import { MutationHookOptions } from "./_types.js";

export type QueryQueryPropertyValuesMutationVariables = {
  request?: components.PropertyValuesRequestBody | undefined;
  options?: RequestOptions;
};

export type QueryQueryPropertyValuesMutationData =
  components.PropertyValuesResponseBody;

/**
 * Query Property Values
 */
export function useQueryQueryPropertyValuesMutation(
  options?: MutationHookOptions<
    QueryQueryPropertyValuesMutationData,
    Error,
    QueryQueryPropertyValuesMutationVariables
  >,
): UseMutationResult<
  QueryQueryPropertyValuesMutationData,
  Error,
  QueryQueryPropertyValuesMutationVariables
> {
  const client = useInkeepAnalyticsContext();
  return useMutation({
    ...buildQueryQueryPropertyValuesMutation(client, options),
    ...options,
  });
}

export function mutationKeyQueryQueryPropertyValues(): MutationKey {
  return ["@inkeep/inkeep-analytics", "Query", "queryPropertyValues"];
}

export function buildQueryQueryPropertyValuesMutation(
  client$: InkeepAnalyticsCore,
  hookOptions?: RequestOptions,
): {
  mutationKey: MutationKey;
  mutationFn: (
    variables: QueryQueryPropertyValuesMutationVariables,
  ) => Promise<QueryQueryPropertyValuesMutationData>;
} {
  return {
    mutationKey: mutationKeyQueryQueryPropertyValues(),
    mutationFn: function queryQueryPropertyValuesMutationFn({
      request,
      options,
    }): Promise<QueryQueryPropertyValuesMutationData> {
      const mergedOptions = {
        ...hookOptions,
        ...options,
        fetchOptions: {
          ...hookOptions?.fetchOptions,
          ...options?.fetchOptions,
          signal: combineSignals(
            hookOptions?.fetchOptions?.signal,
            options?.fetchOptions?.signal,
          ),
        },
      };
      return unwrapAsync(queryQueryPropertyValues(
        client$,
        request,
        mergedOptions,
      ));
    },
  };
}
