// @inkeep/inkeep-analytics/src/react-query/queryQuerySemanticThreads.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import {
  MutationKey,
  useMutation,
  UseMutationResult,
} from "@tanstack/react-query";
import { InkeepAnalyticsCore } from "../core.js";
import { queryQuerySemanticThreads } from "../funcs/queryQuerySemanticThreads.js";
import { combineSignals } from "../lib/primitives.js";
import { RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
import { MutationHookOptions } from "./_types.js";

export type QueryQuerySemanticThreadsMutationVariables = {
  security: operations.QuerySemanticThreadsSecurity;
  request: components.QuerySemanticThreadsRequestBody;
  options?: RequestOptions;
};

export type QueryQuerySemanticThreadsMutationData =
  components.QuerySemanticThreadsResponseBody;

/**
 * Query Semantic Threads
 */
export function useQueryQuerySemanticThreadsMutation(
  options?: MutationHookOptions<
    QueryQuerySemanticThreadsMutationData,
    Error,
    QueryQuerySemanticThreadsMutationVariables
  >,
): UseMutationResult<
  QueryQuerySemanticThreadsMutationData,
  Error,
  QueryQuerySemanticThreadsMutationVariables
> {
  const client = useInkeepAnalyticsContext();
  return useMutation({
    ...buildQueryQuerySemanticThreadsMutation(client, options),
    ...options,
  });
}

export function mutationKeyQueryQuerySemanticThreads(): MutationKey {
  return ["@inkeep/inkeep-analytics", "Query", "querySemanticThreads"];
}

export function buildQueryQuerySemanticThreadsMutation(
  client$: InkeepAnalyticsCore,
  hookOptions?: RequestOptions,
): {
  mutationKey: MutationKey;
  mutationFn: (
    variables: QueryQuerySemanticThreadsMutationVariables,
  ) => Promise<QueryQuerySemanticThreadsMutationData>;
} {
  return {
    mutationKey: mutationKeyQueryQuerySemanticThreads(),
    mutationFn: function queryQuerySemanticThreadsMutationFn({
      security,
      request,
      options,
    }): Promise<QueryQuerySemanticThreadsMutationData> {
      const mergedOptions = {
        ...hookOptions,
        ...options,
        fetchOptions: {
          ...hookOptions?.fetchOptions,
          ...options?.fetchOptions,
          signal: combineSignals(
            hookOptions?.fetchOptions?.signal,
            options?.fetchOptions?.signal,
          ),
        },
      };
      return unwrapAsync(queryQuerySemanticThreads(
        client$,
        security,
        request,
        mergedOptions,
      ));
    },
  };
}
