// @inkeep/inkeep-analytics/src/react-query/queryQueryEvents.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import {
  MutationKey,
  useMutation,
  UseMutationResult,
} from "@tanstack/react-query";
import { InkeepAnalyticsCore } from "../core.js";
import { queryQueryEvents } from "../funcs/queryQueryEvents.js";
import { combineSignals } from "../lib/primitives.js";
import { RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
import { MutationHookOptions } from "./_types.js";

export type QueryQueryEventsMutationVariables = {
  security: operations.QueryEventsSecurity;
  request: components.QueryEventsRequestBody;
  options?: RequestOptions;
};

export type QueryQueryEventsMutationData = components.QueryEventsResponseBody;

/**
 * Query Events
 */
export function useQueryQueryEventsMutation(
  options?: MutationHookOptions<
    QueryQueryEventsMutationData,
    Error,
    QueryQueryEventsMutationVariables
  >,
): UseMutationResult<
  QueryQueryEventsMutationData,
  Error,
  QueryQueryEventsMutationVariables
> {
  const client = useInkeepAnalyticsContext();
  return useMutation({
    ...buildQueryQueryEventsMutation(client, options),
    ...options,
  });
}

export function mutationKeyQueryQueryEvents(): MutationKey {
  return ["@inkeep/inkeep-analytics", "Query", "queryEvents"];
}

export function buildQueryQueryEventsMutation(
  client$: InkeepAnalyticsCore,
  hookOptions?: RequestOptions,
): {
  mutationKey: MutationKey;
  mutationFn: (
    variables: QueryQueryEventsMutationVariables,
  ) => Promise<QueryQueryEventsMutationData>;
} {
  return {
    mutationKey: mutationKeyQueryQueryEvents(),
    mutationFn: function queryQueryEventsMutationFn({
      security,
      request,
      options,
    }): Promise<QueryQueryEventsMutationData> {
      const mergedOptions = {
        ...hookOptions,
        ...options,
        fetchOptions: {
          ...hookOptions?.fetchOptions,
          ...options?.fetchOptions,
          signal: combineSignals(
            hookOptions?.fetchOptions?.signal,
            options?.fetchOptions?.signal,
          ),
        },
      };
      return unwrapAsync(queryQueryEvents(
        client$,
        security,
        request,
        mergedOptions,
      ));
    },
  };
}
