// @inkeep/inkeep-analytics/src/react-query/queryQueryFeedback.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import {
  MutationKey,
  useMutation,
  UseMutationResult,
} from "@tanstack/react-query";
import { InkeepAnalyticsCore } from "../core.js";
import { queryQueryFeedback } from "../funcs/queryQueryFeedback.js";
import { combineSignals } from "../lib/primitives.js";
import { RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
import { MutationHookOptions } from "./_types.js";

export type QueryQueryFeedbackMutationVariables = {
  security: operations.QueryFeedbackSecurity;
  request: components.QueryFeedbackRequestBody;
  options?: RequestOptions;
};

export type QueryQueryFeedbackMutationData =
  components.QueryFeedbackResponseBody;

/**
 * Query Feedback
 */
export function useQueryQueryFeedbackMutation(
  options?: MutationHookOptions<
    QueryQueryFeedbackMutationData,
    Error,
    QueryQueryFeedbackMutationVariables
  >,
): UseMutationResult<
  QueryQueryFeedbackMutationData,
  Error,
  QueryQueryFeedbackMutationVariables
> {
  const client = useInkeepAnalyticsContext();
  return useMutation({
    ...buildQueryQueryFeedbackMutation(client, options),
    ...options,
  });
}

export function mutationKeyQueryQueryFeedback(): MutationKey {
  return ["@inkeep/inkeep-analytics", "Query", "queryFeedback"];
}

export function buildQueryQueryFeedbackMutation(
  client$: InkeepAnalyticsCore,
  hookOptions?: RequestOptions,
): {
  mutationKey: MutationKey;
  mutationFn: (
    variables: QueryQueryFeedbackMutationVariables,
  ) => Promise<QueryQueryFeedbackMutationData>;
} {
  return {
    mutationKey: mutationKeyQueryQueryFeedback(),
    mutationFn: function queryQueryFeedbackMutationFn({
      security,
      request,
      options,
    }): Promise<QueryQueryFeedbackMutationData> {
      const mergedOptions = {
        ...hookOptions,
        ...options,
        fetchOptions: {
          ...hookOptions?.fetchOptions,
          ...options?.fetchOptions,
          signal: combineSignals(
            hookOptions?.fetchOptions?.signal,
            options?.fetchOptions?.signal,
          ),
        },
      };
      return unwrapAsync(queryQueryFeedback(
        client$,
        security,
        request,
        mergedOptions,
      ));
    },
  };
}
