// @inkeep/inkeep-analytics/dist/commonjs/models/errors/notfound.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotFound$ = exports.NotFound$outboundSchema = exports.NotFound$inboundSchema = exports.NotFoundError$ = exports.NotFoundError$outboundSchema = exports.NotFoundError$inboundSchema = exports.NotFoundErrorCode$ = exports.NotFoundErrorCode$outboundSchema = exports.NotFoundErrorCode$inboundSchema = exports.NotFoundCode$ = exports.NotFoundCode$outboundSchema = exports.NotFoundCode$inboundSchema = exports.NotFound = exports.NotFoundErrorCode = exports.NotFoundCode = void 0;
exports.notFoundErrorToJSON = notFoundErrorToJSON;
exports.notFoundErrorFromJSON = notFoundErrorFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const inkeepanalyticserror_js_1 = require("./inkeepanalyticserror.js");
/**
 * A short code indicating the error code returned.
 */
exports.NotFoundCode = {
    NotFound: "not_found",
};
/**
 * A short code indicating the error code returned.
 */
exports.NotFoundErrorCode = {
    NotFound: "not_found",
};
class NotFound extends inkeepanalyticserror_js_1.InkeepAnalyticsError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.title = err.title;
        this.status = err.status;
        this.detail = err.detail;
        if (err.instance != null)
            this.instance = err.instance;
        if (err.requestId != null)
            this.requestId = err.requestId;
        this.code = err.code;
        this.error = err.error;
        this.name = "NotFound";
    }
}
exports.NotFound = NotFound;
/** @internal */
exports.NotFoundCode$inboundSchema = z.nativeEnum(exports.NotFoundCode);
/** @internal */
exports.NotFoundCode$outboundSchema = exports.NotFoundCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NotFoundCode$;
(function (NotFoundCode$) {
    /** @deprecated use `NotFoundCode$inboundSchema` instead. */
    NotFoundCode$.inboundSchema = exports.NotFoundCode$inboundSchema;
    /** @deprecated use `NotFoundCode$outboundSchema` instead. */
    NotFoundCode$.outboundSchema = exports.NotFoundCode$outboundSchema;
})(NotFoundCode$ || (exports.NotFoundCode$ = NotFoundCode$ = {}));
/** @internal */
exports.NotFoundErrorCode$inboundSchema = z.nativeEnum(exports.NotFoundErrorCode);
/** @internal */
exports.NotFoundErrorCode$outboundSchema = exports.NotFoundErrorCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NotFoundErrorCode$;
(function (NotFoundErrorCode$) {
    /** @deprecated use `NotFoundErrorCode$inboundSchema` instead. */
    NotFoundErrorCode$.inboundSchema = exports.NotFoundErrorCode$inboundSchema;
    /** @deprecated use `NotFoundErrorCode$outboundSchema` instead. */
    NotFoundErrorCode$.outboundSchema = exports.NotFoundErrorCode$outboundSchema;
})(NotFoundErrorCode$ || (exports.NotFoundErrorCode$ = NotFoundErrorCode$ = {}));
/** @internal */
exports.NotFoundError$inboundSchema = z.object({
    code: exports.NotFoundErrorCode$inboundSchema,
    message: z.string(),
});
/** @internal */
exports.NotFoundError$outboundSchema = z.object({
    code: exports.NotFoundErrorCode$outboundSchema,
    message: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NotFoundError$;
(function (NotFoundError$) {
    /** @deprecated use `NotFoundError$inboundSchema` instead. */
    NotFoundError$.inboundSchema = exports.NotFoundError$inboundSchema;
    /** @deprecated use `NotFoundError$outboundSchema` instead. */
    NotFoundError$.outboundSchema = exports.NotFoundError$outboundSchema;
})(NotFoundError$ || (exports.NotFoundError$ = NotFoundError$ = {}));
function notFoundErrorToJSON(notFoundError) {
    return JSON.stringify(exports.NotFoundError$outboundSchema.parse(notFoundError));
}
function notFoundErrorFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.NotFoundError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NotFoundError' from JSON`);
}
/** @internal */
exports.NotFound$inboundSchema = z.object({
    title: z.string(),
    status: z.number(),
    detail: z.string(),
    instance: z.string().optional(),
    requestId: z.string().optional(),
    code: exports.NotFoundCode$inboundSchema,
    error: z.lazy(() => exports.NotFoundError$inboundSchema),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new NotFound(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
exports.NotFound$outboundSchema = z.instanceof(NotFound)
    .transform(v => v.data$)
    .pipe(z.object({
    title: z.string(),
    status: z.number(),
    detail: z.string(),
    instance: z.string().optional(),
    requestId: z.string().optional(),
    code: exports.NotFoundCode$outboundSchema,
    error: z.lazy(() => exports.NotFoundError$outboundSchema),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NotFound$;
(function (NotFound$) {
    /** @deprecated use `NotFound$inboundSchema` instead. */
    NotFound$.inboundSchema = exports.NotFound$inboundSchema;
    /** @deprecated use `NotFound$outboundSchema` instead. */
    NotFound$.outboundSchema = exports.NotFound$outboundSchema;
})(NotFound$ || (exports.NotFound$ = NotFound$ = {}));
