// @inkeep/inkeep-analytics/dist/commonjs/models/errors/unprocessableentity.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnprocessableEntity$ = exports.UnprocessableEntity$outboundSchema = exports.UnprocessableEntity$inboundSchema = exports.UnprocessableEntityError$ = exports.UnprocessableEntityError$outboundSchema = exports.UnprocessableEntityError$inboundSchema = exports.UnprocessableEntityErrorCode$ = exports.UnprocessableEntityErrorCode$outboundSchema = exports.UnprocessableEntityErrorCode$inboundSchema = exports.UnprocessableEntityCode$ = exports.UnprocessableEntityCode$outboundSchema = exports.UnprocessableEntityCode$inboundSchema = exports.UnprocessableEntity = exports.UnprocessableEntityErrorCode = exports.UnprocessableEntityCode = void 0;
exports.unprocessableEntityErrorToJSON = unprocessableEntityErrorToJSON;
exports.unprocessableEntityErrorFromJSON = unprocessableEntityErrorFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const inkeepanalyticserror_js_1 = require("./inkeepanalyticserror.js");
/**
 * A short code indicating the error code returned.
 */
exports.UnprocessableEntityCode = {
    UnprocessableEntity: "unprocessable_entity",
};
/**
 * A short code indicating the error code returned.
 */
exports.UnprocessableEntityErrorCode = {
    UnprocessableEntity: "unprocessable_entity",
};
class UnprocessableEntity extends inkeepanalyticserror_js_1.InkeepAnalyticsError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.title = err.title;
        this.status = err.status;
        this.detail = err.detail;
        if (err.instance != null)
            this.instance = err.instance;
        if (err.requestId != null)
            this.requestId = err.requestId;
        this.code = err.code;
        this.error = err.error;
        this.name = "UnprocessableEntity";
    }
}
exports.UnprocessableEntity = UnprocessableEntity;
/** @internal */
exports.UnprocessableEntityCode$inboundSchema = z.nativeEnum(exports.UnprocessableEntityCode);
/** @internal */
exports.UnprocessableEntityCode$outboundSchema = exports.UnprocessableEntityCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UnprocessableEntityCode$;
(function (UnprocessableEntityCode$) {
    /** @deprecated use `UnprocessableEntityCode$inboundSchema` instead. */
    UnprocessableEntityCode$.inboundSchema = exports.UnprocessableEntityCode$inboundSchema;
    /** @deprecated use `UnprocessableEntityCode$outboundSchema` instead. */
    UnprocessableEntityCode$.outboundSchema = exports.UnprocessableEntityCode$outboundSchema;
})(UnprocessableEntityCode$ || (exports.UnprocessableEntityCode$ = UnprocessableEntityCode$ = {}));
/** @internal */
exports.UnprocessableEntityErrorCode$inboundSchema = z.nativeEnum(exports.UnprocessableEntityErrorCode);
/** @internal */
exports.UnprocessableEntityErrorCode$outboundSchema = exports.UnprocessableEntityErrorCode$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UnprocessableEntityErrorCode$;
(function (UnprocessableEntityErrorCode$) {
    /** @deprecated use `UnprocessableEntityErrorCode$inboundSchema` instead. */
    UnprocessableEntityErrorCode$.inboundSchema = exports.UnprocessableEntityErrorCode$inboundSchema;
    /** @deprecated use `UnprocessableEntityErrorCode$outboundSchema` instead. */
    UnprocessableEntityErrorCode$.outboundSchema = exports.UnprocessableEntityErrorCode$outboundSchema;
})(UnprocessableEntityErrorCode$ || (exports.UnprocessableEntityErrorCode$ = UnprocessableEntityErrorCode$ = {}));
/** @internal */
exports.UnprocessableEntityError$inboundSchema = z.object({
    code: exports.UnprocessableEntityErrorCode$inboundSchema,
    message: z.string(),
});
/** @internal */
exports.UnprocessableEntityError$outboundSchema = z.object({
    code: exports.UnprocessableEntityErrorCode$outboundSchema,
    message: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UnprocessableEntityError$;
(function (UnprocessableEntityError$) {
    /** @deprecated use `UnprocessableEntityError$inboundSchema` instead. */
    UnprocessableEntityError$.inboundSchema = exports.UnprocessableEntityError$inboundSchema;
    /** @deprecated use `UnprocessableEntityError$outboundSchema` instead. */
    UnprocessableEntityError$.outboundSchema = exports.UnprocessableEntityError$outboundSchema;
})(UnprocessableEntityError$ || (exports.UnprocessableEntityError$ = UnprocessableEntityError$ = {}));
function unprocessableEntityErrorToJSON(unprocessableEntityError) {
    return JSON.stringify(exports.UnprocessableEntityError$outboundSchema.parse(unprocessableEntityError));
}
function unprocessableEntityErrorFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UnprocessableEntityError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UnprocessableEntityError' from JSON`);
}
/** @internal */
exports.UnprocessableEntity$inboundSchema = z.object({
    title: z.string(),
    status: z.number(),
    detail: z.string(),
    instance: z.string().optional(),
    requestId: z.string().optional(),
    code: exports.UnprocessableEntityCode$inboundSchema,
    error: z.lazy(() => exports.UnprocessableEntityError$inboundSchema),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new UnprocessableEntity(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
exports.UnprocessableEntity$outboundSchema = z.instanceof(UnprocessableEntity)
    .transform(v => v.data$)
    .pipe(z.object({
    title: z.string(),
    status: z.number(),
    detail: z.string(),
    instance: z.string().optional(),
    requestId: z.string().optional(),
    code: exports.UnprocessableEntityCode$outboundSchema,
    error: z.lazy(() => exports.UnprocessableEntityError$outboundSchema),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UnprocessableEntity$;
(function (UnprocessableEntity$) {
    /** @deprecated use `UnprocessableEntity$inboundSchema` instead. */
    UnprocessableEntity$.inboundSchema = exports.UnprocessableEntity$inboundSchema;
    /** @deprecated use `UnprocessableEntity$outboundSchema` instead. */
    UnprocessableEntity$.outboundSchema = exports.UnprocessableEntity$outboundSchema;
})(UnprocessableEntity$ || (exports.UnprocessableEntity$ = UnprocessableEntity$ = {}));
