// @inkeep/inkeep-analytics/dist/commonjs/sdk/query.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Query = void 0;
const queryConversations_js_1 = require("../funcs/queryConversations.js");
const queryExportSemanticThreadsQueryResults_js_1 = require("../funcs/queryExportSemanticThreadsQueryResults.js");
const queryQueryEvents_js_1 = require("../funcs/queryQueryEvents.js");
const queryQueryFeedback_js_1 = require("../funcs/queryQueryFeedback.js");
const queryQueryPropertyKeys_js_1 = require("../funcs/queryQueryPropertyKeys.js");
const queryQueryPropertyValues_js_1 = require("../funcs/queryQueryPropertyValues.js");
const queryQuerySemanticThreads_js_1 = require("../funcs/queryQuerySemanticThreads.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
class Query extends sdks_js_1.ClientSDK {
    /**
     * Query Conversations
     */
    async conversations(security, request, options) {
        return (0, fp_js_1.unwrapAsync)((0, queryConversations_js_1.queryConversations)(this, security, request, options));
    }
    /**
     * Query Events
     */
    async queryEvents(security, request, options) {
        return (0, fp_js_1.unwrapAsync)((0, queryQueryEvents_js_1.queryQueryEvents)(this, security, request, options));
    }
    /**
     * Query Feedback
     */
    async queryFeedback(security, request, options) {
        return (0, fp_js_1.unwrapAsync)((0, queryQueryFeedback_js_1.queryQueryFeedback)(this, security, request, options));
    }
    /**
     * Query Semantic Threads
     */
    async querySemanticThreads(security, request, options) {
        return (0, fp_js_1.unwrapAsync)((0, queryQuerySemanticThreads_js_1.queryQuerySemanticThreads)(this, security, request, options));
    }
    /**
     * Export Semantic Threads Query Results
     */
    async exportSemanticThreadsQueryResults(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, queryExportSemanticThreadsQueryResults_js_1.queryExportSemanticThreadsQueryResults)(this, request, options));
    }
    /**
     * Query Property Keys
     */
    async queryPropertyKeys(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, queryQueryPropertyKeys_js_1.queryQueryPropertyKeys)(this, request, options));
    }
    /**
     * Query Property Values
     */
    async queryPropertyValues(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, queryQueryPropertyValues_js_1.queryQueryPropertyValues)(this, request, options));
    }
}
exports.Query = Query;
