// @inkeep/inkeep-analytics/dist/commonjs/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InkeepAnalytics = void 0;
const topSearchQueries_js_1 = require("../funcs/topSearchQueries.js");
const weeklySearchUsers_js_1 = require("../funcs/weeklySearchUsers.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
const conversation_js_1 = require("./conversation.js");
const conversations_js_1 = require("./conversations.js");
const events_js_1 = require("./events.js");
const feedback_js_1 = require("./feedback.js");
const query_js_1 = require("./query.js");
class InkeepAnalytics extends sdks_js_1.ClientSDK {
    get conversations() {
        return (this._conversations ?? (this._conversations = new conversations_js_1.Conversations(this._options)));
    }
    get conversation() {
        return (this._conversation ?? (this._conversation = new conversation_js_1.Conversation(this._options)));
    }
    get feedback() {
        return (this._feedback ?? (this._feedback = new feedback_js_1.Feedback(this._options)));
    }
    get events() {
        return (this._events ?? (this._events = new events_js_1.Events(this._options)));
    }
    get query() {
        return (this._query ?? (this._query = new query_js_1.Query(this._options)));
    }
    /**
     * Top Search Queries
     */
    async topSearchQueries(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, topSearchQueries_js_1.topSearchQueries)(this, request, options));
    }
    /**
     * Weekly Search Users
     */
    async weeklySearchUsers(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, weeklySearchUsers_js_1.weeklySearchUsers)(this, request, options));
    }
}
exports.InkeepAnalytics = InkeepAnalytics;
