// @inkeep/inkeep-analytics/dist/commonjs/models/components/conversationssimplefieldselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversationsSimpleFieldSelection$ = exports.ConversationsSimpleFieldSelection$outboundSchema = exports.ConversationsSimpleFieldSelection$inboundSchema = exports.ConversationsSimpleFieldSelectionField$ = exports.ConversationsSimpleFieldSelectionField$outboundSchema = exports.ConversationsSimpleFieldSelectionField$inboundSchema = exports.ConversationsSimpleFieldSelectionType$ = exports.ConversationsSimpleFieldSelectionType$outboundSchema = exports.ConversationsSimpleFieldSelectionType$inboundSchema = exports.ConversationsSimpleFieldSelectionField = exports.ConversationsSimpleFieldSelectionType = void 0;
exports.conversationsSimpleFieldSelectionToJSON = conversationsSimpleFieldSelectionToJSON;
exports.conversationsSimpleFieldSelectionFromJSON = conversationsSimpleFieldSelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.ConversationsSimpleFieldSelectionType = {
    Field: "field",
};
exports.ConversationsSimpleFieldSelectionField = {
    Id: "id",
    UserMessageCount: "userMessageCount",
    OrganizationId: "organizationId",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    FirstMessageTime: "firstMessageTime",
    Type: "type",
    Properties: "properties",
    UserProperties: "userProperties",
};
/** @internal */
exports.ConversationsSimpleFieldSelectionType$inboundSchema = z.nativeEnum(exports.ConversationsSimpleFieldSelectionType);
/** @internal */
exports.ConversationsSimpleFieldSelectionType$outboundSchema = exports.ConversationsSimpleFieldSelectionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConversationsSimpleFieldSelectionType$;
(function (ConversationsSimpleFieldSelectionType$) {
    /** @deprecated use `ConversationsSimpleFieldSelectionType$inboundSchema` instead. */
    ConversationsSimpleFieldSelectionType$.inboundSchema = exports.ConversationsSimpleFieldSelectionType$inboundSchema;
    /** @deprecated use `ConversationsSimpleFieldSelectionType$outboundSchema` instead. */
    ConversationsSimpleFieldSelectionType$.outboundSchema = exports.ConversationsSimpleFieldSelectionType$outboundSchema;
})(ConversationsSimpleFieldSelectionType$ || (exports.ConversationsSimpleFieldSelectionType$ = ConversationsSimpleFieldSelectionType$ = {}));
/** @internal */
exports.ConversationsSimpleFieldSelectionField$inboundSchema = z.nativeEnum(exports.ConversationsSimpleFieldSelectionField);
/** @internal */
exports.ConversationsSimpleFieldSelectionField$outboundSchema = exports.ConversationsSimpleFieldSelectionField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConversationsSimpleFieldSelectionField$;
(function (ConversationsSimpleFieldSelectionField$) {
    /** @deprecated use `ConversationsSimpleFieldSelectionField$inboundSchema` instead. */
    ConversationsSimpleFieldSelectionField$.inboundSchema = exports.ConversationsSimpleFieldSelectionField$inboundSchema;
    /** @deprecated use `ConversationsSimpleFieldSelectionField$outboundSchema` instead. */
    ConversationsSimpleFieldSelectionField$.outboundSchema = exports.ConversationsSimpleFieldSelectionField$outboundSchema;
})(ConversationsSimpleFieldSelectionField$ || (exports.ConversationsSimpleFieldSelectionField$ = ConversationsSimpleFieldSelectionField$ = {}));
/** @internal */
exports.ConversationsSimpleFieldSelection$inboundSchema = z.object({
    type: exports.ConversationsSimpleFieldSelectionType$inboundSchema,
    field: exports.ConversationsSimpleFieldSelectionField$inboundSchema,
    path: z.array(z.string()).optional(),
});
/** @internal */
exports.ConversationsSimpleFieldSelection$outboundSchema = z.object({
    type: exports.ConversationsSimpleFieldSelectionType$outboundSchema,
    field: exports.ConversationsSimpleFieldSelectionField$outboundSchema,
    path: z.array(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConversationsSimpleFieldSelection$;
(function (ConversationsSimpleFieldSelection$) {
    /** @deprecated use `ConversationsSimpleFieldSelection$inboundSchema` instead. */
    ConversationsSimpleFieldSelection$.inboundSchema = exports.ConversationsSimpleFieldSelection$inboundSchema;
    /** @deprecated use `ConversationsSimpleFieldSelection$outboundSchema` instead. */
    ConversationsSimpleFieldSelection$.outboundSchema = exports.ConversationsSimpleFieldSelection$outboundSchema;
})(ConversationsSimpleFieldSelection$ || (exports.ConversationsSimpleFieldSelection$ = ConversationsSimpleFieldSelection$ = {}));
function conversationsSimpleFieldSelectionToJSON(conversationsSimpleFieldSelection) {
    return JSON.stringify(exports.ConversationsSimpleFieldSelection$outboundSchema.parse(conversationsSimpleFieldSelection));
}
function conversationsSimpleFieldSelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ConversationsSimpleFieldSelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ConversationsSimpleFieldSelection' from JSON`);
}
