// @inkeep/inkeep-analytics/dist/commonjs/models/components/conversation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Conversation$ = exports.Conversation$outboundSchema = exports.Conversation$inboundSchema = void 0;
exports.conversationToJSON = conversationToJSON;
exports.conversationFromJSON = conversationFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const openaiconversation_js_1 = require("./openaiconversation.js");
const supportcopilotconversation_js_1 = require("./supportcopilotconversation.js");
const supportticketconversation_js_1 = require("./supportticketconversation.js");
/** @internal */
exports.Conversation$inboundSchema = z.union([
    supportticketconversation_js_1.SupportTicketConversation$inboundSchema.and(z.object({ type: z.literal("support_ticket") }).transform((v) => ({
        type: v.type,
    }))),
    supportcopilotconversation_js_1.SupportCopilotConversation$inboundSchema.and(z.object({ type: z.literal("support_copilot") }).transform((v) => ({
        type: v.type,
    }))),
    openaiconversation_js_1.OpenAIConversation$inboundSchema.and(z.object({ type: z.literal("openai") }).transform((v) => ({
        type: v.type,
    }))),
]);
/** @internal */
exports.Conversation$outboundSchema = z.union([
    supportticketconversation_js_1.SupportTicketConversation$outboundSchema.and(z.object({ type: z.literal("support_ticket") }).transform((v) => ({
        type: v.type,
    }))),
    supportcopilotconversation_js_1.SupportCopilotConversation$outboundSchema.and(z.object({ type: z.literal("support_copilot") }).transform((v) => ({
        type: v.type,
    }))),
    openaiconversation_js_1.OpenAIConversation$outboundSchema.and(z.object({ type: z.literal("openai") }).transform((v) => ({
        type: v.type,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Conversation$;
(function (Conversation$) {
    /** @deprecated use `Conversation$inboundSchema` instead. */
    Conversation$.inboundSchema = exports.Conversation$inboundSchema;
    /** @deprecated use `Conversation$outboundSchema` instead. */
    Conversation$.outboundSchema = exports.Conversation$outboundSchema;
})(Conversation$ || (exports.Conversation$ = Conversation$ = {}));
function conversationToJSON(conversation) {
    return JSON.stringify(exports.Conversation$outboundSchema.parse(conversation));
}
function conversationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Conversation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Conversation' from JSON`);
}
