// @inkeep/inkeep-analytics/dist/commonjs/models/components/conversationsorderby.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversationsOrderBy$ = exports.ConversationsOrderBy$outboundSchema = exports.ConversationsOrderBy$inboundSchema = exports.Direction$ = exports.Direction$outboundSchema = exports.Direction$inboundSchema = exports.ConversationsOrderByField$ = exports.ConversationsOrderByField$outboundSchema = exports.ConversationsOrderByField$inboundSchema = exports.Direction = exports.ConversationsOrderByField = void 0;
exports.conversationsOrderByToJSON = conversationsOrderByToJSON;
exports.conversationsOrderByFromJSON = conversationsOrderByFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.ConversationsOrderByField = {
    Id: "id",
    UserMessageCount: "userMessageCount",
    OrganizationId: "organizationId",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    FirstMessageTime: "firstMessageTime",
    Type: "type",
    Properties: "properties",
    UserProperties: "userProperties",
    IdHour: "id_hour",
    IdDay: "id_day",
    IdWeek: "id_week",
    IdMonth: "id_month",
    UserMessageCountHour: "userMessageCount_hour",
    UserMessageCountDay: "userMessageCount_day",
    UserMessageCountWeek: "userMessageCount_week",
    UserMessageCountMonth: "userMessageCount_month",
    OrganizationIdHour: "organizationId_hour",
    OrganizationIdDay: "organizationId_day",
    OrganizationIdWeek: "organizationId_week",
    OrganizationIdMonth: "organizationId_month",
    ProjectIdHour: "projectId_hour",
    ProjectIdDay: "projectId_day",
    ProjectIdWeek: "projectId_week",
    ProjectIdMonth: "projectId_month",
    IntegrationIdHour: "integrationId_hour",
    IntegrationIdDay: "integrationId_day",
    IntegrationIdWeek: "integrationId_week",
    IntegrationIdMonth: "integrationId_month",
    FirstMessageTimeHour: "firstMessageTime_hour",
    FirstMessageTimeDay: "firstMessageTime_day",
    FirstMessageTimeWeek: "firstMessageTime_week",
    FirstMessageTimeMonth: "firstMessageTime_month",
    TypeHour: "type_hour",
    TypeDay: "type_day",
    TypeWeek: "type_week",
    TypeMonth: "type_month",
    PropertiesHour: "properties_hour",
    PropertiesDay: "properties_day",
    PropertiesWeek: "properties_week",
    PropertiesMonth: "properties_month",
    UserPropertiesHour: "userProperties_hour",
    UserPropertiesDay: "userProperties_day",
    UserPropertiesWeek: "userProperties_week",
    UserPropertiesMonth: "userProperties_month",
    SumId: "sum_id",
    SumUserMessageCount: "sum_userMessageCount",
    SumOrganizationId: "sum_organizationId",
    SumProjectId: "sum_projectId",
    SumIntegrationId: "sum_integrationId",
    SumFirstMessageTime: "sum_firstMessageTime",
    SumType: "sum_type",
    SumProperties: "sum_properties",
    SumUserProperties: "sum_userProperties",
    CountId: "count_id",
    CountUserMessageCount: "count_userMessageCount",
    CountOrganizationId: "count_organizationId",
    CountProjectId: "count_projectId",
    CountIntegrationId: "count_integrationId",
    CountFirstMessageTime: "count_firstMessageTime",
    CountType: "count_type",
    CountProperties: "count_properties",
    CountUserProperties: "count_userProperties",
    AvgId: "avg_id",
    AvgUserMessageCount: "avg_userMessageCount",
    AvgOrganizationId: "avg_organizationId",
    AvgProjectId: "avg_projectId",
    AvgIntegrationId: "avg_integrationId",
    AvgFirstMessageTime: "avg_firstMessageTime",
    AvgType: "avg_type",
    AvgProperties: "avg_properties",
    AvgUserProperties: "avg_userProperties",
    MinId: "min_id",
    MinUserMessageCount: "min_userMessageCount",
    MinOrganizationId: "min_organizationId",
    MinProjectId: "min_projectId",
    MinIntegrationId: "min_integrationId",
    MinFirstMessageTime: "min_firstMessageTime",
    MinType: "min_type",
    MinProperties: "min_properties",
    MinUserProperties: "min_userProperties",
    MaxId: "max_id",
    MaxUserMessageCount: "max_userMessageCount",
    MaxOrganizationId: "max_organizationId",
    MaxProjectId: "max_projectId",
    MaxIntegrationId: "max_integrationId",
    MaxFirstMessageTime: "max_firstMessageTime",
    MaxType: "max_type",
    MaxProperties: "max_properties",
    MaxUserProperties: "max_userProperties",
};
exports.Direction = {
    Asc: "asc",
    Desc: "desc",
};
/** @internal */
exports.ConversationsOrderByField$inboundSchema = z.nativeEnum(exports.ConversationsOrderByField);
/** @internal */
exports.ConversationsOrderByField$outboundSchema = exports.ConversationsOrderByField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConversationsOrderByField$;
(function (ConversationsOrderByField$) {
    /** @deprecated use `ConversationsOrderByField$inboundSchema` instead. */
    ConversationsOrderByField$.inboundSchema = exports.ConversationsOrderByField$inboundSchema;
    /** @deprecated use `ConversationsOrderByField$outboundSchema` instead. */
    ConversationsOrderByField$.outboundSchema = exports.ConversationsOrderByField$outboundSchema;
})(ConversationsOrderByField$ || (exports.ConversationsOrderByField$ = ConversationsOrderByField$ = {}));
/** @internal */
exports.Direction$inboundSchema = z
    .nativeEnum(exports.Direction);
/** @internal */
exports.Direction$outboundSchema = exports.Direction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Direction$;
(function (Direction$) {
    /** @deprecated use `Direction$inboundSchema` instead. */
    Direction$.inboundSchema = exports.Direction$inboundSchema;
    /** @deprecated use `Direction$outboundSchema` instead. */
    Direction$.outboundSchema = exports.Direction$outboundSchema;
})(Direction$ || (exports.Direction$ = Direction$ = {}));
/** @internal */
exports.ConversationsOrderBy$inboundSchema = z.object({
    field: exports.ConversationsOrderByField$inboundSchema,
    direction: exports.Direction$inboundSchema,
});
/** @internal */
exports.ConversationsOrderBy$outboundSchema = z.object({
    field: exports.ConversationsOrderByField$outboundSchema,
    direction: exports.Direction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConversationsOrderBy$;
(function (ConversationsOrderBy$) {
    /** @deprecated use `ConversationsOrderBy$inboundSchema` instead. */
    ConversationsOrderBy$.inboundSchema = exports.ConversationsOrderBy$inboundSchema;
    /** @deprecated use `ConversationsOrderBy$outboundSchema` instead. */
    ConversationsOrderBy$.outboundSchema = exports.ConversationsOrderBy$outboundSchema;
})(ConversationsOrderBy$ || (exports.ConversationsOrderBy$ = ConversationsOrderBy$ = {}));
function conversationsOrderByToJSON(conversationsOrderBy) {
    return JSON.stringify(exports.ConversationsOrderBy$outboundSchema.parse(conversationsOrderBy));
}
function conversationsOrderByFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ConversationsOrderBy$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ConversationsOrderBy' from JSON`);
}
