// @inkeep/inkeep-analytics/dist/commonjs/models/components/conversationsselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversationsSelection$ = exports.ConversationsSelection$outboundSchema = exports.ConversationsSelection$inboundSchema = void 0;
exports.conversationsSelectionToJSON = conversationsSelectionToJSON;
exports.conversationsSelectionFromJSON = conversationsSelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const conversationsaggregationselection_js_1 = require("./conversationsaggregationselection.js");
const conversationssimplefieldselection_js_1 = require("./conversationssimplefieldselection.js");
const conversationstimebasedgroupbyselection_js_1 = require("./conversationstimebasedgroupbyselection.js");
/** @internal */
exports.ConversationsSelection$inboundSchema = z.union([
    conversationstimebasedgroupbyselection_js_1.ConversationsTimeBasedGroupBySelection$inboundSchema.and(z.object({ type: z.literal("time") }).transform((v) => ({ type: v.type }))),
    conversationssimplefieldselection_js_1.ConversationsSimpleFieldSelection$inboundSchema.and(z.object({ type: z.literal("field") }).transform((v) => ({ type: v.type }))),
    conversationsaggregationselection_js_1.ConversationsAggregationSelection$inboundSchema.and(z.object({ type: z.literal("aggregation") }).transform((v) => ({
        type: v.type,
    }))),
]);
/** @internal */
exports.ConversationsSelection$outboundSchema = z.union([
    conversationstimebasedgroupbyselection_js_1.ConversationsTimeBasedGroupBySelection$outboundSchema.and(z.object({ type: z.literal("time") }).transform((v) => ({ type: v.type }))),
    conversationssimplefieldselection_js_1.ConversationsSimpleFieldSelection$outboundSchema.and(z.object({ type: z.literal("field") }).transform((v) => ({ type: v.type }))),
    conversationsaggregationselection_js_1.ConversationsAggregationSelection$outboundSchema.and(z.object({ type: z.literal("aggregation") }).transform((v) => ({
        type: v.type,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConversationsSelection$;
(function (ConversationsSelection$) {
    /** @deprecated use `ConversationsSelection$inboundSchema` instead. */
    ConversationsSelection$.inboundSchema = exports.ConversationsSelection$inboundSchema;
    /** @deprecated use `ConversationsSelection$outboundSchema` instead. */
    ConversationsSelection$.outboundSchema = exports.ConversationsSelection$outboundSchema;
})(ConversationsSelection$ || (exports.ConversationsSelection$ = ConversationsSelection$ = {}));
function conversationsSelectionToJSON(conversationsSelection) {
    return JSON.stringify(exports.ConversationsSelection$outboundSchema.parse(conversationsSelection));
}
function conversationsSelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ConversationsSelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ConversationsSelection' from JSON`);
}
