// @inkeep/inkeep-analytics/dist/commonjs/models/components/conversationsfield.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversationsField$ = exports.ConversationsField$outboundSchema = exports.ConversationsField$inboundSchema = exports.ConversationsField = void 0;
const z = __importStar(require("zod"));
/**
 * Available fields for Conversations
 */
exports.ConversationsField = {
    Id: "id",
    UserMessageCount: "userMessageCount",
    OrganizationId: "organizationId",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    FirstMessageTime: "firstMessageTime",
    Type: "type",
    Properties: "properties",
    UserProperties: "userProperties",
};
/** @internal */
exports.ConversationsField$inboundSchema = z.nativeEnum(exports.ConversationsField);
/** @internal */
exports.ConversationsField$outboundSchema = exports.ConversationsField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConversationsField$;
(function (ConversationsField$) {
    /** @deprecated use `ConversationsField$inboundSchema` instead. */
    ConversationsField$.inboundSchema = exports.ConversationsField$inboundSchema;
    /** @deprecated use `ConversationsField$outboundSchema` instead. */
    ConversationsField$.outboundSchema = exports.ConversationsField$outboundSchema;
})(ConversationsField$ || (exports.ConversationsField$ = ConversationsField$ = {}));
