// @inkeep/inkeep-analytics/dist/commonjs/models/components/conversationsfilteror.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversationsFilterOR$ = exports.ConversationsFilterOR$outboundSchema = exports.ConversationsFilterOR$inboundSchema = exports.Or$ = exports.Or$outboundSchema = exports.Or$inboundSchema = void 0;
exports.orToJSON = orToJSON;
exports.orFromJSON = orFromJSON;
exports.conversationsFilterORToJSON = conversationsFilterORToJSON;
exports.conversationsFilterORFromJSON = conversationsFilterORFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const conversationsfilter_js_1 = require("./conversationsfilter.js");
/** @internal */
exports.Or$inboundSchema = z.lazy(() => conversationsfilter_js_1.ConversationsFilter$inboundSchema);
/** @internal */
exports.Or$outboundSchema = z
    .lazy(() => conversationsfilter_js_1.ConversationsFilter$outboundSchema);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Or$;
(function (Or$) {
    /** @deprecated use `Or$inboundSchema` instead. */
    Or$.inboundSchema = exports.Or$inboundSchema;
    /** @deprecated use `Or$outboundSchema` instead. */
    Or$.outboundSchema = exports.Or$outboundSchema;
})(Or$ || (exports.Or$ = Or$ = {}));
function orToJSON(or) {
    return JSON.stringify(exports.Or$outboundSchema.parse(or));
}
function orFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Or$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Or' from JSON`);
}
/** @internal */
exports.ConversationsFilterOR$inboundSchema = z.object({
    or: z.array(z.lazy(() => conversationsfilter_js_1.ConversationsFilter$inboundSchema)),
});
/** @internal */
exports.ConversationsFilterOR$outboundSchema = z.object({
    or: z.array(z.lazy(() => conversationsfilter_js_1.ConversationsFilter$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ConversationsFilterOR$;
(function (ConversationsFilterOR$) {
    /** @deprecated use `ConversationsFilterOR$inboundSchema` instead. */
    ConversationsFilterOR$.inboundSchema = exports.ConversationsFilterOR$inboundSchema;
    /** @deprecated use `ConversationsFilterOR$outboundSchema` instead. */
    ConversationsFilterOR$.outboundSchema = exports.ConversationsFilterOR$outboundSchema;
})(ConversationsFilterOR$ || (exports.ConversationsFilterOR$ = ConversationsFilterOR$ = {}));
function conversationsFilterORToJSON(conversationsFilterOR) {
    return JSON.stringify(exports.ConversationsFilterOR$outboundSchema.parse(conversationsFilterOR));
}
function conversationsFilterORFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ConversationsFilterOR$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ConversationsFilterOR' from JSON`);
}
