// @inkeep/inkeep-analytics/dist/commonjs/models/components/eventsaggregationselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventsAggregationSelection$ = exports.EventsAggregationSelection$outboundSchema = exports.EventsAggregationSelection$inboundSchema = exports.EventsAggregationSelectionField$ = exports.EventsAggregationSelectionField$outboundSchema = exports.EventsAggregationSelectionField$inboundSchema = exports.EventsAggregationSelectionAggregation$ = exports.EventsAggregationSelectionAggregation$outboundSchema = exports.EventsAggregationSelectionAggregation$inboundSchema = exports.EventsAggregationSelectionType$ = exports.EventsAggregationSelectionType$outboundSchema = exports.EventsAggregationSelectionType$inboundSchema = exports.EventsAggregationSelectionField = exports.EventsAggregationSelectionAggregation = exports.EventsAggregationSelectionType = void 0;
exports.eventsAggregationSelectionToJSON = eventsAggregationSelectionToJSON;
exports.eventsAggregationSelectionFromJSON = eventsAggregationSelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.EventsAggregationSelectionType = {
    Aggregation: "aggregation",
};
exports.EventsAggregationSelectionAggregation = {
    Sum: "sum",
    Count: "count",
    CountDistinct: "countDistinct",
    Avg: "avg",
    Min: "min",
    Max: "max",
};
exports.EventsAggregationSelectionField = {
    Id: "id",
    OrganizationId: "organizationId",
    MessageId: "messageId",
    ConversationId: "conversationId",
    CreatedAt: "createdAt",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    EventType: "eventType",
    Type: "type",
    SearchQuery: "searchQuery",
    Properties: "properties",
    UserProperties: "userProperties",
};
/** @internal */
exports.EventsAggregationSelectionType$inboundSchema = z.nativeEnum(exports.EventsAggregationSelectionType);
/** @internal */
exports.EventsAggregationSelectionType$outboundSchema = exports.EventsAggregationSelectionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsAggregationSelectionType$;
(function (EventsAggregationSelectionType$) {
    /** @deprecated use `EventsAggregationSelectionType$inboundSchema` instead. */
    EventsAggregationSelectionType$.inboundSchema = exports.EventsAggregationSelectionType$inboundSchema;
    /** @deprecated use `EventsAggregationSelectionType$outboundSchema` instead. */
    EventsAggregationSelectionType$.outboundSchema = exports.EventsAggregationSelectionType$outboundSchema;
})(EventsAggregationSelectionType$ || (exports.EventsAggregationSelectionType$ = EventsAggregationSelectionType$ = {}));
/** @internal */
exports.EventsAggregationSelectionAggregation$inboundSchema = z.nativeEnum(exports.EventsAggregationSelectionAggregation);
/** @internal */
exports.EventsAggregationSelectionAggregation$outboundSchema = exports.EventsAggregationSelectionAggregation$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsAggregationSelectionAggregation$;
(function (EventsAggregationSelectionAggregation$) {
    /** @deprecated use `EventsAggregationSelectionAggregation$inboundSchema` instead. */
    EventsAggregationSelectionAggregation$.inboundSchema = exports.EventsAggregationSelectionAggregation$inboundSchema;
    /** @deprecated use `EventsAggregationSelectionAggregation$outboundSchema` instead. */
    EventsAggregationSelectionAggregation$.outboundSchema = exports.EventsAggregationSelectionAggregation$outboundSchema;
})(EventsAggregationSelectionAggregation$ || (exports.EventsAggregationSelectionAggregation$ = EventsAggregationSelectionAggregation$ = {}));
/** @internal */
exports.EventsAggregationSelectionField$inboundSchema = z.nativeEnum(exports.EventsAggregationSelectionField);
/** @internal */
exports.EventsAggregationSelectionField$outboundSchema = exports.EventsAggregationSelectionField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsAggregationSelectionField$;
(function (EventsAggregationSelectionField$) {
    /** @deprecated use `EventsAggregationSelectionField$inboundSchema` instead. */
    EventsAggregationSelectionField$.inboundSchema = exports.EventsAggregationSelectionField$inboundSchema;
    /** @deprecated use `EventsAggregationSelectionField$outboundSchema` instead. */
    EventsAggregationSelectionField$.outboundSchema = exports.EventsAggregationSelectionField$outboundSchema;
})(EventsAggregationSelectionField$ || (exports.EventsAggregationSelectionField$ = EventsAggregationSelectionField$ = {}));
/** @internal */
exports.EventsAggregationSelection$inboundSchema = z.object({
    type: exports.EventsAggregationSelectionType$inboundSchema,
    aggregation: exports.EventsAggregationSelectionAggregation$inboundSchema,
    field: exports.EventsAggregationSelectionField$inboundSchema.optional(),
    path: z.array(z.string()).optional(),
});
/** @internal */
exports.EventsAggregationSelection$outboundSchema = z.object({
    type: exports.EventsAggregationSelectionType$outboundSchema,
    aggregation: exports.EventsAggregationSelectionAggregation$outboundSchema,
    field: exports.EventsAggregationSelectionField$outboundSchema.optional(),
    path: z.array(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsAggregationSelection$;
(function (EventsAggregationSelection$) {
    /** @deprecated use `EventsAggregationSelection$inboundSchema` instead. */
    EventsAggregationSelection$.inboundSchema = exports.EventsAggregationSelection$inboundSchema;
    /** @deprecated use `EventsAggregationSelection$outboundSchema` instead. */
    EventsAggregationSelection$.outboundSchema = exports.EventsAggregationSelection$outboundSchema;
})(EventsAggregationSelection$ || (exports.EventsAggregationSelection$ = EventsAggregationSelection$ = {}));
function eventsAggregationSelectionToJSON(eventsAggregationSelection) {
    return JSON.stringify(exports.EventsAggregationSelection$outboundSchema.parse(eventsAggregationSelection));
}
function eventsAggregationSelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsAggregationSelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsAggregationSelection' from JSON`);
}
