// @inkeep/inkeep-analytics/dist/commonjs/models/components/eventstimebasedgroupbyselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventsTimeBasedGroupBySelection$ = exports.EventsTimeBasedGroupBySelection$outboundSchema = exports.EventsTimeBasedGroupBySelection$inboundSchema = exports.EventsTimeBasedGroupBySelectionField$ = exports.EventsTimeBasedGroupBySelectionField$outboundSchema = exports.EventsTimeBasedGroupBySelectionField$inboundSchema = exports.EventsTimeBasedGroupBySelectionType$ = exports.EventsTimeBasedGroupBySelectionType$outboundSchema = exports.EventsTimeBasedGroupBySelectionType$inboundSchema = exports.EventsTimeBasedGroupBySelectionField = exports.EventsTimeBasedGroupBySelectionType = void 0;
exports.eventsTimeBasedGroupBySelectionToJSON = eventsTimeBasedGroupBySelectionToJSON;
exports.eventsTimeBasedGroupBySelectionFromJSON = eventsTimeBasedGroupBySelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const timeunit_js_1 = require("./timeunit.js");
exports.EventsTimeBasedGroupBySelectionType = {
    Time: "time",
};
exports.EventsTimeBasedGroupBySelectionField = {
    Id: "id",
    OrganizationId: "organizationId",
    MessageId: "messageId",
    ConversationId: "conversationId",
    CreatedAt: "createdAt",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    EventType: "eventType",
    Type: "type",
    SearchQuery: "searchQuery",
    Properties: "properties",
    UserProperties: "userProperties",
};
/** @internal */
exports.EventsTimeBasedGroupBySelectionType$inboundSchema = z.nativeEnum(exports.EventsTimeBasedGroupBySelectionType);
/** @internal */
exports.EventsTimeBasedGroupBySelectionType$outboundSchema = exports.EventsTimeBasedGroupBySelectionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsTimeBasedGroupBySelectionType$;
(function (EventsTimeBasedGroupBySelectionType$) {
    /** @deprecated use `EventsTimeBasedGroupBySelectionType$inboundSchema` instead. */
    EventsTimeBasedGroupBySelectionType$.inboundSchema = exports.EventsTimeBasedGroupBySelectionType$inboundSchema;
    /** @deprecated use `EventsTimeBasedGroupBySelectionType$outboundSchema` instead. */
    EventsTimeBasedGroupBySelectionType$.outboundSchema = exports.EventsTimeBasedGroupBySelectionType$outboundSchema;
})(EventsTimeBasedGroupBySelectionType$ || (exports.EventsTimeBasedGroupBySelectionType$ = EventsTimeBasedGroupBySelectionType$ = {}));
/** @internal */
exports.EventsTimeBasedGroupBySelectionField$inboundSchema = z.nativeEnum(exports.EventsTimeBasedGroupBySelectionField);
/** @internal */
exports.EventsTimeBasedGroupBySelectionField$outboundSchema = exports.EventsTimeBasedGroupBySelectionField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsTimeBasedGroupBySelectionField$;
(function (EventsTimeBasedGroupBySelectionField$) {
    /** @deprecated use `EventsTimeBasedGroupBySelectionField$inboundSchema` instead. */
    EventsTimeBasedGroupBySelectionField$.inboundSchema = exports.EventsTimeBasedGroupBySelectionField$inboundSchema;
    /** @deprecated use `EventsTimeBasedGroupBySelectionField$outboundSchema` instead. */
    EventsTimeBasedGroupBySelectionField$.outboundSchema = exports.EventsTimeBasedGroupBySelectionField$outboundSchema;
})(EventsTimeBasedGroupBySelectionField$ || (exports.EventsTimeBasedGroupBySelectionField$ = EventsTimeBasedGroupBySelectionField$ = {}));
/** @internal */
exports.EventsTimeBasedGroupBySelection$inboundSchema = z.object({
    type: exports.EventsTimeBasedGroupBySelectionType$inboundSchema,
    timeUnit: timeunit_js_1.TimeUnit$inboundSchema,
    field: exports.EventsTimeBasedGroupBySelectionField$inboundSchema.optional(),
});
/** @internal */
exports.EventsTimeBasedGroupBySelection$outboundSchema = z.object({
    type: exports.EventsTimeBasedGroupBySelectionType$outboundSchema,
    timeUnit: timeunit_js_1.TimeUnit$outboundSchema,
    field: exports.EventsTimeBasedGroupBySelectionField$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsTimeBasedGroupBySelection$;
(function (EventsTimeBasedGroupBySelection$) {
    /** @deprecated use `EventsTimeBasedGroupBySelection$inboundSchema` instead. */
    EventsTimeBasedGroupBySelection$.inboundSchema = exports.EventsTimeBasedGroupBySelection$inboundSchema;
    /** @deprecated use `EventsTimeBasedGroupBySelection$outboundSchema` instead. */
    EventsTimeBasedGroupBySelection$.outboundSchema = exports.EventsTimeBasedGroupBySelection$outboundSchema;
})(EventsTimeBasedGroupBySelection$ || (exports.EventsTimeBasedGroupBySelection$ = EventsTimeBasedGroupBySelection$ = {}));
function eventsTimeBasedGroupBySelectionToJSON(eventsTimeBasedGroupBySelection) {
    return JSON.stringify(exports.EventsTimeBasedGroupBySelection$outboundSchema.parse(eventsTimeBasedGroupBySelection));
}
function eventsTimeBasedGroupBySelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsTimeBasedGroupBySelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsTimeBasedGroupBySelection' from JSON`);
}
