// @inkeep/inkeep-analytics/dist/commonjs/models/components/eventsorderby.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventsOrderBy$ = exports.EventsOrderBy$outboundSchema = exports.EventsOrderBy$inboundSchema = exports.EventsOrderByDirection$ = exports.EventsOrderByDirection$outboundSchema = exports.EventsOrderByDirection$inboundSchema = exports.EventsOrderByField$ = exports.EventsOrderByField$outboundSchema = exports.EventsOrderByField$inboundSchema = exports.EventsOrderByDirection = exports.EventsOrderByField = void 0;
exports.eventsOrderByToJSON = eventsOrderByToJSON;
exports.eventsOrderByFromJSON = eventsOrderByFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.EventsOrderByField = {
    Id: "id",
    OrganizationId: "organizationId",
    MessageId: "messageId",
    ConversationId: "conversationId",
    CreatedAt: "createdAt",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    EventType: "eventType",
    Type: "type",
    SearchQuery: "searchQuery",
    Properties: "properties",
    UserProperties: "userProperties",
    IdHour: "id_hour",
    IdDay: "id_day",
    IdWeek: "id_week",
    IdMonth: "id_month",
    OrganizationIdHour: "organizationId_hour",
    OrganizationIdDay: "organizationId_day",
    OrganizationIdWeek: "organizationId_week",
    OrganizationIdMonth: "organizationId_month",
    MessageIdHour: "messageId_hour",
    MessageIdDay: "messageId_day",
    MessageIdWeek: "messageId_week",
    MessageIdMonth: "messageId_month",
    ConversationIdHour: "conversationId_hour",
    ConversationIdDay: "conversationId_day",
    ConversationIdWeek: "conversationId_week",
    ConversationIdMonth: "conversationId_month",
    CreatedAtHour: "createdAt_hour",
    CreatedAtDay: "createdAt_day",
    CreatedAtWeek: "createdAt_week",
    CreatedAtMonth: "createdAt_month",
    ProjectIdHour: "projectId_hour",
    ProjectIdDay: "projectId_day",
    ProjectIdWeek: "projectId_week",
    ProjectIdMonth: "projectId_month",
    IntegrationIdHour: "integrationId_hour",
    IntegrationIdDay: "integrationId_day",
    IntegrationIdWeek: "integrationId_week",
    IntegrationIdMonth: "integrationId_month",
    EventTypeHour: "eventType_hour",
    EventTypeDay: "eventType_day",
    EventTypeWeek: "eventType_week",
    EventTypeMonth: "eventType_month",
    TypeHour: "type_hour",
    TypeDay: "type_day",
    TypeWeek: "type_week",
    TypeMonth: "type_month",
    SearchQueryHour: "searchQuery_hour",
    SearchQueryDay: "searchQuery_day",
    SearchQueryWeek: "searchQuery_week",
    SearchQueryMonth: "searchQuery_month",
    PropertiesHour: "properties_hour",
    PropertiesDay: "properties_day",
    PropertiesWeek: "properties_week",
    PropertiesMonth: "properties_month",
    UserPropertiesHour: "userProperties_hour",
    UserPropertiesDay: "userProperties_day",
    UserPropertiesWeek: "userProperties_week",
    UserPropertiesMonth: "userProperties_month",
    SumId: "sum_id",
    SumOrganizationId: "sum_organizationId",
    SumMessageId: "sum_messageId",
    SumConversationId: "sum_conversationId",
    SumCreatedAt: "sum_createdAt",
    SumProjectId: "sum_projectId",
    SumIntegrationId: "sum_integrationId",
    SumEventType: "sum_eventType",
    SumType: "sum_type",
    SumSearchQuery: "sum_searchQuery",
    SumProperties: "sum_properties",
    SumUserProperties: "sum_userProperties",
    CountId: "count_id",
    CountOrganizationId: "count_organizationId",
    CountMessageId: "count_messageId",
    CountConversationId: "count_conversationId",
    CountCreatedAt: "count_createdAt",
    CountProjectId: "count_projectId",
    CountIntegrationId: "count_integrationId",
    CountEventType: "count_eventType",
    CountType: "count_type",
    CountSearchQuery: "count_searchQuery",
    CountProperties: "count_properties",
    CountUserProperties: "count_userProperties",
    AvgId: "avg_id",
    AvgOrganizationId: "avg_organizationId",
    AvgMessageId: "avg_messageId",
    AvgConversationId: "avg_conversationId",
    AvgCreatedAt: "avg_createdAt",
    AvgProjectId: "avg_projectId",
    AvgIntegrationId: "avg_integrationId",
    AvgEventType: "avg_eventType",
    AvgType: "avg_type",
    AvgSearchQuery: "avg_searchQuery",
    AvgProperties: "avg_properties",
    AvgUserProperties: "avg_userProperties",
    MinId: "min_id",
    MinOrganizationId: "min_organizationId",
    MinMessageId: "min_messageId",
    MinConversationId: "min_conversationId",
    MinCreatedAt: "min_createdAt",
    MinProjectId: "min_projectId",
    MinIntegrationId: "min_integrationId",
    MinEventType: "min_eventType",
    MinType: "min_type",
    MinSearchQuery: "min_searchQuery",
    MinProperties: "min_properties",
    MinUserProperties: "min_userProperties",
    MaxId: "max_id",
    MaxOrganizationId: "max_organizationId",
    MaxMessageId: "max_messageId",
    MaxConversationId: "max_conversationId",
    MaxCreatedAt: "max_createdAt",
    MaxProjectId: "max_projectId",
    MaxIntegrationId: "max_integrationId",
    MaxEventType: "max_eventType",
    MaxType: "max_type",
    MaxSearchQuery: "max_searchQuery",
    MaxProperties: "max_properties",
    MaxUserProperties: "max_userProperties",
};
exports.EventsOrderByDirection = {
    Asc: "asc",
    Desc: "desc",
};
/** @internal */
exports.EventsOrderByField$inboundSchema = z.nativeEnum(exports.EventsOrderByField);
/** @internal */
exports.EventsOrderByField$outboundSchema = exports.EventsOrderByField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsOrderByField$;
(function (EventsOrderByField$) {
    /** @deprecated use `EventsOrderByField$inboundSchema` instead. */
    EventsOrderByField$.inboundSchema = exports.EventsOrderByField$inboundSchema;
    /** @deprecated use `EventsOrderByField$outboundSchema` instead. */
    EventsOrderByField$.outboundSchema = exports.EventsOrderByField$outboundSchema;
})(EventsOrderByField$ || (exports.EventsOrderByField$ = EventsOrderByField$ = {}));
/** @internal */
exports.EventsOrderByDirection$inboundSchema = z.nativeEnum(exports.EventsOrderByDirection);
/** @internal */
exports.EventsOrderByDirection$outboundSchema = exports.EventsOrderByDirection$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsOrderByDirection$;
(function (EventsOrderByDirection$) {
    /** @deprecated use `EventsOrderByDirection$inboundSchema` instead. */
    EventsOrderByDirection$.inboundSchema = exports.EventsOrderByDirection$inboundSchema;
    /** @deprecated use `EventsOrderByDirection$outboundSchema` instead. */
    EventsOrderByDirection$.outboundSchema = exports.EventsOrderByDirection$outboundSchema;
})(EventsOrderByDirection$ || (exports.EventsOrderByDirection$ = EventsOrderByDirection$ = {}));
/** @internal */
exports.EventsOrderBy$inboundSchema = z.object({
    field: exports.EventsOrderByField$inboundSchema,
    direction: exports.EventsOrderByDirection$inboundSchema,
});
/** @internal */
exports.EventsOrderBy$outboundSchema = z.object({
    field: exports.EventsOrderByField$outboundSchema,
    direction: exports.EventsOrderByDirection$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsOrderBy$;
(function (EventsOrderBy$) {
    /** @deprecated use `EventsOrderBy$inboundSchema` instead. */
    EventsOrderBy$.inboundSchema = exports.EventsOrderBy$inboundSchema;
    /** @deprecated use `EventsOrderBy$outboundSchema` instead. */
    EventsOrderBy$.outboundSchema = exports.EventsOrderBy$outboundSchema;
})(EventsOrderBy$ || (exports.EventsOrderBy$ = EventsOrderBy$ = {}));
function eventsOrderByToJSON(eventsOrderBy) {
    return JSON.stringify(exports.EventsOrderBy$outboundSchema.parse(eventsOrderBy));
}
function eventsOrderByFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsOrderBy$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsOrderBy' from JSON`);
}
