// @inkeep/inkeep-analytics/dist/commonjs/models/components/eventsfilteror.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventsFilterOR$ = exports.EventsFilterOR$outboundSchema = exports.EventsFilterOR$inboundSchema = exports.EventsFilterOROR$ = exports.EventsFilterOROR$outboundSchema = exports.EventsFilterOROR$inboundSchema = void 0;
exports.eventsFilterORORToJSON = eventsFilterORORToJSON;
exports.eventsFilterORORFromJSON = eventsFilterORORFromJSON;
exports.eventsFilterORToJSON = eventsFilterORToJSON;
exports.eventsFilterORFromJSON = eventsFilterORFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const eventsfilter_js_1 = require("./eventsfilter.js");
/** @internal */
exports.EventsFilterOROR$inboundSchema = z.lazy(() => eventsfilter_js_1.EventsFilter$inboundSchema);
/** @internal */
exports.EventsFilterOROR$outboundSchema = z.lazy(() => eventsfilter_js_1.EventsFilter$outboundSchema);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsFilterOROR$;
(function (EventsFilterOROR$) {
    /** @deprecated use `EventsFilterOROR$inboundSchema` instead. */
    EventsFilterOROR$.inboundSchema = exports.EventsFilterOROR$inboundSchema;
    /** @deprecated use `EventsFilterOROR$outboundSchema` instead. */
    EventsFilterOROR$.outboundSchema = exports.EventsFilterOROR$outboundSchema;
})(EventsFilterOROR$ || (exports.EventsFilterOROR$ = EventsFilterOROR$ = {}));
function eventsFilterORORToJSON(eventsFilterOROR) {
    return JSON.stringify(exports.EventsFilterOROR$outboundSchema.parse(eventsFilterOROR));
}
function eventsFilterORORFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsFilterOROR$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsFilterOROR' from JSON`);
}
/** @internal */
exports.EventsFilterOR$inboundSchema = z.object({
    or: z.array(z.lazy(() => eventsfilter_js_1.EventsFilter$inboundSchema)),
});
/** @internal */
exports.EventsFilterOR$outboundSchema = z.object({
    or: z.array(z.lazy(() => eventsfilter_js_1.EventsFilter$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventsFilterOR$;
(function (EventsFilterOR$) {
    /** @deprecated use `EventsFilterOR$inboundSchema` instead. */
    EventsFilterOR$.inboundSchema = exports.EventsFilterOR$inboundSchema;
    /** @deprecated use `EventsFilterOR$outboundSchema` instead. */
    EventsFilterOR$.outboundSchema = exports.EventsFilterOR$outboundSchema;
})(EventsFilterOR$ || (exports.EventsFilterOR$ = EventsFilterOR$ = {}));
function eventsFilterORToJSON(eventsFilterOR) {
    return JSON.stringify(exports.EventsFilterOR$outboundSchema.parse(eventsFilterOR));
}
function eventsFilterORFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.EventsFilterOR$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsFilterOR' from JSON`);
}
