// @inkeep/inkeep-analytics/dist/commonjs/models/components/feedbackgroupby.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedbackGroupBy$ = exports.FeedbackGroupBy$outboundSchema = exports.FeedbackGroupBy$inboundSchema = void 0;
exports.feedbackGroupByToJSON = feedbackGroupByToJSON;
exports.feedbackGroupByFromJSON = feedbackGroupByFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const feedbackfield_js_1 = require("./feedbackfield.js");
/** @internal */
exports.FeedbackGroupBy$inboundSchema = z.object({
    field: feedbackfield_js_1.FeedbackField$inboundSchema,
    path: z.array(z.string()).optional(),
    includeInSelect: z.boolean().default(true),
});
/** @internal */
exports.FeedbackGroupBy$outboundSchema = z.object({
    field: feedbackfield_js_1.FeedbackField$outboundSchema,
    path: z.array(z.string()).optional(),
    includeInSelect: z.boolean().default(true),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackGroupBy$;
(function (FeedbackGroupBy$) {
    /** @deprecated use `FeedbackGroupBy$inboundSchema` instead. */
    FeedbackGroupBy$.inboundSchema = exports.FeedbackGroupBy$inboundSchema;
    /** @deprecated use `FeedbackGroupBy$outboundSchema` instead. */
    FeedbackGroupBy$.outboundSchema = exports.FeedbackGroupBy$outboundSchema;
})(FeedbackGroupBy$ || (exports.FeedbackGroupBy$ = FeedbackGroupBy$ = {}));
function feedbackGroupByToJSON(feedbackGroupBy) {
    return JSON.stringify(exports.FeedbackGroupBy$outboundSchema.parse(feedbackGroupBy));
}
function feedbackGroupByFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeedbackGroupBy$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackGroupBy' from JSON`);
}
