// @inkeep/inkeep-analytics/dist/commonjs/models/components/feedbackorderby.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedbackOrderBy$ = exports.FeedbackOrderBy$outboundSchema = exports.FeedbackOrderBy$inboundSchema = exports.FeedbackOrderByDirection$ = exports.FeedbackOrderByDirection$outboundSchema = exports.FeedbackOrderByDirection$inboundSchema = exports.FeedbackOrderByField$ = exports.FeedbackOrderByField$outboundSchema = exports.FeedbackOrderByField$inboundSchema = exports.FeedbackOrderByDirection = exports.FeedbackOrderByField = void 0;
exports.feedbackOrderByToJSON = feedbackOrderByToJSON;
exports.feedbackOrderByFromJSON = feedbackOrderByFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.FeedbackOrderByField = {
    Id: "id",
    Type: "type",
    MessageId: "messageId",
    CreatedAt: "createdAt",
    OrganizationId: "organizationId",
    Reasons: "reasons",
    Properties: "properties",
    UserProperties: "userProperties",
    ConversationId: "conversationId",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    UpdatedAt: "updatedAt",
    Status: "status",
    IdHour: "id_hour",
    IdDay: "id_day",
    IdWeek: "id_week",
    IdMonth: "id_month",
    TypeHour: "type_hour",
    TypeDay: "type_day",
    TypeWeek: "type_week",
    TypeMonth: "type_month",
    MessageIdHour: "messageId_hour",
    MessageIdDay: "messageId_day",
    MessageIdWeek: "messageId_week",
    MessageIdMonth: "messageId_month",
    CreatedAtHour: "createdAt_hour",
    CreatedAtDay: "createdAt_day",
    CreatedAtWeek: "createdAt_week",
    CreatedAtMonth: "createdAt_month",
    OrganizationIdHour: "organizationId_hour",
    OrganizationIdDay: "organizationId_day",
    OrganizationIdWeek: "organizationId_week",
    OrganizationIdMonth: "organizationId_month",
    ReasonsHour: "reasons_hour",
    ReasonsDay: "reasons_day",
    ReasonsWeek: "reasons_week",
    ReasonsMonth: "reasons_month",
    PropertiesHour: "properties_hour",
    PropertiesDay: "properties_day",
    PropertiesWeek: "properties_week",
    PropertiesMonth: "properties_month",
    UserPropertiesHour: "userProperties_hour",
    UserPropertiesDay: "userProperties_day",
    UserPropertiesWeek: "userProperties_week",
    UserPropertiesMonth: "userProperties_month",
    ConversationIdHour: "conversationId_hour",
    ConversationIdDay: "conversationId_day",
    ConversationIdWeek: "conversationId_week",
    ConversationIdMonth: "conversationId_month",
    ProjectIdHour: "projectId_hour",
    ProjectIdDay: "projectId_day",
    ProjectIdWeek: "projectId_week",
    ProjectIdMonth: "projectId_month",
    IntegrationIdHour: "integrationId_hour",
    IntegrationIdDay: "integrationId_day",
    IntegrationIdWeek: "integrationId_week",
    IntegrationIdMonth: "integrationId_month",
    UpdatedAtHour: "updatedAt_hour",
    UpdatedAtDay: "updatedAt_day",
    UpdatedAtWeek: "updatedAt_week",
    UpdatedAtMonth: "updatedAt_month",
    StatusHour: "status_hour",
    StatusDay: "status_day",
    StatusWeek: "status_week",
    StatusMonth: "status_month",
    SumId: "sum_id",
    SumType: "sum_type",
    SumMessageId: "sum_messageId",
    SumCreatedAt: "sum_createdAt",
    SumOrganizationId: "sum_organizationId",
    SumReasons: "sum_reasons",
    SumProperties: "sum_properties",
    SumUserProperties: "sum_userProperties",
    SumConversationId: "sum_conversationId",
    SumProjectId: "sum_projectId",
    SumIntegrationId: "sum_integrationId",
    SumUpdatedAt: "sum_updatedAt",
    SumStatus: "sum_status",
    CountId: "count_id",
    CountType: "count_type",
    CountMessageId: "count_messageId",
    CountCreatedAt: "count_createdAt",
    CountOrganizationId: "count_organizationId",
    CountReasons: "count_reasons",
    CountProperties: "count_properties",
    CountUserProperties: "count_userProperties",
    CountConversationId: "count_conversationId",
    CountProjectId: "count_projectId",
    CountIntegrationId: "count_integrationId",
    CountUpdatedAt: "count_updatedAt",
    CountStatus: "count_status",
    AvgId: "avg_id",
    AvgType: "avg_type",
    AvgMessageId: "avg_messageId",
    AvgCreatedAt: "avg_createdAt",
    AvgOrganizationId: "avg_organizationId",
    AvgReasons: "avg_reasons",
    AvgProperties: "avg_properties",
    AvgUserProperties: "avg_userProperties",
    AvgConversationId: "avg_conversationId",
    AvgProjectId: "avg_projectId",
    AvgIntegrationId: "avg_integrationId",
    AvgUpdatedAt: "avg_updatedAt",
    AvgStatus: "avg_status",
    MinId: "min_id",
    MinType: "min_type",
    MinMessageId: "min_messageId",
    MinCreatedAt: "min_createdAt",
    MinOrganizationId: "min_organizationId",
    MinReasons: "min_reasons",
    MinProperties: "min_properties",
    MinUserProperties: "min_userProperties",
    MinConversationId: "min_conversationId",
    MinProjectId: "min_projectId",
    MinIntegrationId: "min_integrationId",
    MinUpdatedAt: "min_updatedAt",
    MinStatus: "min_status",
    MaxId: "max_id",
    MaxType: "max_type",
    MaxMessageId: "max_messageId",
    MaxCreatedAt: "max_createdAt",
    MaxOrganizationId: "max_organizationId",
    MaxReasons: "max_reasons",
    MaxProperties: "max_properties",
    MaxUserProperties: "max_userProperties",
    MaxConversationId: "max_conversationId",
    MaxProjectId: "max_projectId",
    MaxIntegrationId: "max_integrationId",
    MaxUpdatedAt: "max_updatedAt",
    MaxStatus: "max_status",
};
exports.FeedbackOrderByDirection = {
    Asc: "asc",
    Desc: "desc",
};
/** @internal */
exports.FeedbackOrderByField$inboundSchema = z.nativeEnum(exports.FeedbackOrderByField);
/** @internal */
exports.FeedbackOrderByField$outboundSchema = exports.FeedbackOrderByField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackOrderByField$;
(function (FeedbackOrderByField$) {
    /** @deprecated use `FeedbackOrderByField$inboundSchema` instead. */
    FeedbackOrderByField$.inboundSchema = exports.FeedbackOrderByField$inboundSchema;
    /** @deprecated use `FeedbackOrderByField$outboundSchema` instead. */
    FeedbackOrderByField$.outboundSchema = exports.FeedbackOrderByField$outboundSchema;
})(FeedbackOrderByField$ || (exports.FeedbackOrderByField$ = FeedbackOrderByField$ = {}));
/** @internal */
exports.FeedbackOrderByDirection$inboundSchema = z.nativeEnum(exports.FeedbackOrderByDirection);
/** @internal */
exports.FeedbackOrderByDirection$outboundSchema = exports.FeedbackOrderByDirection$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackOrderByDirection$;
(function (FeedbackOrderByDirection$) {
    /** @deprecated use `FeedbackOrderByDirection$inboundSchema` instead. */
    FeedbackOrderByDirection$.inboundSchema = exports.FeedbackOrderByDirection$inboundSchema;
    /** @deprecated use `FeedbackOrderByDirection$outboundSchema` instead. */
    FeedbackOrderByDirection$.outboundSchema = exports.FeedbackOrderByDirection$outboundSchema;
})(FeedbackOrderByDirection$ || (exports.FeedbackOrderByDirection$ = FeedbackOrderByDirection$ = {}));
/** @internal */
exports.FeedbackOrderBy$inboundSchema = z.object({
    field: exports.FeedbackOrderByField$inboundSchema,
    direction: exports.FeedbackOrderByDirection$inboundSchema,
});
/** @internal */
exports.FeedbackOrderBy$outboundSchema = z.object({
    field: exports.FeedbackOrderByField$outboundSchema,
    direction: exports.FeedbackOrderByDirection$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackOrderBy$;
(function (FeedbackOrderBy$) {
    /** @deprecated use `FeedbackOrderBy$inboundSchema` instead. */
    FeedbackOrderBy$.inboundSchema = exports.FeedbackOrderBy$inboundSchema;
    /** @deprecated use `FeedbackOrderBy$outboundSchema` instead. */
    FeedbackOrderBy$.outboundSchema = exports.FeedbackOrderBy$outboundSchema;
})(FeedbackOrderBy$ || (exports.FeedbackOrderBy$ = FeedbackOrderBy$ = {}));
function feedbackOrderByToJSON(feedbackOrderBy) {
    return JSON.stringify(exports.FeedbackOrderBy$outboundSchema.parse(feedbackOrderBy));
}
function feedbackOrderByFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeedbackOrderBy$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackOrderBy' from JSON`);
}
