// @inkeep/inkeep-analytics/dist/commonjs/models/components/feedbackfilteror.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedbackFilterOR$ = exports.FeedbackFilterOR$outboundSchema = exports.FeedbackFilterOR$inboundSchema = exports.FeedbackFilterOROR$ = exports.FeedbackFilterOROR$outboundSchema = exports.FeedbackFilterOROR$inboundSchema = void 0;
exports.feedbackFilterORORToJSON = feedbackFilterORORToJSON;
exports.feedbackFilterORORFromJSON = feedbackFilterORORFromJSON;
exports.feedbackFilterORToJSON = feedbackFilterORToJSON;
exports.feedbackFilterORFromJSON = feedbackFilterORFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const feedbackfilter_js_1 = require("./feedbackfilter.js");
/** @internal */
exports.FeedbackFilterOROR$inboundSchema = z.lazy(() => feedbackfilter_js_1.FeedbackFilter$inboundSchema);
/** @internal */
exports.FeedbackFilterOROR$outboundSchema = z.lazy(() => feedbackfilter_js_1.FeedbackFilter$outboundSchema);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackFilterOROR$;
(function (FeedbackFilterOROR$) {
    /** @deprecated use `FeedbackFilterOROR$inboundSchema` instead. */
    FeedbackFilterOROR$.inboundSchema = exports.FeedbackFilterOROR$inboundSchema;
    /** @deprecated use `FeedbackFilterOROR$outboundSchema` instead. */
    FeedbackFilterOROR$.outboundSchema = exports.FeedbackFilterOROR$outboundSchema;
})(FeedbackFilterOROR$ || (exports.FeedbackFilterOROR$ = FeedbackFilterOROR$ = {}));
function feedbackFilterORORToJSON(feedbackFilterOROR) {
    return JSON.stringify(exports.FeedbackFilterOROR$outboundSchema.parse(feedbackFilterOROR));
}
function feedbackFilterORORFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeedbackFilterOROR$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackFilterOROR' from JSON`);
}
/** @internal */
exports.FeedbackFilterOR$inboundSchema = z.object({
    or: z.array(z.lazy(() => feedbackfilter_js_1.FeedbackFilter$inboundSchema)),
});
/** @internal */
exports.FeedbackFilterOR$outboundSchema = z.object({
    or: z.array(z.lazy(() => feedbackfilter_js_1.FeedbackFilter$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackFilterOR$;
(function (FeedbackFilterOR$) {
    /** @deprecated use `FeedbackFilterOR$inboundSchema` instead. */
    FeedbackFilterOR$.inboundSchema = exports.FeedbackFilterOR$inboundSchema;
    /** @deprecated use `FeedbackFilterOR$outboundSchema` instead. */
    FeedbackFilterOR$.outboundSchema = exports.FeedbackFilterOR$outboundSchema;
})(FeedbackFilterOR$ || (exports.FeedbackFilterOR$ = FeedbackFilterOR$ = {}));
function feedbackFilterORToJSON(feedbackFilterOR) {
    return JSON.stringify(exports.FeedbackFilterOR$outboundSchema.parse(feedbackFilterOR));
}
function feedbackFilterORFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeedbackFilterOR$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackFilterOR' from JSON`);
}
