// @inkeep/inkeep-analytics/dist/commonjs/models/components/feedbackfilterand.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedbackFilterAND$ = exports.FeedbackFilterAND$outboundSchema = exports.FeedbackFilterAND$inboundSchema = exports.FeedbackFilterANDAND$ = exports.FeedbackFilterANDAND$outboundSchema = exports.FeedbackFilterANDAND$inboundSchema = void 0;
exports.feedbackFilterANDANDToJSON = feedbackFilterANDANDToJSON;
exports.feedbackFilterANDANDFromJSON = feedbackFilterANDANDFromJSON;
exports.feedbackFilterANDToJSON = feedbackFilterANDToJSON;
exports.feedbackFilterANDFromJSON = feedbackFilterANDFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const feedbackfilter_js_1 = require("./feedbackfilter.js");
/** @internal */
exports.FeedbackFilterANDAND$inboundSchema = z.lazy(() => feedbackfilter_js_1.FeedbackFilter$inboundSchema);
/** @internal */
exports.FeedbackFilterANDAND$outboundSchema = z.lazy(() => feedbackfilter_js_1.FeedbackFilter$outboundSchema);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackFilterANDAND$;
(function (FeedbackFilterANDAND$) {
    /** @deprecated use `FeedbackFilterANDAND$inboundSchema` instead. */
    FeedbackFilterANDAND$.inboundSchema = exports.FeedbackFilterANDAND$inboundSchema;
    /** @deprecated use `FeedbackFilterANDAND$outboundSchema` instead. */
    FeedbackFilterANDAND$.outboundSchema = exports.FeedbackFilterANDAND$outboundSchema;
})(FeedbackFilterANDAND$ || (exports.FeedbackFilterANDAND$ = FeedbackFilterANDAND$ = {}));
function feedbackFilterANDANDToJSON(feedbackFilterANDAND) {
    return JSON.stringify(exports.FeedbackFilterANDAND$outboundSchema.parse(feedbackFilterANDAND));
}
function feedbackFilterANDANDFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeedbackFilterANDAND$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackFilterANDAND' from JSON`);
}
/** @internal */
exports.FeedbackFilterAND$inboundSchema = z.object({
    and: z.array(z.lazy(() => feedbackfilter_js_1.FeedbackFilter$inboundSchema)),
});
/** @internal */
exports.FeedbackFilterAND$outboundSchema = z.object({
    and: z.array(z.lazy(() => feedbackfilter_js_1.FeedbackFilter$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackFilterAND$;
(function (FeedbackFilterAND$) {
    /** @deprecated use `FeedbackFilterAND$inboundSchema` instead. */
    FeedbackFilterAND$.inboundSchema = exports.FeedbackFilterAND$inboundSchema;
    /** @deprecated use `FeedbackFilterAND$outboundSchema` instead. */
    FeedbackFilterAND$.outboundSchema = exports.FeedbackFilterAND$outboundSchema;
})(FeedbackFilterAND$ || (exports.FeedbackFilterAND$ = FeedbackFilterAND$ = {}));
function feedbackFilterANDToJSON(feedbackFilterAND) {
    return JSON.stringify(exports.FeedbackFilterAND$outboundSchema.parse(feedbackFilterAND));
}
function feedbackFilterANDFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeedbackFilterAND$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackFilterAND' from JSON`);
}
