// @inkeep/inkeep-analytics/dist/commonjs/models/components/feedbackaggregationselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedbackAggregationSelection$ = exports.FeedbackAggregationSelection$outboundSchema = exports.FeedbackAggregationSelection$inboundSchema = exports.FeedbackAggregationSelectionField$ = exports.FeedbackAggregationSelectionField$outboundSchema = exports.FeedbackAggregationSelectionField$inboundSchema = exports.FeedbackAggregationSelectionAggregation$ = exports.FeedbackAggregationSelectionAggregation$outboundSchema = exports.FeedbackAggregationSelectionAggregation$inboundSchema = exports.FeedbackAggregationSelectionType$ = exports.FeedbackAggregationSelectionType$outboundSchema = exports.FeedbackAggregationSelectionType$inboundSchema = exports.FeedbackAggregationSelectionField = exports.FeedbackAggregationSelectionAggregation = exports.FeedbackAggregationSelectionType = void 0;
exports.feedbackAggregationSelectionToJSON = feedbackAggregationSelectionToJSON;
exports.feedbackAggregationSelectionFromJSON = feedbackAggregationSelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.FeedbackAggregationSelectionType = {
    Aggregation: "aggregation",
};
exports.FeedbackAggregationSelectionAggregation = {
    Sum: "sum",
    Count: "count",
    CountDistinct: "countDistinct",
    Avg: "avg",
    Min: "min",
    Max: "max",
};
exports.FeedbackAggregationSelectionField = {
    Id: "id",
    Type: "type",
    MessageId: "messageId",
    CreatedAt: "createdAt",
    OrganizationId: "organizationId",
    Reasons: "reasons",
    Properties: "properties",
    UserProperties: "userProperties",
    ConversationId: "conversationId",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    UpdatedAt: "updatedAt",
    Status: "status",
};
/** @internal */
exports.FeedbackAggregationSelectionType$inboundSchema = z.nativeEnum(exports.FeedbackAggregationSelectionType);
/** @internal */
exports.FeedbackAggregationSelectionType$outboundSchema = exports.FeedbackAggregationSelectionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackAggregationSelectionType$;
(function (FeedbackAggregationSelectionType$) {
    /** @deprecated use `FeedbackAggregationSelectionType$inboundSchema` instead. */
    FeedbackAggregationSelectionType$.inboundSchema = exports.FeedbackAggregationSelectionType$inboundSchema;
    /** @deprecated use `FeedbackAggregationSelectionType$outboundSchema` instead. */
    FeedbackAggregationSelectionType$.outboundSchema = exports.FeedbackAggregationSelectionType$outboundSchema;
})(FeedbackAggregationSelectionType$ || (exports.FeedbackAggregationSelectionType$ = FeedbackAggregationSelectionType$ = {}));
/** @internal */
exports.FeedbackAggregationSelectionAggregation$inboundSchema = z
    .nativeEnum(exports.FeedbackAggregationSelectionAggregation);
/** @internal */
exports.FeedbackAggregationSelectionAggregation$outboundSchema = exports.FeedbackAggregationSelectionAggregation$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackAggregationSelectionAggregation$;
(function (FeedbackAggregationSelectionAggregation$) {
    /** @deprecated use `FeedbackAggregationSelectionAggregation$inboundSchema` instead. */
    FeedbackAggregationSelectionAggregation$.inboundSchema = exports.FeedbackAggregationSelectionAggregation$inboundSchema;
    /** @deprecated use `FeedbackAggregationSelectionAggregation$outboundSchema` instead. */
    FeedbackAggregationSelectionAggregation$.outboundSchema = exports.FeedbackAggregationSelectionAggregation$outboundSchema;
})(FeedbackAggregationSelectionAggregation$ || (exports.FeedbackAggregationSelectionAggregation$ = FeedbackAggregationSelectionAggregation$ = {}));
/** @internal */
exports.FeedbackAggregationSelectionField$inboundSchema = z.nativeEnum(exports.FeedbackAggregationSelectionField);
/** @internal */
exports.FeedbackAggregationSelectionField$outboundSchema = exports.FeedbackAggregationSelectionField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackAggregationSelectionField$;
(function (FeedbackAggregationSelectionField$) {
    /** @deprecated use `FeedbackAggregationSelectionField$inboundSchema` instead. */
    FeedbackAggregationSelectionField$.inboundSchema = exports.FeedbackAggregationSelectionField$inboundSchema;
    /** @deprecated use `FeedbackAggregationSelectionField$outboundSchema` instead. */
    FeedbackAggregationSelectionField$.outboundSchema = exports.FeedbackAggregationSelectionField$outboundSchema;
})(FeedbackAggregationSelectionField$ || (exports.FeedbackAggregationSelectionField$ = FeedbackAggregationSelectionField$ = {}));
/** @internal */
exports.FeedbackAggregationSelection$inboundSchema = z.object({
    type: exports.FeedbackAggregationSelectionType$inboundSchema,
    aggregation: exports.FeedbackAggregationSelectionAggregation$inboundSchema,
    field: exports.FeedbackAggregationSelectionField$inboundSchema.optional(),
    path: z.array(z.string()).optional(),
});
/** @internal */
exports.FeedbackAggregationSelection$outboundSchema = z.object({
    type: exports.FeedbackAggregationSelectionType$outboundSchema,
    aggregation: exports.FeedbackAggregationSelectionAggregation$outboundSchema,
    field: exports.FeedbackAggregationSelectionField$outboundSchema.optional(),
    path: z.array(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackAggregationSelection$;
(function (FeedbackAggregationSelection$) {
    /** @deprecated use `FeedbackAggregationSelection$inboundSchema` instead. */
    FeedbackAggregationSelection$.inboundSchema = exports.FeedbackAggregationSelection$inboundSchema;
    /** @deprecated use `FeedbackAggregationSelection$outboundSchema` instead. */
    FeedbackAggregationSelection$.outboundSchema = exports.FeedbackAggregationSelection$outboundSchema;
})(FeedbackAggregationSelection$ || (exports.FeedbackAggregationSelection$ = FeedbackAggregationSelection$ = {}));
function feedbackAggregationSelectionToJSON(feedbackAggregationSelection) {
    return JSON.stringify(exports.FeedbackAggregationSelection$outboundSchema.parse(feedbackAggregationSelection));
}
function feedbackAggregationSelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeedbackAggregationSelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackAggregationSelection' from JSON`);
}
