// @inkeep/inkeep-analytics/dist/commonjs/models/components/feedbackselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeedbackSelection$ = exports.FeedbackSelection$outboundSchema = exports.FeedbackSelection$inboundSchema = void 0;
exports.feedbackSelectionToJSON = feedbackSelectionToJSON;
exports.feedbackSelectionFromJSON = feedbackSelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const feedbackaggregationselection_js_1 = require("./feedbackaggregationselection.js");
const feedbacksimplefieldselection_js_1 = require("./feedbacksimplefieldselection.js");
const feedbacktimebasedgroupbyselection_js_1 = require("./feedbacktimebasedgroupbyselection.js");
/** @internal */
exports.FeedbackSelection$inboundSchema = z.union([
    feedbacktimebasedgroupbyselection_js_1.FeedbackTimeBasedGroupBySelection$inboundSchema.and(z.object({ type: z.literal("time") }).transform((v) => ({ type: v.type }))),
    feedbacksimplefieldselection_js_1.FeedbackSimpleFieldSelection$inboundSchema.and(z.object({ type: z.literal("field") }).transform((v) => ({ type: v.type }))),
    feedbackaggregationselection_js_1.FeedbackAggregationSelection$inboundSchema.and(z.object({ type: z.literal("aggregation") }).transform((v) => ({
        type: v.type,
    }))),
]);
/** @internal */
exports.FeedbackSelection$outboundSchema = z.union([
    feedbacktimebasedgroupbyselection_js_1.FeedbackTimeBasedGroupBySelection$outboundSchema.and(z.object({ type: z.literal("time") }).transform((v) => ({ type: v.type }))),
    feedbacksimplefieldselection_js_1.FeedbackSimpleFieldSelection$outboundSchema.and(z.object({ type: z.literal("field") }).transform((v) => ({ type: v.type }))),
    feedbackaggregationselection_js_1.FeedbackAggregationSelection$outboundSchema.and(z.object({ type: z.literal("aggregation") }).transform((v) => ({
        type: v.type,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeedbackSelection$;
(function (FeedbackSelection$) {
    /** @deprecated use `FeedbackSelection$inboundSchema` instead. */
    FeedbackSelection$.inboundSchema = exports.FeedbackSelection$inboundSchema;
    /** @deprecated use `FeedbackSelection$outboundSchema` instead. */
    FeedbackSelection$.outboundSchema = exports.FeedbackSelection$outboundSchema;
})(FeedbackSelection$ || (exports.FeedbackSelection$ = FeedbackSelection$ = {}));
function feedbackSelectionToJSON(feedbackSelection) {
    return JSON.stringify(exports.FeedbackSelection$outboundSchema.parse(feedbackSelection));
}
function feedbackSelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FeedbackSelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackSelection' from JSON`);
}
