// @inkeep/inkeep-analytics/dist/commonjs/models/components/insertevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InsertEvent$ = exports.InsertEvent$outboundSchema = exports.InsertEvent$inboundSchema = void 0;
exports.insertEventToJSON = insertEventToJSON;
exports.insertEventFromJSON = insertEventFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const insertconversationevent_js_1 = require("./insertconversationevent.js");
const insertmessageevent_js_1 = require("./insertmessageevent.js");
const insertsearchevent_js_1 = require("./insertsearchevent.js");
/** @internal */
exports.InsertEvent$inboundSchema = z.union([
    insertmessageevent_js_1.InsertMessageEvent$inboundSchema.and(z.object({ entityType: z.literal("message") }).transform((v) => ({
        entityType: v.entityType,
    }))),
    insertconversationevent_js_1.InsertConversationEvent$inboundSchema.and(z.object({ entityType: z.literal("conversation") }).transform((v) => ({
        entityType: v.entityType,
    }))),
    insertsearchevent_js_1.InsertSearchEvent$inboundSchema.and(z.object({ entityType: z.literal("search") }).transform((v) => ({
        entityType: v.entityType,
    }))),
]);
/** @internal */
exports.InsertEvent$outboundSchema = z.union([
    insertmessageevent_js_1.InsertMessageEvent$outboundSchema.and(z.object({ entityType: z.literal("message") }).transform((v) => ({
        entityType: v.entityType,
    }))),
    insertconversationevent_js_1.InsertConversationEvent$outboundSchema.and(z.object({ entityType: z.literal("conversation") }).transform((v) => ({
        entityType: v.entityType,
    }))),
    insertsearchevent_js_1.InsertSearchEvent$outboundSchema.and(z.object({ entityType: z.literal("search") }).transform((v) => ({
        entityType: v.entityType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InsertEvent$;
(function (InsertEvent$) {
    /** @deprecated use `InsertEvent$inboundSchema` instead. */
    InsertEvent$.inboundSchema = exports.InsertEvent$inboundSchema;
    /** @deprecated use `InsertEvent$outboundSchema` instead. */
    InsertEvent$.outboundSchema = exports.InsertEvent$outboundSchema;
})(InsertEvent$ || (exports.InsertEvent$ = InsertEvent$ = {}));
function insertEventToJSON(insertEvent) {
    return JSON.stringify(exports.InsertEvent$outboundSchema.parse(insertEvent));
}
function insertEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.InsertEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InsertEvent' from JSON`);
}
