// @inkeep/inkeep-analytics/dist/commonjs/models/components/openaitoolcallfunction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenAIToolCallFunction$ = exports.OpenAIToolCallFunction$outboundSchema = exports.OpenAIToolCallFunction$inboundSchema = void 0;
exports.openAIToolCallFunctionToJSON = openAIToolCallFunctionToJSON;
exports.openAIToolCallFunctionFromJSON = openAIToolCallFunctionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.OpenAIToolCallFunction$inboundSchema = z.object({
    name: z.string(),
    arguments: z.string(),
});
/** @internal */
exports.OpenAIToolCallFunction$outboundSchema = z.object({
    name: z.string(),
    arguments: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OpenAIToolCallFunction$;
(function (OpenAIToolCallFunction$) {
    /** @deprecated use `OpenAIToolCallFunction$inboundSchema` instead. */
    OpenAIToolCallFunction$.inboundSchema = exports.OpenAIToolCallFunction$inboundSchema;
    /** @deprecated use `OpenAIToolCallFunction$outboundSchema` instead. */
    OpenAIToolCallFunction$.outboundSchema = exports.OpenAIToolCallFunction$outboundSchema;
})(OpenAIToolCallFunction$ || (exports.OpenAIToolCallFunction$ = OpenAIToolCallFunction$ = {}));
function openAIToolCallFunctionToJSON(openAIToolCallFunction) {
    return JSON.stringify(exports.OpenAIToolCallFunction$outboundSchema.parse(openAIToolCallFunction));
}
function openAIToolCallFunctionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OpenAIToolCallFunction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIToolCallFunction' from JSON`);
}
