// @inkeep/inkeep-analytics/dist/commonjs/models/components/openaitoolcall.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenAIToolCall$ = exports.OpenAIToolCall$outboundSchema = exports.OpenAIToolCall$inboundSchema = exports.OpenAIToolCallType$ = exports.OpenAIToolCallType$outboundSchema = exports.OpenAIToolCallType$inboundSchema = exports.OpenAIToolCallType = void 0;
exports.openAIToolCallToJSON = openAIToolCallToJSON;
exports.openAIToolCallFromJSON = openAIToolCallFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const openaitoolcallfunction_js_1 = require("./openaitoolcallfunction.js");
exports.OpenAIToolCallType = {
    Function: "function",
};
/** @internal */
exports.OpenAIToolCallType$inboundSchema = z.nativeEnum(exports.OpenAIToolCallType);
/** @internal */
exports.OpenAIToolCallType$outboundSchema = exports.OpenAIToolCallType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OpenAIToolCallType$;
(function (OpenAIToolCallType$) {
    /** @deprecated use `OpenAIToolCallType$inboundSchema` instead. */
    OpenAIToolCallType$.inboundSchema = exports.OpenAIToolCallType$inboundSchema;
    /** @deprecated use `OpenAIToolCallType$outboundSchema` instead. */
    OpenAIToolCallType$.outboundSchema = exports.OpenAIToolCallType$outboundSchema;
})(OpenAIToolCallType$ || (exports.OpenAIToolCallType$ = OpenAIToolCallType$ = {}));
/** @internal */
exports.OpenAIToolCall$inboundSchema = z.object({
    id: z.string(),
    type: exports.OpenAIToolCallType$inboundSchema,
    function: openaitoolcallfunction_js_1.OpenAIToolCallFunction$inboundSchema,
});
/** @internal */
exports.OpenAIToolCall$outboundSchema = z.object({
    id: z.string(),
    type: exports.OpenAIToolCallType$outboundSchema,
    function: openaitoolcallfunction_js_1.OpenAIToolCallFunction$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OpenAIToolCall$;
(function (OpenAIToolCall$) {
    /** @deprecated use `OpenAIToolCall$inboundSchema` instead. */
    OpenAIToolCall$.inboundSchema = exports.OpenAIToolCall$inboundSchema;
    /** @deprecated use `OpenAIToolCall$outboundSchema` instead. */
    OpenAIToolCall$.outboundSchema = exports.OpenAIToolCall$outboundSchema;
})(OpenAIToolCall$ || (exports.OpenAIToolCall$ = OpenAIToolCall$ = {}));
function openAIToolCallToJSON(openAIToolCall) {
    return JSON.stringify(exports.OpenAIToolCall$outboundSchema.parse(openAIToolCall));
}
function openAIToolCallFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OpenAIToolCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIToolCall' from JSON`);
}
