// @inkeep/inkeep-analytics/dist/commonjs/models/components/openaiimageurlobject.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenAIImageUrlObject$ = exports.OpenAIImageUrlObject$outboundSchema = exports.OpenAIImageUrlObject$inboundSchema = exports.Detail$ = exports.Detail$outboundSchema = exports.Detail$inboundSchema = exports.Detail = void 0;
exports.openAIImageUrlObjectToJSON = openAIImageUrlObjectToJSON;
exports.openAIImageUrlObjectFromJSON = openAIImageUrlObjectFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.Detail = {
    Auto: "auto",
    Low: "low",
    High: "high",
};
/** @internal */
exports.Detail$inboundSchema = z
    .nativeEnum(exports.Detail);
/** @internal */
exports.Detail$outboundSchema = exports.Detail$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Detail$;
(function (Detail$) {
    /** @deprecated use `Detail$inboundSchema` instead. */
    Detail$.inboundSchema = exports.Detail$inboundSchema;
    /** @deprecated use `Detail$outboundSchema` instead. */
    Detail$.outboundSchema = exports.Detail$outboundSchema;
})(Detail$ || (exports.Detail$ = Detail$ = {}));
/** @internal */
exports.OpenAIImageUrlObject$inboundSchema = z.object({
    url: z.string(),
    detail: exports.Detail$inboundSchema.default("auto"),
});
/** @internal */
exports.OpenAIImageUrlObject$outboundSchema = z.object({
    url: z.string(),
    detail: exports.Detail$outboundSchema.default("auto"),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OpenAIImageUrlObject$;
(function (OpenAIImageUrlObject$) {
    /** @deprecated use `OpenAIImageUrlObject$inboundSchema` instead. */
    OpenAIImageUrlObject$.inboundSchema = exports.OpenAIImageUrlObject$inboundSchema;
    /** @deprecated use `OpenAIImageUrlObject$outboundSchema` instead. */
    OpenAIImageUrlObject$.outboundSchema = exports.OpenAIImageUrlObject$outboundSchema;
})(OpenAIImageUrlObject$ || (exports.OpenAIImageUrlObject$ = OpenAIImageUrlObject$ = {}));
function openAIImageUrlObjectToJSON(openAIImageUrlObject) {
    return JSON.stringify(exports.OpenAIImageUrlObject$outboundSchema.parse(openAIImageUrlObject));
}
function openAIImageUrlObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OpenAIImageUrlObject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIImageUrlObject' from JSON`);
}
