// @inkeep/inkeep-analytics/dist/commonjs/models/components/propertyvaluesrequestbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyValuesRequestBody$ = exports.PropertyValuesRequestBody$outboundSchema = exports.PropertyValuesRequestBody$inboundSchema = exports.PropertyValuesRequestBodyViews$ = exports.PropertyValuesRequestBodyViews$outboundSchema = exports.PropertyValuesRequestBodyViews$inboundSchema = exports.PropertyValuesRequestBodyField$ = exports.PropertyValuesRequestBodyField$outboundSchema = exports.PropertyValuesRequestBodyField$inboundSchema = exports.PropertyValuesRequestBodyViews = exports.PropertyValuesRequestBodyField = void 0;
exports.propertyValuesRequestBodyToJSON = propertyValuesRequestBodyToJSON;
exports.propertyValuesRequestBodyFromJSON = propertyValuesRequestBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Which JSON field to query values from
 */
exports.PropertyValuesRequestBodyField = {
    Properties: "properties",
    UserProperties: "userProperties",
};
exports.PropertyValuesRequestBodyViews = {
    EventsView: "events_view",
    ConversationsView: "conversations_view",
    SemanticThreadsView: "semantic_threads_view",
    FeedbackView: "feedback_view",
};
/** @internal */
exports.PropertyValuesRequestBodyField$inboundSchema = z.nativeEnum(exports.PropertyValuesRequestBodyField);
/** @internal */
exports.PropertyValuesRequestBodyField$outboundSchema = exports.PropertyValuesRequestBodyField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PropertyValuesRequestBodyField$;
(function (PropertyValuesRequestBodyField$) {
    /** @deprecated use `PropertyValuesRequestBodyField$inboundSchema` instead. */
    PropertyValuesRequestBodyField$.inboundSchema = exports.PropertyValuesRequestBodyField$inboundSchema;
    /** @deprecated use `PropertyValuesRequestBodyField$outboundSchema` instead. */
    PropertyValuesRequestBodyField$.outboundSchema = exports.PropertyValuesRequestBodyField$outboundSchema;
})(PropertyValuesRequestBodyField$ || (exports.PropertyValuesRequestBodyField$ = PropertyValuesRequestBodyField$ = {}));
/** @internal */
exports.PropertyValuesRequestBodyViews$inboundSchema = z.nativeEnum(exports.PropertyValuesRequestBodyViews);
/** @internal */
exports.PropertyValuesRequestBodyViews$outboundSchema = exports.PropertyValuesRequestBodyViews$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PropertyValuesRequestBodyViews$;
(function (PropertyValuesRequestBodyViews$) {
    /** @deprecated use `PropertyValuesRequestBodyViews$inboundSchema` instead. */
    PropertyValuesRequestBodyViews$.inboundSchema = exports.PropertyValuesRequestBodyViews$inboundSchema;
    /** @deprecated use `PropertyValuesRequestBodyViews$outboundSchema` instead. */
    PropertyValuesRequestBodyViews$.outboundSchema = exports.PropertyValuesRequestBodyViews$outboundSchema;
})(PropertyValuesRequestBodyViews$ || (exports.PropertyValuesRequestBodyViews$ = PropertyValuesRequestBodyViews$ = {}));
/** @internal */
exports.PropertyValuesRequestBody$inboundSchema = z.object({
    field: exports.PropertyValuesRequestBodyField$inboundSchema,
    key: z.string(),
    views: z.array(exports.PropertyValuesRequestBodyViews$inboundSchema).optional(),
    limit: z.number().default(100),
});
/** @internal */
exports.PropertyValuesRequestBody$outboundSchema = z.object({
    field: exports.PropertyValuesRequestBodyField$outboundSchema,
    key: z.string(),
    views: z.array(exports.PropertyValuesRequestBodyViews$outboundSchema).optional(),
    limit: z.number().default(100),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PropertyValuesRequestBody$;
(function (PropertyValuesRequestBody$) {
    /** @deprecated use `PropertyValuesRequestBody$inboundSchema` instead. */
    PropertyValuesRequestBody$.inboundSchema = exports.PropertyValuesRequestBody$inboundSchema;
    /** @deprecated use `PropertyValuesRequestBody$outboundSchema` instead. */
    PropertyValuesRequestBody$.outboundSchema = exports.PropertyValuesRequestBody$outboundSchema;
})(PropertyValuesRequestBody$ || (exports.PropertyValuesRequestBody$ = PropertyValuesRequestBody$ = {}));
function propertyValuesRequestBodyToJSON(propertyValuesRequestBody) {
    return JSON.stringify(exports.PropertyValuesRequestBody$outboundSchema.parse(propertyValuesRequestBody));
}
function propertyValuesRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PropertyValuesRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PropertyValuesRequestBody' from JSON`);
}
