// @inkeep/inkeep-analytics/dist/commonjs/models/components/querysemanticthreadsrequestbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.QuerySemanticThreadsRequestBody$ = exports.QuerySemanticThreadsRequestBody$outboundSchema = exports.QuerySemanticThreadsRequestBody$inboundSchema = void 0;
exports.querySemanticThreadsRequestBodyToJSON = querySemanticThreadsRequestBodyToJSON;
exports.querySemanticThreadsRequestBodyFromJSON = querySemanticThreadsRequestBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const semanticthreadfilter_js_1 = require("./semanticthreadfilter.js");
const semanticthreadgroupby_js_1 = require("./semanticthreadgroupby.js");
const semanticthreadorderby_js_1 = require("./semanticthreadorderby.js");
const semanticthreadselection_js_1 = require("./semanticthreadselection.js");
/** @internal */
exports.QuerySemanticThreadsRequestBody$inboundSchema = z.object({
    query: z.string().optional(),
    select: z.array(semanticthreadselection_js_1.SemanticThreadSelection$inboundSchema).optional(),
    groupBy: z.array(semanticthreadgroupby_js_1.SemanticThreadGroupBy$inboundSchema).optional(),
    where: semanticthreadfilter_js_1.SemanticThreadFilter$inboundSchema.optional(),
    orderBy: z.array(semanticthreadorderby_js_1.SemanticThreadOrderBy$inboundSchema).optional(),
    notes: z.string().optional(),
});
/** @internal */
exports.QuerySemanticThreadsRequestBody$outboundSchema = z.object({
    query: z.string().optional(),
    select: z.array(semanticthreadselection_js_1.SemanticThreadSelection$outboundSchema).optional(),
    groupBy: z.array(semanticthreadgroupby_js_1.SemanticThreadGroupBy$outboundSchema).optional(),
    where: semanticthreadfilter_js_1.SemanticThreadFilter$outboundSchema.optional(),
    orderBy: z.array(semanticthreadorderby_js_1.SemanticThreadOrderBy$outboundSchema).optional(),
    notes: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QuerySemanticThreadsRequestBody$;
(function (QuerySemanticThreadsRequestBody$) {
    /** @deprecated use `QuerySemanticThreadsRequestBody$inboundSchema` instead. */
    QuerySemanticThreadsRequestBody$.inboundSchema = exports.QuerySemanticThreadsRequestBody$inboundSchema;
    /** @deprecated use `QuerySemanticThreadsRequestBody$outboundSchema` instead. */
    QuerySemanticThreadsRequestBody$.outboundSchema = exports.QuerySemanticThreadsRequestBody$outboundSchema;
})(QuerySemanticThreadsRequestBody$ || (exports.QuerySemanticThreadsRequestBody$ = QuerySemanticThreadsRequestBody$ = {}));
function querySemanticThreadsRequestBodyToJSON(querySemanticThreadsRequestBody) {
    return JSON.stringify(exports.QuerySemanticThreadsRequestBody$outboundSchema.parse(querySemanticThreadsRequestBody));
}
function querySemanticThreadsRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.QuerySemanticThreadsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QuerySemanticThreadsRequestBody' from JSON`);
}
