// @inkeep/inkeep-analytics/dist/commonjs/models/components/queryeventsresponsebody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryEventsResponseBody$ = exports.QueryEventsResponseBody$outboundSchema = exports.QueryEventsResponseBody$inboundSchema = exports.QueryEventsResponseBodyStatus$ = exports.QueryEventsResponseBodyStatus$outboundSchema = exports.QueryEventsResponseBodyStatus$inboundSchema = exports.QueryEventsResponseBodyStatus = void 0;
exports.queryEventsResponseBodyToJSON = queryEventsResponseBodyToJSON;
exports.queryEventsResponseBodyFromJSON = queryEventsResponseBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const queryeventsresponsedata_js_1 = require("./queryeventsresponsedata.js");
/**
 * Status of the response
 */
exports.QueryEventsResponseBodyStatus = {
    Ok: "ok",
};
/** @internal */
exports.QueryEventsResponseBodyStatus$inboundSchema = z.nativeEnum(exports.QueryEventsResponseBodyStatus);
/** @internal */
exports.QueryEventsResponseBodyStatus$outboundSchema = exports.QueryEventsResponseBodyStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryEventsResponseBodyStatus$;
(function (QueryEventsResponseBodyStatus$) {
    /** @deprecated use `QueryEventsResponseBodyStatus$inboundSchema` instead. */
    QueryEventsResponseBodyStatus$.inboundSchema = exports.QueryEventsResponseBodyStatus$inboundSchema;
    /** @deprecated use `QueryEventsResponseBodyStatus$outboundSchema` instead. */
    QueryEventsResponseBodyStatus$.outboundSchema = exports.QueryEventsResponseBodyStatus$outboundSchema;
})(QueryEventsResponseBodyStatus$ || (exports.QueryEventsResponseBodyStatus$ = QueryEventsResponseBodyStatus$ = {}));
/** @internal */
exports.QueryEventsResponseBody$inboundSchema = z.object({
    status: exports.QueryEventsResponseBodyStatus$inboundSchema,
    data: queryeventsresponsedata_js_1.QueryEventsResponseData$inboundSchema,
});
/** @internal */
exports.QueryEventsResponseBody$outboundSchema = z.object({
    status: exports.QueryEventsResponseBodyStatus$outboundSchema,
    data: queryeventsresponsedata_js_1.QueryEventsResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryEventsResponseBody$;
(function (QueryEventsResponseBody$) {
    /** @deprecated use `QueryEventsResponseBody$inboundSchema` instead. */
    QueryEventsResponseBody$.inboundSchema = exports.QueryEventsResponseBody$inboundSchema;
    /** @deprecated use `QueryEventsResponseBody$outboundSchema` instead. */
    QueryEventsResponseBody$.outboundSchema = exports.QueryEventsResponseBody$outboundSchema;
})(QueryEventsResponseBody$ || (exports.QueryEventsResponseBody$ = QueryEventsResponseBody$ = {}));
function queryEventsResponseBodyToJSON(queryEventsResponseBody) {
    return JSON.stringify(exports.QueryEventsResponseBody$outboundSchema.parse(queryEventsResponseBody));
}
function queryEventsResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.QueryEventsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryEventsResponseBody' from JSON`);
}
