// @inkeep/inkeep-analytics/dist/commonjs/models/components/queryfeedbackrequestbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryFeedbackRequestBody$ = exports.QueryFeedbackRequestBody$outboundSchema = exports.QueryFeedbackRequestBody$inboundSchema = void 0;
exports.queryFeedbackRequestBodyToJSON = queryFeedbackRequestBodyToJSON;
exports.queryFeedbackRequestBodyFromJSON = queryFeedbackRequestBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const feedbackfilter_js_1 = require("./feedbackfilter.js");
const feedbackgroupby_js_1 = require("./feedbackgroupby.js");
const feedbackorderby_js_1 = require("./feedbackorderby.js");
const feedbackselection_js_1 = require("./feedbackselection.js");
/** @internal */
exports.QueryFeedbackRequestBody$inboundSchema = z.object({
    select: z.array(feedbackselection_js_1.FeedbackSelection$inboundSchema).optional(),
    groupBy: z.array(feedbackgroupby_js_1.FeedbackGroupBy$inboundSchema).optional(),
    where: feedbackfilter_js_1.FeedbackFilter$inboundSchema.optional(),
    orderBy: z.array(feedbackorderby_js_1.FeedbackOrderBy$inboundSchema).optional(),
    notes: z.string().optional(),
});
/** @internal */
exports.QueryFeedbackRequestBody$outboundSchema = z.object({
    select: z.array(feedbackselection_js_1.FeedbackSelection$outboundSchema).optional(),
    groupBy: z.array(feedbackgroupby_js_1.FeedbackGroupBy$outboundSchema).optional(),
    where: feedbackfilter_js_1.FeedbackFilter$outboundSchema.optional(),
    orderBy: z.array(feedbackorderby_js_1.FeedbackOrderBy$outboundSchema).optional(),
    notes: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryFeedbackRequestBody$;
(function (QueryFeedbackRequestBody$) {
    /** @deprecated use `QueryFeedbackRequestBody$inboundSchema` instead. */
    QueryFeedbackRequestBody$.inboundSchema = exports.QueryFeedbackRequestBody$inboundSchema;
    /** @deprecated use `QueryFeedbackRequestBody$outboundSchema` instead. */
    QueryFeedbackRequestBody$.outboundSchema = exports.QueryFeedbackRequestBody$outboundSchema;
})(QueryFeedbackRequestBody$ || (exports.QueryFeedbackRequestBody$ = QueryFeedbackRequestBody$ = {}));
function queryFeedbackRequestBodyToJSON(queryFeedbackRequestBody) {
    return JSON.stringify(exports.QueryFeedbackRequestBody$outboundSchema.parse(queryFeedbackRequestBody));
}
function queryFeedbackRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.QueryFeedbackRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryFeedbackRequestBody' from JSON`);
}
