// @inkeep/inkeep-analytics/dist/commonjs/models/components/queryfeedbackresponsebody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryFeedbackResponseBody$ = exports.QueryFeedbackResponseBody$outboundSchema = exports.QueryFeedbackResponseBody$inboundSchema = exports.QueryFeedbackResponseBodyStatus$ = exports.QueryFeedbackResponseBodyStatus$outboundSchema = exports.QueryFeedbackResponseBodyStatus$inboundSchema = exports.QueryFeedbackResponseBodyStatus = void 0;
exports.queryFeedbackResponseBodyToJSON = queryFeedbackResponseBodyToJSON;
exports.queryFeedbackResponseBodyFromJSON = queryFeedbackResponseBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const queryfeedbackresponsedata_js_1 = require("./queryfeedbackresponsedata.js");
/**
 * Status of the response
 */
exports.QueryFeedbackResponseBodyStatus = {
    Ok: "ok",
};
/** @internal */
exports.QueryFeedbackResponseBodyStatus$inboundSchema = z.nativeEnum(exports.QueryFeedbackResponseBodyStatus);
/** @internal */
exports.QueryFeedbackResponseBodyStatus$outboundSchema = exports.QueryFeedbackResponseBodyStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryFeedbackResponseBodyStatus$;
(function (QueryFeedbackResponseBodyStatus$) {
    /** @deprecated use `QueryFeedbackResponseBodyStatus$inboundSchema` instead. */
    QueryFeedbackResponseBodyStatus$.inboundSchema = exports.QueryFeedbackResponseBodyStatus$inboundSchema;
    /** @deprecated use `QueryFeedbackResponseBodyStatus$outboundSchema` instead. */
    QueryFeedbackResponseBodyStatus$.outboundSchema = exports.QueryFeedbackResponseBodyStatus$outboundSchema;
})(QueryFeedbackResponseBodyStatus$ || (exports.QueryFeedbackResponseBodyStatus$ = QueryFeedbackResponseBodyStatus$ = {}));
/** @internal */
exports.QueryFeedbackResponseBody$inboundSchema = z.object({
    status: exports.QueryFeedbackResponseBodyStatus$inboundSchema,
    data: queryfeedbackresponsedata_js_1.QueryFeedbackResponseData$inboundSchema,
});
/** @internal */
exports.QueryFeedbackResponseBody$outboundSchema = z.object({
    status: exports.QueryFeedbackResponseBodyStatus$outboundSchema,
    data: queryfeedbackresponsedata_js_1.QueryFeedbackResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryFeedbackResponseBody$;
(function (QueryFeedbackResponseBody$) {
    /** @deprecated use `QueryFeedbackResponseBody$inboundSchema` instead. */
    QueryFeedbackResponseBody$.inboundSchema = exports.QueryFeedbackResponseBody$inboundSchema;
    /** @deprecated use `QueryFeedbackResponseBody$outboundSchema` instead. */
    QueryFeedbackResponseBody$.outboundSchema = exports.QueryFeedbackResponseBody$outboundSchema;
})(QueryFeedbackResponseBody$ || (exports.QueryFeedbackResponseBody$ = QueryFeedbackResponseBody$ = {}));
function queryFeedbackResponseBodyToJSON(queryFeedbackResponseBody) {
    return JSON.stringify(exports.QueryFeedbackResponseBody$outboundSchema.parse(queryFeedbackResponseBody));
}
function queryFeedbackResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.QueryFeedbackResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryFeedbackResponseBody' from JSON`);
}
