// @inkeep/inkeep-analytics/dist/commonjs/models/components/queryconversationsrequestbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryConversationsRequestBody$ = exports.QueryConversationsRequestBody$outboundSchema = exports.QueryConversationsRequestBody$inboundSchema = void 0;
exports.queryConversationsRequestBodyToJSON = queryConversationsRequestBodyToJSON;
exports.queryConversationsRequestBodyFromJSON = queryConversationsRequestBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const conversationsfilter_js_1 = require("./conversationsfilter.js");
const conversationsgroupby_js_1 = require("./conversationsgroupby.js");
const conversationsorderby_js_1 = require("./conversationsorderby.js");
const conversationsselection_js_1 = require("./conversationsselection.js");
/** @internal */
exports.QueryConversationsRequestBody$inboundSchema = z.object({
    select: z.array(conversationsselection_js_1.ConversationsSelection$inboundSchema).optional(),
    groupBy: z.array(conversationsgroupby_js_1.ConversationsGroupBy$inboundSchema).optional(),
    where: conversationsfilter_js_1.ConversationsFilter$inboundSchema.optional(),
    orderBy: z.array(conversationsorderby_js_1.ConversationsOrderBy$inboundSchema).optional(),
    notes: z.string().optional(),
});
/** @internal */
exports.QueryConversationsRequestBody$outboundSchema = z.object({
    select: z.array(conversationsselection_js_1.ConversationsSelection$outboundSchema).optional(),
    groupBy: z.array(conversationsgroupby_js_1.ConversationsGroupBy$outboundSchema).optional(),
    where: conversationsfilter_js_1.ConversationsFilter$outboundSchema.optional(),
    orderBy: z.array(conversationsorderby_js_1.ConversationsOrderBy$outboundSchema).optional(),
    notes: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryConversationsRequestBody$;
(function (QueryConversationsRequestBody$) {
    /** @deprecated use `QueryConversationsRequestBody$inboundSchema` instead. */
    QueryConversationsRequestBody$.inboundSchema = exports.QueryConversationsRequestBody$inboundSchema;
    /** @deprecated use `QueryConversationsRequestBody$outboundSchema` instead. */
    QueryConversationsRequestBody$.outboundSchema = exports.QueryConversationsRequestBody$outboundSchema;
})(QueryConversationsRequestBody$ || (exports.QueryConversationsRequestBody$ = QueryConversationsRequestBody$ = {}));
function queryConversationsRequestBodyToJSON(queryConversationsRequestBody) {
    return JSON.stringify(exports.QueryConversationsRequestBody$outboundSchema.parse(queryConversationsRequestBody));
}
function queryConversationsRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.QueryConversationsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryConversationsRequestBody' from JSON`);
}
