// @inkeep/inkeep-analytics/dist/commonjs/models/components/selectevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SelectEvent$ = exports.SelectEvent$outboundSchema = exports.SelectEvent$inboundSchema = void 0;
exports.selectEventToJSON = selectEventToJSON;
exports.selectEventFromJSON = selectEventFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const selectconversationevent_js_1 = require("./selectconversationevent.js");
const selectmessageevent_js_1 = require("./selectmessageevent.js");
const selectsearchevent_js_1 = require("./selectsearchevent.js");
/** @internal */
exports.SelectEvent$inboundSchema = z.union([
    selectmessageevent_js_1.SelectMessageEvent$inboundSchema.and(z.object({ entityType: z.literal("message") }).transform((v) => ({
        entityType: v.entityType,
    }))),
    selectconversationevent_js_1.SelectConversationEvent$inboundSchema.and(z.object({ entityType: z.literal("conversation") }).transform((v) => ({
        entityType: v.entityType,
    }))),
    selectsearchevent_js_1.SelectSearchEvent$inboundSchema.and(z.object({ entityType: z.literal("search") }).transform((v) => ({
        entityType: v.entityType,
    }))),
]);
/** @internal */
exports.SelectEvent$outboundSchema = z.union([
    selectmessageevent_js_1.SelectMessageEvent$outboundSchema.and(z.object({ entityType: z.literal("message") }).transform((v) => ({
        entityType: v.entityType,
    }))),
    selectconversationevent_js_1.SelectConversationEvent$outboundSchema.and(z.object({ entityType: z.literal("conversation") }).transform((v) => ({
        entityType: v.entityType,
    }))),
    selectsearchevent_js_1.SelectSearchEvent$outboundSchema.and(z.object({ entityType: z.literal("search") }).transform((v) => ({
        entityType: v.entityType,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SelectEvent$;
(function (SelectEvent$) {
    /** @deprecated use `SelectEvent$inboundSchema` instead. */
    SelectEvent$.inboundSchema = exports.SelectEvent$inboundSchema;
    /** @deprecated use `SelectEvent$outboundSchema` instead. */
    SelectEvent$.outboundSchema = exports.SelectEvent$outboundSchema;
})(SelectEvent$ || (exports.SelectEvent$ = SelectEvent$ = {}));
function selectEventToJSON(selectEvent) {
    return JSON.stringify(exports.SelectEvent$outboundSchema.parse(selectEvent));
}
function selectEventFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SelectEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SelectEvent' from JSON`);
}
