// @inkeep/inkeep-analytics/dist/commonjs/models/components/semanticthreadfilterand.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SemanticThreadFilterAND$ = exports.SemanticThreadFilterAND$outboundSchema = exports.SemanticThreadFilterAND$inboundSchema = exports.SemanticThreadFilterANDAND$ = exports.SemanticThreadFilterANDAND$outboundSchema = exports.SemanticThreadFilterANDAND$inboundSchema = void 0;
exports.semanticThreadFilterANDANDToJSON = semanticThreadFilterANDANDToJSON;
exports.semanticThreadFilterANDANDFromJSON = semanticThreadFilterANDANDFromJSON;
exports.semanticThreadFilterANDToJSON = semanticThreadFilterANDToJSON;
exports.semanticThreadFilterANDFromJSON = semanticThreadFilterANDFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const semanticthreadfilter_js_1 = require("./semanticthreadfilter.js");
/** @internal */
exports.SemanticThreadFilterANDAND$inboundSchema = z.lazy(() => semanticthreadfilter_js_1.SemanticThreadFilter$inboundSchema);
/** @internal */
exports.SemanticThreadFilterANDAND$outboundSchema = z.lazy(() => semanticthreadfilter_js_1.SemanticThreadFilter$outboundSchema);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadFilterANDAND$;
(function (SemanticThreadFilterANDAND$) {
    /** @deprecated use `SemanticThreadFilterANDAND$inboundSchema` instead. */
    SemanticThreadFilterANDAND$.inboundSchema = exports.SemanticThreadFilterANDAND$inboundSchema;
    /** @deprecated use `SemanticThreadFilterANDAND$outboundSchema` instead. */
    SemanticThreadFilterANDAND$.outboundSchema = exports.SemanticThreadFilterANDAND$outboundSchema;
})(SemanticThreadFilterANDAND$ || (exports.SemanticThreadFilterANDAND$ = SemanticThreadFilterANDAND$ = {}));
function semanticThreadFilterANDANDToJSON(semanticThreadFilterANDAND) {
    return JSON.stringify(exports.SemanticThreadFilterANDAND$outboundSchema.parse(semanticThreadFilterANDAND));
}
function semanticThreadFilterANDANDFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SemanticThreadFilterANDAND$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SemanticThreadFilterANDAND' from JSON`);
}
/** @internal */
exports.SemanticThreadFilterAND$inboundSchema = z.object({
    and: z.array(z.lazy(() => semanticthreadfilter_js_1.SemanticThreadFilter$inboundSchema)),
});
/** @internal */
exports.SemanticThreadFilterAND$outboundSchema = z.object({
    and: z.array(z.lazy(() => semanticthreadfilter_js_1.SemanticThreadFilter$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadFilterAND$;
(function (SemanticThreadFilterAND$) {
    /** @deprecated use `SemanticThreadFilterAND$inboundSchema` instead. */
    SemanticThreadFilterAND$.inboundSchema = exports.SemanticThreadFilterAND$inboundSchema;
    /** @deprecated use `SemanticThreadFilterAND$outboundSchema` instead. */
    SemanticThreadFilterAND$.outboundSchema = exports.SemanticThreadFilterAND$outboundSchema;
})(SemanticThreadFilterAND$ || (exports.SemanticThreadFilterAND$ = SemanticThreadFilterAND$ = {}));
function semanticThreadFilterANDToJSON(semanticThreadFilterAND) {
    return JSON.stringify(exports.SemanticThreadFilterAND$outboundSchema.parse(semanticThreadFilterAND));
}
function semanticThreadFilterANDFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SemanticThreadFilterAND$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SemanticThreadFilterAND' from JSON`);
}
