// @inkeep/inkeep-analytics/dist/commonjs/models/components/semanticthreadselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SemanticThreadSelection$ = exports.SemanticThreadSelection$outboundSchema = exports.SemanticThreadSelection$inboundSchema = void 0;
exports.semanticThreadSelectionToJSON = semanticThreadSelectionToJSON;
exports.semanticThreadSelectionFromJSON = semanticThreadSelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const semanticthreadaggregationselection_js_1 = require("./semanticthreadaggregationselection.js");
const semanticthreadsimplefieldselection_js_1 = require("./semanticthreadsimplefieldselection.js");
const semanticthreadtimebasedgroupbyselection_js_1 = require("./semanticthreadtimebasedgroupbyselection.js");
/** @internal */
exports.SemanticThreadSelection$inboundSchema = z.union([
    semanticthreadtimebasedgroupbyselection_js_1.SemanticThreadTimeBasedGroupBySelection$inboundSchema.and(z.object({ type: z.literal("time") }).transform((v) => ({ type: v.type }))),
    semanticthreadsimplefieldselection_js_1.SemanticThreadSimpleFieldSelection$inboundSchema.and(z.object({ type: z.literal("field") }).transform((v) => ({ type: v.type }))),
    semanticthreadaggregationselection_js_1.SemanticThreadAggregationSelection$inboundSchema.and(z.object({ type: z.literal("aggregation") }).transform((v) => ({
        type: v.type,
    }))),
]);
/** @internal */
exports.SemanticThreadSelection$outboundSchema = z.union([
    semanticthreadtimebasedgroupbyselection_js_1.SemanticThreadTimeBasedGroupBySelection$outboundSchema.and(z.object({ type: z.literal("time") }).transform((v) => ({ type: v.type }))),
    semanticthreadsimplefieldselection_js_1.SemanticThreadSimpleFieldSelection$outboundSchema.and(z.object({ type: z.literal("field") }).transform((v) => ({ type: v.type }))),
    semanticthreadaggregationselection_js_1.SemanticThreadAggregationSelection$outboundSchema.and(z.object({ type: z.literal("aggregation") }).transform((v) => ({
        type: v.type,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadSelection$;
(function (SemanticThreadSelection$) {
    /** @deprecated use `SemanticThreadSelection$inboundSchema` instead. */
    SemanticThreadSelection$.inboundSchema = exports.SemanticThreadSelection$inboundSchema;
    /** @deprecated use `SemanticThreadSelection$outboundSchema` instead. */
    SemanticThreadSelection$.outboundSchema = exports.SemanticThreadSelection$outboundSchema;
})(SemanticThreadSelection$ || (exports.SemanticThreadSelection$ = SemanticThreadSelection$ = {}));
function semanticThreadSelectionToJSON(semanticThreadSelection) {
    return JSON.stringify(exports.SemanticThreadSelection$outboundSchema.parse(semanticThreadSelection));
}
function semanticThreadSelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SemanticThreadSelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SemanticThreadSelection' from JSON`);
}
