// @inkeep/inkeep-analytics/dist/commonjs/models/components/semanticthreadfilteror.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SemanticThreadFilterOR$ = exports.SemanticThreadFilterOR$outboundSchema = exports.SemanticThreadFilterOR$inboundSchema = exports.SemanticThreadFilterOROR$ = exports.SemanticThreadFilterOROR$outboundSchema = exports.SemanticThreadFilterOROR$inboundSchema = void 0;
exports.semanticThreadFilterORORToJSON = semanticThreadFilterORORToJSON;
exports.semanticThreadFilterORORFromJSON = semanticThreadFilterORORFromJSON;
exports.semanticThreadFilterORToJSON = semanticThreadFilterORToJSON;
exports.semanticThreadFilterORFromJSON = semanticThreadFilterORFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const semanticthreadfilter_js_1 = require("./semanticthreadfilter.js");
/** @internal */
exports.SemanticThreadFilterOROR$inboundSchema = z.lazy(() => semanticthreadfilter_js_1.SemanticThreadFilter$inboundSchema);
/** @internal */
exports.SemanticThreadFilterOROR$outboundSchema = z.lazy(() => semanticthreadfilter_js_1.SemanticThreadFilter$outboundSchema);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadFilterOROR$;
(function (SemanticThreadFilterOROR$) {
    /** @deprecated use `SemanticThreadFilterOROR$inboundSchema` instead. */
    SemanticThreadFilterOROR$.inboundSchema = exports.SemanticThreadFilterOROR$inboundSchema;
    /** @deprecated use `SemanticThreadFilterOROR$outboundSchema` instead. */
    SemanticThreadFilterOROR$.outboundSchema = exports.SemanticThreadFilterOROR$outboundSchema;
})(SemanticThreadFilterOROR$ || (exports.SemanticThreadFilterOROR$ = SemanticThreadFilterOROR$ = {}));
function semanticThreadFilterORORToJSON(semanticThreadFilterOROR) {
    return JSON.stringify(exports.SemanticThreadFilterOROR$outboundSchema.parse(semanticThreadFilterOROR));
}
function semanticThreadFilterORORFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SemanticThreadFilterOROR$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SemanticThreadFilterOROR' from JSON`);
}
/** @internal */
exports.SemanticThreadFilterOR$inboundSchema = z.object({
    or: z.array(z.lazy(() => semanticthreadfilter_js_1.SemanticThreadFilter$inboundSchema)),
});
/** @internal */
exports.SemanticThreadFilterOR$outboundSchema = z.object({
    or: z.array(z.lazy(() => semanticthreadfilter_js_1.SemanticThreadFilter$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadFilterOR$;
(function (SemanticThreadFilterOR$) {
    /** @deprecated use `SemanticThreadFilterOR$inboundSchema` instead. */
    SemanticThreadFilterOR$.inboundSchema = exports.SemanticThreadFilterOR$inboundSchema;
    /** @deprecated use `SemanticThreadFilterOR$outboundSchema` instead. */
    SemanticThreadFilterOR$.outboundSchema = exports.SemanticThreadFilterOR$outboundSchema;
})(SemanticThreadFilterOR$ || (exports.SemanticThreadFilterOR$ = SemanticThreadFilterOR$ = {}));
function semanticThreadFilterORToJSON(semanticThreadFilterOR) {
    return JSON.stringify(exports.SemanticThreadFilterOR$outboundSchema.parse(semanticThreadFilterOR));
}
function semanticThreadFilterORFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SemanticThreadFilterOR$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SemanticThreadFilterOR' from JSON`);
}
