// @inkeep/inkeep-analytics/dist/commonjs/models/components/semanticthreadtimebasedgroupbyselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SemanticThreadTimeBasedGroupBySelection$ = exports.SemanticThreadTimeBasedGroupBySelection$outboundSchema = exports.SemanticThreadTimeBasedGroupBySelection$inboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionField$ = exports.SemanticThreadTimeBasedGroupBySelectionField$outboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionField$inboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionType$ = exports.SemanticThreadTimeBasedGroupBySelectionType$outboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionType$inboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionField = exports.SemanticThreadTimeBasedGroupBySelectionType = void 0;
exports.semanticThreadTimeBasedGroupBySelectionToJSON = semanticThreadTimeBasedGroupBySelectionToJSON;
exports.semanticThreadTimeBasedGroupBySelectionFromJSON = semanticThreadTimeBasedGroupBySelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const timeunit_js_1 = require("./timeunit.js");
exports.SemanticThreadTimeBasedGroupBySelectionType = {
    Time: "time",
};
exports.SemanticThreadTimeBasedGroupBySelectionField = {
    Id: "id",
    MessageIds: "messageIds",
    MessagesCount: "messagesCount",
    OrganizationId: "organizationId",
    Question: "question",
    QuestionLength: "questionLength",
    Answer: "answer",
    IsOnTopic: "isOnTopic",
    IsClear: "isClear",
    IsDocumented: "isDocumented",
    IsFeatureSupported: "isFeatureSupported",
    IsAnswerConfident: "isAnswerConfident",
    GapSummary: "gapSummary",
    Subject: "subject",
    Category: "category",
    Languages: "languages",
    FirstPartyEntities: "firstPartyEntities",
    ThirdPartyEntities: "thirdPartyEntities",
    Emotion: "emotion",
    Sentiment: "sentiment",
    QuestionType: "questionType",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    Properties: "properties",
    UserProperties: "userProperties",
    FirstMessageTime: "firstMessageTime",
    LastMessageTime: "lastMessageTime",
    LastUpdated: "lastUpdated",
    HasPositiveFeedback: "hasPositiveFeedback",
    HasNegativeFeedback: "hasNegativeFeedback",
    ConversationIds: "conversationIds",
    AreAIAnnotationsVerified: "areAIAnnotationsVerified",
    IsInitialQuestionAnswer: "isInitialQuestionAnswer",
};
/** @internal */
exports.SemanticThreadTimeBasedGroupBySelectionType$inboundSchema = z
    .nativeEnum(exports.SemanticThreadTimeBasedGroupBySelectionType);
/** @internal */
exports.SemanticThreadTimeBasedGroupBySelectionType$outboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadTimeBasedGroupBySelectionType$;
(function (SemanticThreadTimeBasedGroupBySelectionType$) {
    /** @deprecated use `SemanticThreadTimeBasedGroupBySelectionType$inboundSchema` instead. */
    SemanticThreadTimeBasedGroupBySelectionType$.inboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionType$inboundSchema;
    /** @deprecated use `SemanticThreadTimeBasedGroupBySelectionType$outboundSchema` instead. */
    SemanticThreadTimeBasedGroupBySelectionType$.outboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionType$outboundSchema;
})(SemanticThreadTimeBasedGroupBySelectionType$ || (exports.SemanticThreadTimeBasedGroupBySelectionType$ = SemanticThreadTimeBasedGroupBySelectionType$ = {}));
/** @internal */
exports.SemanticThreadTimeBasedGroupBySelectionField$inboundSchema = z
    .nativeEnum(exports.SemanticThreadTimeBasedGroupBySelectionField);
/** @internal */
exports.SemanticThreadTimeBasedGroupBySelectionField$outboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadTimeBasedGroupBySelectionField$;
(function (SemanticThreadTimeBasedGroupBySelectionField$) {
    /** @deprecated use `SemanticThreadTimeBasedGroupBySelectionField$inboundSchema` instead. */
    SemanticThreadTimeBasedGroupBySelectionField$.inboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionField$inboundSchema;
    /** @deprecated use `SemanticThreadTimeBasedGroupBySelectionField$outboundSchema` instead. */
    SemanticThreadTimeBasedGroupBySelectionField$.outboundSchema = exports.SemanticThreadTimeBasedGroupBySelectionField$outboundSchema;
})(SemanticThreadTimeBasedGroupBySelectionField$ || (exports.SemanticThreadTimeBasedGroupBySelectionField$ = SemanticThreadTimeBasedGroupBySelectionField$ = {}));
/** @internal */
exports.SemanticThreadTimeBasedGroupBySelection$inboundSchema = z.object({
    type: exports.SemanticThreadTimeBasedGroupBySelectionType$inboundSchema,
    timeUnit: timeunit_js_1.TimeUnit$inboundSchema,
    field: exports.SemanticThreadTimeBasedGroupBySelectionField$inboundSchema.optional(),
});
/** @internal */
exports.SemanticThreadTimeBasedGroupBySelection$outboundSchema = z.object({
    type: exports.SemanticThreadTimeBasedGroupBySelectionType$outboundSchema,
    timeUnit: timeunit_js_1.TimeUnit$outboundSchema,
    field: exports.SemanticThreadTimeBasedGroupBySelectionField$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadTimeBasedGroupBySelection$;
(function (SemanticThreadTimeBasedGroupBySelection$) {
    /** @deprecated use `SemanticThreadTimeBasedGroupBySelection$inboundSchema` instead. */
    SemanticThreadTimeBasedGroupBySelection$.inboundSchema = exports.SemanticThreadTimeBasedGroupBySelection$inboundSchema;
    /** @deprecated use `SemanticThreadTimeBasedGroupBySelection$outboundSchema` instead. */
    SemanticThreadTimeBasedGroupBySelection$.outboundSchema = exports.SemanticThreadTimeBasedGroupBySelection$outboundSchema;
})(SemanticThreadTimeBasedGroupBySelection$ || (exports.SemanticThreadTimeBasedGroupBySelection$ = SemanticThreadTimeBasedGroupBySelection$ = {}));
function semanticThreadTimeBasedGroupBySelectionToJSON(semanticThreadTimeBasedGroupBySelection) {
    return JSON.stringify(exports.SemanticThreadTimeBasedGroupBySelection$outboundSchema.parse(semanticThreadTimeBasedGroupBySelection));
}
function semanticThreadTimeBasedGroupBySelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SemanticThreadTimeBasedGroupBySelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SemanticThreadTimeBasedGroupBySelection' from JSON`);
}
