// @inkeep/inkeep-analytics/dist/commonjs/models/components/semanticthreadsimplefieldselection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SemanticThreadSimpleFieldSelection$ = exports.SemanticThreadSimpleFieldSelection$outboundSchema = exports.SemanticThreadSimpleFieldSelection$inboundSchema = exports.SemanticThreadSimpleFieldSelectionField$ = exports.SemanticThreadSimpleFieldSelectionField$outboundSchema = exports.SemanticThreadSimpleFieldSelectionField$inboundSchema = exports.SemanticThreadSimpleFieldSelectionType$ = exports.SemanticThreadSimpleFieldSelectionType$outboundSchema = exports.SemanticThreadSimpleFieldSelectionType$inboundSchema = exports.SemanticThreadSimpleFieldSelectionField = exports.SemanticThreadSimpleFieldSelectionType = void 0;
exports.semanticThreadSimpleFieldSelectionToJSON = semanticThreadSimpleFieldSelectionToJSON;
exports.semanticThreadSimpleFieldSelectionFromJSON = semanticThreadSimpleFieldSelectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.SemanticThreadSimpleFieldSelectionType = {
    Field: "field",
};
exports.SemanticThreadSimpleFieldSelectionField = {
    Id: "id",
    MessageIds: "messageIds",
    MessagesCount: "messagesCount",
    OrganizationId: "organizationId",
    Question: "question",
    QuestionLength: "questionLength",
    Answer: "answer",
    IsOnTopic: "isOnTopic",
    IsClear: "isClear",
    IsDocumented: "isDocumented",
    IsFeatureSupported: "isFeatureSupported",
    IsAnswerConfident: "isAnswerConfident",
    GapSummary: "gapSummary",
    Subject: "subject",
    Category: "category",
    Languages: "languages",
    FirstPartyEntities: "firstPartyEntities",
    ThirdPartyEntities: "thirdPartyEntities",
    Emotion: "emotion",
    Sentiment: "sentiment",
    QuestionType: "questionType",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    Properties: "properties",
    UserProperties: "userProperties",
    FirstMessageTime: "firstMessageTime",
    LastMessageTime: "lastMessageTime",
    LastUpdated: "lastUpdated",
    HasPositiveFeedback: "hasPositiveFeedback",
    HasNegativeFeedback: "hasNegativeFeedback",
    ConversationIds: "conversationIds",
    AreAIAnnotationsVerified: "areAIAnnotationsVerified",
    IsInitialQuestionAnswer: "isInitialQuestionAnswer",
};
/** @internal */
exports.SemanticThreadSimpleFieldSelectionType$inboundSchema = z.nativeEnum(exports.SemanticThreadSimpleFieldSelectionType);
/** @internal */
exports.SemanticThreadSimpleFieldSelectionType$outboundSchema = exports.SemanticThreadSimpleFieldSelectionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadSimpleFieldSelectionType$;
(function (SemanticThreadSimpleFieldSelectionType$) {
    /** @deprecated use `SemanticThreadSimpleFieldSelectionType$inboundSchema` instead. */
    SemanticThreadSimpleFieldSelectionType$.inboundSchema = exports.SemanticThreadSimpleFieldSelectionType$inboundSchema;
    /** @deprecated use `SemanticThreadSimpleFieldSelectionType$outboundSchema` instead. */
    SemanticThreadSimpleFieldSelectionType$.outboundSchema = exports.SemanticThreadSimpleFieldSelectionType$outboundSchema;
})(SemanticThreadSimpleFieldSelectionType$ || (exports.SemanticThreadSimpleFieldSelectionType$ = SemanticThreadSimpleFieldSelectionType$ = {}));
/** @internal */
exports.SemanticThreadSimpleFieldSelectionField$inboundSchema = z
    .nativeEnum(exports.SemanticThreadSimpleFieldSelectionField);
/** @internal */
exports.SemanticThreadSimpleFieldSelectionField$outboundSchema = exports.SemanticThreadSimpleFieldSelectionField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadSimpleFieldSelectionField$;
(function (SemanticThreadSimpleFieldSelectionField$) {
    /** @deprecated use `SemanticThreadSimpleFieldSelectionField$inboundSchema` instead. */
    SemanticThreadSimpleFieldSelectionField$.inboundSchema = exports.SemanticThreadSimpleFieldSelectionField$inboundSchema;
    /** @deprecated use `SemanticThreadSimpleFieldSelectionField$outboundSchema` instead. */
    SemanticThreadSimpleFieldSelectionField$.outboundSchema = exports.SemanticThreadSimpleFieldSelectionField$outboundSchema;
})(SemanticThreadSimpleFieldSelectionField$ || (exports.SemanticThreadSimpleFieldSelectionField$ = SemanticThreadSimpleFieldSelectionField$ = {}));
/** @internal */
exports.SemanticThreadSimpleFieldSelection$inboundSchema = z.object({
    type: exports.SemanticThreadSimpleFieldSelectionType$inboundSchema,
    field: exports.SemanticThreadSimpleFieldSelectionField$inboundSchema,
    path: z.array(z.string()).optional(),
});
/** @internal */
exports.SemanticThreadSimpleFieldSelection$outboundSchema = z.object({
    type: exports.SemanticThreadSimpleFieldSelectionType$outboundSchema,
    field: exports.SemanticThreadSimpleFieldSelectionField$outboundSchema,
    path: z.array(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SemanticThreadSimpleFieldSelection$;
(function (SemanticThreadSimpleFieldSelection$) {
    /** @deprecated use `SemanticThreadSimpleFieldSelection$inboundSchema` instead. */
    SemanticThreadSimpleFieldSelection$.inboundSchema = exports.SemanticThreadSimpleFieldSelection$inboundSchema;
    /** @deprecated use `SemanticThreadSimpleFieldSelection$outboundSchema` instead. */
    SemanticThreadSimpleFieldSelection$.outboundSchema = exports.SemanticThreadSimpleFieldSelection$outboundSchema;
})(SemanticThreadSimpleFieldSelection$ || (exports.SemanticThreadSimpleFieldSelection$ = SemanticThreadSimpleFieldSelection$ = {}));
function semanticThreadSimpleFieldSelectionToJSON(semanticThreadSimpleFieldSelection) {
    return JSON.stringify(exports.SemanticThreadSimpleFieldSelection$outboundSchema.parse(semanticThreadSimpleFieldSelection));
}
function semanticThreadSimpleFieldSelectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SemanticThreadSimpleFieldSelection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SemanticThreadSimpleFieldSelection' from JSON`);
}
