// @inkeep/inkeep-analytics/src/models/components/conversationsfield.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { ClosedEnum } from "../../types/enums.js";

/**
 * Available fields for Conversations
 */
export const ConversationsField = {
  Id: "id",
  UserMessageCount: "userMessageCount",
  OrganizationId: "organizationId",
  ProjectId: "projectId",
  IntegrationId: "integrationId",
  FirstMessageTime: "firstMessageTime",
  Type: "type",
  Properties: "properties",
  UserProperties: "userProperties",
} as const;
/**
 * Available fields for Conversations
 */
export type ConversationsField = ClosedEnum<typeof ConversationsField>;

/** @internal */
export const ConversationsField$inboundSchema: z.ZodNativeEnum<
  typeof ConversationsField
> = z.nativeEnum(ConversationsField);

/** @internal */
export const ConversationsField$outboundSchema: z.ZodNativeEnum<
  typeof ConversationsField
> = ConversationsField$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ConversationsField$ {
  /** @deprecated use `ConversationsField$inboundSchema` instead. */
  export const inboundSchema = ConversationsField$inboundSchema;
  /** @deprecated use `ConversationsField$outboundSchema` instead. */
  export const outboundSchema = ConversationsField$outboundSchema;
}
