// @inkeep/inkeep-analytics/src/models/components/conversationsgroupby.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  ConversationsField,
  ConversationsField$inboundSchema,
  ConversationsField$outboundSchema,
} from "./conversationsfield.js";

/**
 * Group by options for conversations
 */
export type ConversationsGroupBy = {
  /**
   * Available fields for Conversations
   */
  field: ConversationsField;
  path?: Array<string> | undefined;
  includeInSelect?: boolean | undefined;
};

/** @internal */
export const ConversationsGroupBy$inboundSchema: z.ZodType<
  ConversationsGroupBy,
  z.ZodTypeDef,
  unknown
> = z.object({
  field: ConversationsField$inboundSchema,
  path: z.array(z.string()).optional(),
  includeInSelect: z.boolean().default(true),
});

/** @internal */
export type ConversationsGroupBy$Outbound = {
  field: string;
  path?: Array<string> | undefined;
  includeInSelect: boolean;
};

/** @internal */
export const ConversationsGroupBy$outboundSchema: z.ZodType<
  ConversationsGroupBy$Outbound,
  z.ZodTypeDef,
  ConversationsGroupBy
> = z.object({
  field: ConversationsField$outboundSchema,
  path: z.array(z.string()).optional(),
  includeInSelect: z.boolean().default(true),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ConversationsGroupBy$ {
  /** @deprecated use `ConversationsGroupBy$inboundSchema` instead. */
  export const inboundSchema = ConversationsGroupBy$inboundSchema;
  /** @deprecated use `ConversationsGroupBy$outboundSchema` instead. */
  export const outboundSchema = ConversationsGroupBy$outboundSchema;
  /** @deprecated use `ConversationsGroupBy$Outbound` instead. */
  export type Outbound = ConversationsGroupBy$Outbound;
}

export function conversationsGroupByToJSON(
  conversationsGroupBy: ConversationsGroupBy,
): string {
  return JSON.stringify(
    ConversationsGroupBy$outboundSchema.parse(conversationsGroupBy),
  );
}

export function conversationsGroupByFromJSON(
  jsonString: string,
): SafeParseResult<ConversationsGroupBy, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ConversationsGroupBy$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ConversationsGroupBy' from JSON`,
  );
}
