// @inkeep/inkeep-analytics/src/models/components/eventsgroupby.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  EventsField,
  EventsField$inboundSchema,
  EventsField$outboundSchema,
} from "./eventsfield.js";

/**
 * Group by options for events
 */
export type EventsGroupBy = {
  /**
   * Available fields for Events
   */
  field: EventsField;
  path?: Array<string> | undefined;
  includeInSelect?: boolean | undefined;
};

/** @internal */
export const EventsGroupBy$inboundSchema: z.ZodType<
  EventsGroupBy,
  z.ZodTypeDef,
  unknown
> = z.object({
  field: EventsField$inboundSchema,
  path: z.array(z.string()).optional(),
  includeInSelect: z.boolean().default(true),
});

/** @internal */
export type EventsGroupBy$Outbound = {
  field: string;
  path?: Array<string> | undefined;
  includeInSelect: boolean;
};

/** @internal */
export const EventsGroupBy$outboundSchema: z.ZodType<
  EventsGroupBy$Outbound,
  z.ZodTypeDef,
  EventsGroupBy
> = z.object({
  field: EventsField$outboundSchema,
  path: z.array(z.string()).optional(),
  includeInSelect: z.boolean().default(true),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace EventsGroupBy$ {
  /** @deprecated use `EventsGroupBy$inboundSchema` instead. */
  export const inboundSchema = EventsGroupBy$inboundSchema;
  /** @deprecated use `EventsGroupBy$outboundSchema` instead. */
  export const outboundSchema = EventsGroupBy$outboundSchema;
  /** @deprecated use `EventsGroupBy$Outbound` instead. */
  export type Outbound = EventsGroupBy$Outbound;
}

export function eventsGroupByToJSON(eventsGroupBy: EventsGroupBy): string {
  return JSON.stringify(EventsGroupBy$outboundSchema.parse(eventsGroupBy));
}

export function eventsGroupByFromJSON(
  jsonString: string,
): SafeParseResult<EventsGroupBy, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => EventsGroupBy$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'EventsGroupBy' from JSON`,
  );
}
