// @inkeep/inkeep-analytics/src/models/components/openaitoolcallfunction.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type OpenAIToolCallFunction = {
  name: string;
  arguments: string;
};

/** @internal */
export const OpenAIToolCallFunction$inboundSchema: z.ZodType<
  OpenAIToolCallFunction,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  arguments: z.string(),
});

/** @internal */
export type OpenAIToolCallFunction$Outbound = {
  name: string;
  arguments: string;
};

/** @internal */
export const OpenAIToolCallFunction$outboundSchema: z.ZodType<
  OpenAIToolCallFunction$Outbound,
  z.ZodTypeDef,
  OpenAIToolCallFunction
> = z.object({
  name: z.string(),
  arguments: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace OpenAIToolCallFunction$ {
  /** @deprecated use `OpenAIToolCallFunction$inboundSchema` instead. */
  export const inboundSchema = OpenAIToolCallFunction$inboundSchema;
  /** @deprecated use `OpenAIToolCallFunction$outboundSchema` instead. */
  export const outboundSchema = OpenAIToolCallFunction$outboundSchema;
  /** @deprecated use `OpenAIToolCallFunction$Outbound` instead. */
  export type Outbound = OpenAIToolCallFunction$Outbound;
}

export function openAIToolCallFunctionToJSON(
  openAIToolCallFunction: OpenAIToolCallFunction,
): string {
  return JSON.stringify(
    OpenAIToolCallFunction$outboundSchema.parse(openAIToolCallFunction),
  );
}

export function openAIToolCallFunctionFromJSON(
  jsonString: string,
): SafeParseResult<OpenAIToolCallFunction, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => OpenAIToolCallFunction$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'OpenAIToolCallFunction' from JSON`,
  );
}
