// @inkeep/inkeep-analytics/src/models/components/semanticthreadfilter.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  SemanticThreadFilterAND,
  SemanticThreadFilterAND$inboundSchema,
  SemanticThreadFilterAND$Outbound,
  SemanticThreadFilterAND$outboundSchema,
} from "./semanticthreadfilterand.js";
import {
  SemanticThreadFilterCondition,
  SemanticThreadFilterCondition$inboundSchema,
  SemanticThreadFilterCondition$Outbound,
  SemanticThreadFilterCondition$outboundSchema,
} from "./semanticthreadfiltercondition.js";
import {
  SemanticThreadFilterOR,
  SemanticThreadFilterOR$inboundSchema,
  SemanticThreadFilterOR$Outbound,
  SemanticThreadFilterOR$outboundSchema,
} from "./semanticthreadfilteror.js";

export type SemanticThreadFilter =
  | SemanticThreadFilterCondition
  | SemanticThreadFilterAND
  | SemanticThreadFilterOR;

/** @internal */
export const SemanticThreadFilter$inboundSchema: z.ZodType<
  SemanticThreadFilter,
  z.ZodTypeDef,
  unknown
> = z.union([
  SemanticThreadFilterCondition$inboundSchema,
  z.lazy(() => SemanticThreadFilterAND$inboundSchema),
  z.lazy(() => SemanticThreadFilterOR$inboundSchema),
]);

/** @internal */
export type SemanticThreadFilter$Outbound =
  | SemanticThreadFilterCondition$Outbound
  | SemanticThreadFilterAND$Outbound
  | SemanticThreadFilterOR$Outbound;

/** @internal */
export const SemanticThreadFilter$outboundSchema: z.ZodType<
  SemanticThreadFilter$Outbound,
  z.ZodTypeDef,
  SemanticThreadFilter
> = z.union([
  SemanticThreadFilterCondition$outboundSchema,
  z.lazy(() => SemanticThreadFilterAND$outboundSchema),
  z.lazy(() => SemanticThreadFilterOR$outboundSchema),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace SemanticThreadFilter$ {
  /** @deprecated use `SemanticThreadFilter$inboundSchema` instead. */
  export const inboundSchema = SemanticThreadFilter$inboundSchema;
  /** @deprecated use `SemanticThreadFilter$outboundSchema` instead. */
  export const outboundSchema = SemanticThreadFilter$outboundSchema;
  /** @deprecated use `SemanticThreadFilter$Outbound` instead. */
  export type Outbound = SemanticThreadFilter$Outbound;
}

export function semanticThreadFilterToJSON(
  semanticThreadFilter: SemanticThreadFilter,
): string {
  return JSON.stringify(
    SemanticThreadFilter$outboundSchema.parse(semanticThreadFilter),
  );
}

export function semanticThreadFilterFromJSON(
  jsonString: string,
): SafeParseResult<SemanticThreadFilter, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => SemanticThreadFilter$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'SemanticThreadFilter' from JSON`,
  );
}
