// @inkeep/inkeep-analytics/src/models/components/timeunit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { ClosedEnum } from "../../types/enums.js";

/**
 * Valid time units for time-based operations
 */
export const TimeUnit = {
  Hour: "hour",
  Day: "day",
  Week: "week",
  Month: "month",
} as const;
/**
 * Valid time units for time-based operations
 */
export type TimeUnit = ClosedEnum<typeof TimeUnit>;

/** @internal */
export const TimeUnit$inboundSchema: z.ZodNativeEnum<typeof TimeUnit> = z
  .nativeEnum(TimeUnit);

/** @internal */
export const TimeUnit$outboundSchema: z.ZodNativeEnum<typeof TimeUnit> =
  TimeUnit$inboundSchema;

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TimeUnit$ {
  /** @deprecated use `TimeUnit$inboundSchema` instead. */
  export const inboundSchema = TimeUnit$inboundSchema;
  /** @deprecated use `TimeUnit$outboundSchema` instead. */
  export const outboundSchema = TimeUnit$outboundSchema;
}
