// @inkeep/inkeep-analytics/dist/esm/sdk/query.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { queryConversations } from "../funcs/queryConversations.js";
import { queryExportSemanticThreadsQueryResults } from "../funcs/queryExportSemanticThreadsQueryResults.js";
import { queryQueryEvents } from "../funcs/queryQueryEvents.js";
import { queryQueryFeedback } from "../funcs/queryQueryFeedback.js";
import { queryQueryPropertyKeys } from "../funcs/queryQueryPropertyKeys.js";
import { queryQueryPropertyValues } from "../funcs/queryQueryPropertyValues.js";
import { queryQuerySemanticThreads } from "../funcs/queryQuerySemanticThreads.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class Query extends ClientSDK {
    /**
     * Query Conversations
     */
    async conversations(security, request, options) {
        return unwrapAsync(queryConversations(this, security, request, options));
    }
    /**
     * Query Events
     */
    async queryEvents(security, request, options) {
        return unwrapAsync(queryQueryEvents(this, security, request, options));
    }
    /**
     * Query Feedback
     */
    async queryFeedback(security, request, options) {
        return unwrapAsync(queryQueryFeedback(this, security, request, options));
    }
    /**
     * Query Semantic Threads
     */
    async querySemanticThreads(security, request, options) {
        return unwrapAsync(queryQuerySemanticThreads(this, security, request, options));
    }
    /**
     * Export Semantic Threads Query Results
     */
    async exportSemanticThreadsQueryResults(request, options) {
        return unwrapAsync(queryExportSemanticThreadsQueryResults(this, request, options));
    }
    /**
     * Query Property Keys
     */
    async queryPropertyKeys(request, options) {
        return unwrapAsync(queryQueryPropertyKeys(this, request, options));
    }
    /**
     * Query Property Values
     */
    async queryPropertyValues(request, options) {
        return unwrapAsync(queryQueryPropertyValues(this, request, options));
    }
}
