// @inkeep/inkeep-analytics/dist/commonjs/models/operations/deleteconversation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteConversationResponseBody$ = exports.DeleteConversationResponseBody$outboundSchema = exports.DeleteConversationResponseBody$inboundSchema = exports.DeleteConversationRequest$ = exports.DeleteConversationRequest$outboundSchema = exports.DeleteConversationRequest$inboundSchema = exports.DeleteConversationSecurity$ = exports.DeleteConversationSecurity$outboundSchema = exports.DeleteConversationSecurity$inboundSchema = void 0;
exports.deleteConversationSecurityToJSON = deleteConversationSecurityToJSON;
exports.deleteConversationSecurityFromJSON = deleteConversationSecurityFromJSON;
exports.deleteConversationRequestToJSON = deleteConversationRequestToJSON;
exports.deleteConversationRequestFromJSON = deleteConversationRequestFromJSON;
exports.deleteConversationResponseBodyToJSON = deleteConversationResponseBodyToJSON;
exports.deleteConversationResponseBodyFromJSON = deleteConversationResponseBodyFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.DeleteConversationSecurity$inboundSchema = z.object({
    WebIntegrationKey: z.string().optional(),
    ApiIntegrationKey: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "WebIntegrationKey": "webIntegrationKey",
        "ApiIntegrationKey": "apiIntegrationKey",
    });
});
/** @internal */
exports.DeleteConversationSecurity$outboundSchema = z.object({
    webIntegrationKey: z.string().optional(),
    apiIntegrationKey: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        webIntegrationKey: "WebIntegrationKey",
        apiIntegrationKey: "ApiIntegrationKey",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteConversationSecurity$;
(function (DeleteConversationSecurity$) {
    /** @deprecated use `DeleteConversationSecurity$inboundSchema` instead. */
    DeleteConversationSecurity$.inboundSchema = exports.DeleteConversationSecurity$inboundSchema;
    /** @deprecated use `DeleteConversationSecurity$outboundSchema` instead. */
    DeleteConversationSecurity$.outboundSchema = exports.DeleteConversationSecurity$outboundSchema;
})(DeleteConversationSecurity$ || (exports.DeleteConversationSecurity$ = DeleteConversationSecurity$ = {}));
function deleteConversationSecurityToJSON(deleteConversationSecurity) {
    return JSON.stringify(exports.DeleteConversationSecurity$outboundSchema.parse(deleteConversationSecurity));
}
function deleteConversationSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteConversationSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteConversationSecurity' from JSON`);
}
/** @internal */
exports.DeleteConversationRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.DeleteConversationRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteConversationRequest$;
(function (DeleteConversationRequest$) {
    /** @deprecated use `DeleteConversationRequest$inboundSchema` instead. */
    DeleteConversationRequest$.inboundSchema = exports.DeleteConversationRequest$inboundSchema;
    /** @deprecated use `DeleteConversationRequest$outboundSchema` instead. */
    DeleteConversationRequest$.outboundSchema = exports.DeleteConversationRequest$outboundSchema;
})(DeleteConversationRequest$ || (exports.DeleteConversationRequest$ = DeleteConversationRequest$ = {}));
function deleteConversationRequestToJSON(deleteConversationRequest) {
    return JSON.stringify(exports.DeleteConversationRequest$outboundSchema.parse(deleteConversationRequest));
}
function deleteConversationRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteConversationRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteConversationRequest' from JSON`);
}
/** @internal */
exports.DeleteConversationResponseBody$inboundSchema = z.object({
    success: z.boolean(),
    message: z.string(),
});
/** @internal */
exports.DeleteConversationResponseBody$outboundSchema = z.object({
    success: z.boolean(),
    message: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteConversationResponseBody$;
(function (DeleteConversationResponseBody$) {
    /** @deprecated use `DeleteConversationResponseBody$inboundSchema` instead. */
    DeleteConversationResponseBody$.inboundSchema = exports.DeleteConversationResponseBody$inboundSchema;
    /** @deprecated use `DeleteConversationResponseBody$outboundSchema` instead. */
    DeleteConversationResponseBody$.outboundSchema = exports.DeleteConversationResponseBody$outboundSchema;
})(DeleteConversationResponseBody$ || (exports.DeleteConversationResponseBody$ = DeleteConversationResponseBody$ = {}));
function deleteConversationResponseBodyToJSON(deleteConversationResponseBody) {
    return JSON.stringify(exports.DeleteConversationResponseBody$outboundSchema.parse(deleteConversationResponseBody));
}
function deleteConversationResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteConversationResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteConversationResponseBody' from JSON`);
}
