// @inkeep/inkeep-analytics/dist/commonjs/models/operations/getallconversation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAllConversationResponseBody$ = exports.GetAllConversationResponseBody$outboundSchema = exports.GetAllConversationResponseBody$inboundSchema = exports.GetAllConversationRequest$ = exports.GetAllConversationRequest$outboundSchema = exports.GetAllConversationRequest$inboundSchema = void 0;
exports.getAllConversationRequestToJSON = getAllConversationRequestToJSON;
exports.getAllConversationRequestFromJSON = getAllConversationRequestFromJSON;
exports.getAllConversationResponseBodyToJSON = getAllConversationResponseBodyToJSON;
exports.getAllConversationResponseBodyFromJSON = getAllConversationResponseBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.GetAllConversationRequest$inboundSchema = z.object({
    limit: z.nullable(z.number().int().default(20)),
    offset: z.nullable(z.number().int().default(0)),
});
/** @internal */
exports.GetAllConversationRequest$outboundSchema = z.object({
    limit: z.nullable(z.number().int().default(20)),
    offset: z.nullable(z.number().int().default(0)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAllConversationRequest$;
(function (GetAllConversationRequest$) {
    /** @deprecated use `GetAllConversationRequest$inboundSchema` instead. */
    GetAllConversationRequest$.inboundSchema = exports.GetAllConversationRequest$inboundSchema;
    /** @deprecated use `GetAllConversationRequest$outboundSchema` instead. */
    GetAllConversationRequest$.outboundSchema = exports.GetAllConversationRequest$outboundSchema;
})(GetAllConversationRequest$ || (exports.GetAllConversationRequest$ = GetAllConversationRequest$ = {}));
function getAllConversationRequestToJSON(getAllConversationRequest) {
    return JSON.stringify(exports.GetAllConversationRequest$outboundSchema.parse(getAllConversationRequest));
}
function getAllConversationRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetAllConversationRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAllConversationRequest' from JSON`);
}
/** @internal */
exports.GetAllConversationResponseBody$inboundSchema = z.object({
    conversations: z.array(components.Conversation$inboundSchema),
    total: z.number(),
    pageSize: z.number(),
    count: z.number(),
});
/** @internal */
exports.GetAllConversationResponseBody$outboundSchema = z.object({
    conversations: z.array(components.Conversation$outboundSchema),
    total: z.number(),
    pageSize: z.number(),
    count: z.number(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAllConversationResponseBody$;
(function (GetAllConversationResponseBody$) {
    /** @deprecated use `GetAllConversationResponseBody$inboundSchema` instead. */
    GetAllConversationResponseBody$.inboundSchema = exports.GetAllConversationResponseBody$inboundSchema;
    /** @deprecated use `GetAllConversationResponseBody$outboundSchema` instead. */
    GetAllConversationResponseBody$.outboundSchema = exports.GetAllConversationResponseBody$outboundSchema;
})(GetAllConversationResponseBody$ || (exports.GetAllConversationResponseBody$ = GetAllConversationResponseBody$ = {}));
function getAllConversationResponseBodyToJSON(getAllConversationResponseBody) {
    return JSON.stringify(exports.GetAllConversationResponseBody$outboundSchema.parse(getAllConversationResponseBody));
}
function getAllConversationResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetAllConversationResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAllConversationResponseBody' from JSON`);
}
