// @inkeep/inkeep-analytics/dist/commonjs/models/operations/querysemanticthreads.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.QuerySemanticThreadsSecurity$ = exports.QuerySemanticThreadsSecurity$outboundSchema = exports.QuerySemanticThreadsSecurity$inboundSchema = void 0;
exports.querySemanticThreadsSecurityToJSON = querySemanticThreadsSecurityToJSON;
exports.querySemanticThreadsSecurityFromJSON = querySemanticThreadsSecurityFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.QuerySemanticThreadsSecurity$inboundSchema = z.object({
    WebIntegrationKey: z.string().optional(),
    ApiIntegrationKey: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "WebIntegrationKey": "webIntegrationKey",
        "ApiIntegrationKey": "apiIntegrationKey",
    });
});
/** @internal */
exports.QuerySemanticThreadsSecurity$outboundSchema = z.object({
    webIntegrationKey: z.string().optional(),
    apiIntegrationKey: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        webIntegrationKey: "WebIntegrationKey",
        apiIntegrationKey: "ApiIntegrationKey",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QuerySemanticThreadsSecurity$;
(function (QuerySemanticThreadsSecurity$) {
    /** @deprecated use `QuerySemanticThreadsSecurity$inboundSchema` instead. */
    QuerySemanticThreadsSecurity$.inboundSchema = exports.QuerySemanticThreadsSecurity$inboundSchema;
    /** @deprecated use `QuerySemanticThreadsSecurity$outboundSchema` instead. */
    QuerySemanticThreadsSecurity$.outboundSchema = exports.QuerySemanticThreadsSecurity$outboundSchema;
})(QuerySemanticThreadsSecurity$ || (exports.QuerySemanticThreadsSecurity$ = QuerySemanticThreadsSecurity$ = {}));
function querySemanticThreadsSecurityToJSON(querySemanticThreadsSecurity) {
    return JSON.stringify(exports.QuerySemanticThreadsSecurity$outboundSchema.parse(querySemanticThreadsSecurity));
}
function querySemanticThreadsSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.QuerySemanticThreadsSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QuerySemanticThreadsSecurity' from JSON`);
}
