// @inkeep/inkeep-analytics/dist/esm/react-query/conversationsDelete.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { conversationsDelete } from "../funcs/conversationsDelete.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
/**
 * Delete Conversation
 */
export function useConversationsDeleteMutation(options) {
    const client = useInkeepAnalyticsContext();
    return useMutation({
        ...buildConversationsDeleteMutation(client, options),
        ...options,
    });
}
export function mutationKeyConversationsDelete() {
    return ["@inkeep/inkeep-analytics", "conversations", "delete"];
}
export function buildConversationsDeleteMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyConversationsDelete(),
        mutationFn: function conversationsDeleteMutationFn({ security, request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(conversationsDelete(client$, security, request, mergedOptions));
        },
    };
}
