// @inkeep/inkeep-analytics/dist/esm/react-query/conversationsGet.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { conversationsGet } from "../funcs/conversationsGet.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
/**
 * Get Conversation
 */
export function useConversationsGet(security, request, options) {
    const client = useInkeepAnalyticsContext();
    return useQuery({
        ...buildConversationsGetQuery(client, security, request, options),
        ...options,
    });
}
/**
 * Get Conversation
 */
export function useConversationsGetSuspense(security, request, options) {
    const client = useInkeepAnalyticsContext();
    return useSuspenseQuery({
        ...buildConversationsGetQuery(client, security, request, options),
        ...options,
    });
}
export function prefetchConversationsGet(queryClient, client$, security, request) {
    return queryClient.prefetchQuery({
        ...buildConversationsGetQuery(client$, security, request),
    });
}
export function setConversationsGetData(client, queryKeyBase, data) {
    const key = queryKeyConversationsGet(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateConversationsGet(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: [
            "@inkeep/inkeep-analytics",
            "conversations",
            "get",
            ...queryKeyBase,
        ],
    });
}
export function invalidateAllConversationsGet(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@inkeep/inkeep-analytics", "conversations", "get"],
    });
}
export function buildConversationsGetQuery(client$, security, request, options) {
    return {
        queryKey: queryKeyConversationsGet(request.id),
        queryFn: async function conversationsGetQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(conversationsGet(client$, security, request, mergedOptions));
        },
    };
}
export function queryKeyConversationsGet(id) {
    return ["@inkeep/inkeep-analytics", "conversations", "get", id];
}
