// @inkeep/inkeep-analytics/dist/esm/react-query/conversationsLog.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { conversationsLog } from "../funcs/conversationsLog.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
/**
 * Log Conversation
 *
 * @remarks
 * Logs a new conversation or updates an existing one with new messages. Always include all messages.
 *
 * **API Key Types:** `WEB`, `API`
 */
export function useConversationsLogMutation(options) {
    const client = useInkeepAnalyticsContext();
    return useMutation({
        ...buildConversationsLogMutation(client, options),
        ...options,
    });
}
export function mutationKeyConversationsLog() {
    return ["@inkeep/inkeep-analytics", "conversations", "log"];
}
export function buildConversationsLogMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyConversationsLog(),
        mutationFn: function conversationsLogMutationFn({ security, request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(conversationsLog(client$, security, request, mergedOptions));
        },
    };
}
