// @inkeep/inkeep-analytics/dist/esm/react-query/eventsLog.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { eventsLog } from "../funcs/eventsLog.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
/**
 * Log Event
 */
export function useEventsLogMutation(options) {
    const client = useInkeepAnalyticsContext();
    return useMutation({
        ...buildEventsLogMutation(client, options),
        ...options,
    });
}
export function mutationKeyEventsLog() {
    return ["@inkeep/inkeep-analytics", "events", "log"];
}
export function buildEventsLogMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyEventsLog(),
        mutationFn: function eventsLogMutationFn({ security, request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(eventsLog(client$, security, request, mergedOptions));
        },
    };
}
