// @inkeep/inkeep-analytics/dist/esm/react-query/feedbackList.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { feedbackList } from "../funcs/feedbackList.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
/**
 * Get All Feedback
 */
export function useFeedbackList(request, options) {
    const client = useInkeepAnalyticsContext();
    return useQuery({
        ...buildFeedbackListQuery(client, request, options),
        ...options,
    });
}
/**
 * Get All Feedback
 */
export function useFeedbackListSuspense(request, options) {
    const client = useInkeepAnalyticsContext();
    return useSuspenseQuery({
        ...buildFeedbackListQuery(client, request, options),
        ...options,
    });
}
export function prefetchFeedbackList(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildFeedbackListQuery(client$, request),
    });
}
export function setFeedbackListData(client, queryKeyBase, data) {
    const key = queryKeyFeedbackList(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateFeedbackList(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@inkeep/inkeep-analytics", "feedback", "list", ...queryKeyBase],
    });
}
export function invalidateAllFeedbackList(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@inkeep/inkeep-analytics", "feedback", "list"],
    });
}
export function buildFeedbackListQuery(client$, request, options) {
    return {
        queryKey: queryKeyFeedbackList({
            limit: request.limit,
            offset: request.offset,
        }),
        queryFn: async function feedbackListQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(feedbackList(client$, request, mergedOptions));
        },
    };
}
export function queryKeyFeedbackList(parameters) {
    return ["@inkeep/inkeep-analytics", "feedback", "list", parameters];
}
