// @inkeep/inkeep-analytics/dist/esm/react-query/queryExportSemanticThreadsQueryResults.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { queryExportSemanticThreadsQueryResults } from "../funcs/queryExportSemanticThreadsQueryResults.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
/**
 * Export Semantic Threads Query Results
 */
export function useQueryExportSemanticThreadsQueryResultsMutation(options) {
    const client = useInkeepAnalyticsContext();
    return useMutation({
        ...buildQueryExportSemanticThreadsQueryResultsMutation(client, options),
        ...options,
    });
}
export function mutationKeyQueryExportSemanticThreadsQueryResults() {
    return [
        "@inkeep/inkeep-analytics",
        "Query",
        "exportSemanticThreadsQueryResults",
    ];
}
export function buildQueryExportSemanticThreadsQueryResultsMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyQueryExportSemanticThreadsQueryResults(),
        mutationFn: function queryExportSemanticThreadsQueryResultsMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(queryExportSemanticThreadsQueryResults(client$, request, mergedOptions));
        },
    };
}
