// @inkeep/inkeep-analytics/dist/esm/react-query/queryQueryPropertyKeys.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useMutation, } from "@tanstack/react-query";
import { queryQueryPropertyKeys } from "../funcs/queryQueryPropertyKeys.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
/**
 * Query Property Keys
 */
export function useQueryQueryPropertyKeysMutation(options) {
    const client = useInkeepAnalyticsContext();
    return useMutation({
        ...buildQueryQueryPropertyKeysMutation(client, options),
        ...options,
    });
}
export function mutationKeyQueryQueryPropertyKeys() {
    return ["@inkeep/inkeep-analytics", "Query", "queryPropertyKeys"];
}
export function buildQueryQueryPropertyKeysMutation(client$, hookOptions) {
    return {
        mutationKey: mutationKeyQueryQueryPropertyKeys(),
        mutationFn: function queryQueryPropertyKeysMutationFn({ request, options, }) {
            const mergedOptions = {
                ...hookOptions,
                ...options,
                fetchOptions: {
                    ...hookOptions?.fetchOptions,
                    ...options?.fetchOptions,
                    signal: combineSignals(hookOptions?.fetchOptions?.signal, options?.fetchOptions?.signal),
                },
            };
            return unwrapAsync(queryQueryPropertyKeys(client$, request, mergedOptions));
        },
    };
}
