// @inkeep/inkeep-analytics/dist/esm/react-query/topSearchQueries.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { useQuery, useSuspenseQuery, } from "@tanstack/react-query";
import { topSearchQueries } from "../funcs/topSearchQueries.js";
import { combineSignals } from "../lib/primitives.js";
import { unwrapAsync } from "../types/fp.js";
import { useInkeepAnalyticsContext } from "./_context.js";
/**
 * Top Search Queries
 */
export function useTopSearchQueries(request, options) {
    const client = useInkeepAnalyticsContext();
    return useQuery({
        ...buildTopSearchQueriesQuery(client, request, options),
        ...options,
    });
}
/**
 * Top Search Queries
 */
export function useTopSearchQueriesSuspense(request, options) {
    const client = useInkeepAnalyticsContext();
    return useSuspenseQuery({
        ...buildTopSearchQueriesQuery(client, request, options),
        ...options,
    });
}
export function prefetchTopSearchQueries(queryClient, client$, request) {
    return queryClient.prefetchQuery({
        ...buildTopSearchQueriesQuery(client$, request),
    });
}
export function setTopSearchQueriesData(client, queryKeyBase, data) {
    const key = queryKeyTopSearchQueries(...queryKeyBase);
    return client.setQueryData(key, data);
}
export function invalidateTopSearchQueries(client, queryKeyBase, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@inkeep/inkeep-analytics", "topSearchQueries", ...queryKeyBase],
    });
}
export function invalidateAllTopSearchQueries(client, filters) {
    return client.invalidateQueries({
        ...filters,
        queryKey: ["@inkeep/inkeep-analytics", "topSearchQueries"],
    });
}
export function buildTopSearchQueriesQuery(client$, request, options) {
    return {
        queryKey: queryKeyTopSearchQueries({
            after: request.after,
            projectId: request.projectId,
        }),
        queryFn: async function topSearchQueriesQueryFn(ctx) {
            const sig = combineSignals(ctx.signal, options?.fetchOptions?.signal);
            const mergedOptions = {
                ...options,
                fetchOptions: { ...options?.fetchOptions, signal: sig },
            };
            return unwrapAsync(topSearchQueries(client$, request, mergedOptions));
        },
    };
}
export function queryKeyTopSearchQueries(parameters) {
    return ["@inkeep/inkeep-analytics", "topSearchQueries", parameters];
}
