// @inkeep/inkeep-analytics/src/models/operations/logconversation.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type LogConversationSecurity = {
  webIntegrationKey?: string | undefined;
  apiIntegrationKey?: string | undefined;
};

/** @internal */
export const LogConversationSecurity$inboundSchema: z.ZodType<
  LogConversationSecurity,
  z.ZodTypeDef,
  unknown
> = z.object({
  WebIntegrationKey: z.string().optional(),
  ApiIntegrationKey: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "WebIntegrationKey": "webIntegrationKey",
    "ApiIntegrationKey": "apiIntegrationKey",
  });
});

/** @internal */
export type LogConversationSecurity$Outbound = {
  WebIntegrationKey?: string | undefined;
  ApiIntegrationKey?: string | undefined;
};

/** @internal */
export const LogConversationSecurity$outboundSchema: z.ZodType<
  LogConversationSecurity$Outbound,
  z.ZodTypeDef,
  LogConversationSecurity
> = z.object({
  webIntegrationKey: z.string().optional(),
  apiIntegrationKey: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    webIntegrationKey: "WebIntegrationKey",
    apiIntegrationKey: "ApiIntegrationKey",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LogConversationSecurity$ {
  /** @deprecated use `LogConversationSecurity$inboundSchema` instead. */
  export const inboundSchema = LogConversationSecurity$inboundSchema;
  /** @deprecated use `LogConversationSecurity$outboundSchema` instead. */
  export const outboundSchema = LogConversationSecurity$outboundSchema;
  /** @deprecated use `LogConversationSecurity$Outbound` instead. */
  export type Outbound = LogConversationSecurity$Outbound;
}

export function logConversationSecurityToJSON(
  logConversationSecurity: LogConversationSecurity,
): string {
  return JSON.stringify(
    LogConversationSecurity$outboundSchema.parse(logConversationSecurity),
  );
}

export function logConversationSecurityFromJSON(
  jsonString: string,
): SafeParseResult<LogConversationSecurity, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => LogConversationSecurity$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'LogConversationSecurity' from JSON`,
  );
}
