// @inkeep/inkeep-analytics/src/models/operations/logevent.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type LogEventSecurity = {
  webIntegrationKey?: string | undefined;
  apiIntegrationKey?: string | undefined;
};

/** @internal */
export const LogEventSecurity$inboundSchema: z.ZodType<
  LogEventSecurity,
  z.ZodTypeDef,
  unknown
> = z.object({
  WebIntegrationKey: z.string().optional(),
  ApiIntegrationKey: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "WebIntegrationKey": "webIntegrationKey",
    "ApiIntegrationKey": "apiIntegrationKey",
  });
});

/** @internal */
export type LogEventSecurity$Outbound = {
  WebIntegrationKey?: string | undefined;
  ApiIntegrationKey?: string | undefined;
};

/** @internal */
export const LogEventSecurity$outboundSchema: z.ZodType<
  LogEventSecurity$Outbound,
  z.ZodTypeDef,
  LogEventSecurity
> = z.object({
  webIntegrationKey: z.string().optional(),
  apiIntegrationKey: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    webIntegrationKey: "WebIntegrationKey",
    apiIntegrationKey: "ApiIntegrationKey",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LogEventSecurity$ {
  /** @deprecated use `LogEventSecurity$inboundSchema` instead. */
  export const inboundSchema = LogEventSecurity$inboundSchema;
  /** @deprecated use `LogEventSecurity$outboundSchema` instead. */
  export const outboundSchema = LogEventSecurity$outboundSchema;
  /** @deprecated use `LogEventSecurity$Outbound` instead. */
  export type Outbound = LogEventSecurity$Outbound;
}

export function logEventSecurityToJSON(
  logEventSecurity: LogEventSecurity,
): string {
  return JSON.stringify(
    LogEventSecurity$outboundSchema.parse(logEventSecurity),
  );
}

export function logEventSecurityFromJSON(
  jsonString: string,
): SafeParseResult<LogEventSecurity, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => LogEventSecurity$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'LogEventSecurity' from JSON`,
  );
}
