// @inkeep/inkeep-analytics/src/models/operations/queryfeedback.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type QueryFeedbackSecurity = {
  webIntegrationKey?: string | undefined;
  apiIntegrationKey?: string | undefined;
};

/** @internal */
export const QueryFeedbackSecurity$inboundSchema: z.ZodType<
  QueryFeedbackSecurity,
  z.ZodTypeDef,
  unknown
> = z.object({
  WebIntegrationKey: z.string().optional(),
  ApiIntegrationKey: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "WebIntegrationKey": "webIntegrationKey",
    "ApiIntegrationKey": "apiIntegrationKey",
  });
});

/** @internal */
export type QueryFeedbackSecurity$Outbound = {
  WebIntegrationKey?: string | undefined;
  ApiIntegrationKey?: string | undefined;
};

/** @internal */
export const QueryFeedbackSecurity$outboundSchema: z.ZodType<
  QueryFeedbackSecurity$Outbound,
  z.ZodTypeDef,
  QueryFeedbackSecurity
> = z.object({
  webIntegrationKey: z.string().optional(),
  apiIntegrationKey: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    webIntegrationKey: "WebIntegrationKey",
    apiIntegrationKey: "ApiIntegrationKey",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace QueryFeedbackSecurity$ {
  /** @deprecated use `QueryFeedbackSecurity$inboundSchema` instead. */
  export const inboundSchema = QueryFeedbackSecurity$inboundSchema;
  /** @deprecated use `QueryFeedbackSecurity$outboundSchema` instead. */
  export const outboundSchema = QueryFeedbackSecurity$outboundSchema;
  /** @deprecated use `QueryFeedbackSecurity$Outbound` instead. */
  export type Outbound = QueryFeedbackSecurity$Outbound;
}

export function queryFeedbackSecurityToJSON(
  queryFeedbackSecurity: QueryFeedbackSecurity,
): string {
  return JSON.stringify(
    QueryFeedbackSecurity$outboundSchema.parse(queryFeedbackSecurity),
  );
}

export function queryFeedbackSecurityFromJSON(
  jsonString: string,
): SafeParseResult<QueryFeedbackSecurity, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => QueryFeedbackSecurity$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'QueryFeedbackSecurity' from JSON`,
  );
}
