// @inkeep/inkeep-analytics/src/mcp-server/tools/conversationGetConversationByExternalId.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { conversationGetConversationByExternalId } from "../../funcs/conversationGetConversationByExternalId.js";
import * as operations from "../../models/operations/index.js";
import { formatResult, ToolDefinition } from "../tools.js";

const args = {
  request: operations.GetConversationByExternalIdRequest$inboundSchema,
};

export const tool$conversationGetConversationByExternalId: ToolDefinition<
  typeof args
> = {
  name: "conversation-get-conversation-by-external-id",
  description: `Get Conversation by External ID`,
  args,
  tool: async (client, args, ctx) => {
    const [result, apiCall] = await conversationGetConversationByExternalId(
      client,
      args.request,
      { fetchOptions: { signal: ctx.signal } },
    ).$inspect();

    if (!result.ok) {
      return {
        content: [{ type: "text", text: result.error.message }],
        isError: true,
      };
    }

    const value = result.value;

    return formatResult(value, apiCall);
  },
};
