// @inkeep/inkeep-analytics/dist/esm/models/components/conversationsfield.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
/**
 * Available fields for Conversations
 */
export const ConversationsField = {
    Id: "id",
    UserMessageCount: "userMessageCount",
    OrganizationId: "organizationId",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    FirstMessageTime: "firstMessageTime",
    Type: "type",
    Properties: "properties",
    UserProperties: "userProperties",
};
/** @internal */
export const ConversationsField$inboundSchema = z.nativeEnum(ConversationsField);
/** @internal */
export const ConversationsField$outboundSchema = ConversationsField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ConversationsField$;
(function (ConversationsField$) {
    /** @deprecated use `ConversationsField$inboundSchema` instead. */
    ConversationsField$.inboundSchema = ConversationsField$inboundSchema;
    /** @deprecated use `ConversationsField$outboundSchema` instead. */
    ConversationsField$.outboundSchema = ConversationsField$outboundSchema;
})(ConversationsField$ || (ConversationsField$ = {}));
