// @inkeep/inkeep-analytics/dist/esm/models/components/eventsfield.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
/**
 * Available fields for Events
 */
export const EventsField = {
    Id: "id",
    OrganizationId: "organizationId",
    MessageId: "messageId",
    ConversationId: "conversationId",
    CreatedAt: "createdAt",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    EventType: "eventType",
    Type: "type",
    SearchQuery: "searchQuery",
    Properties: "properties",
    UserProperties: "userProperties",
};
/** @internal */
export const EventsField$inboundSchema = z
    .nativeEnum(EventsField);
/** @internal */
export const EventsField$outboundSchema = EventsField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var EventsField$;
(function (EventsField$) {
    /** @deprecated use `EventsField$inboundSchema` instead. */
    EventsField$.inboundSchema = EventsField$inboundSchema;
    /** @deprecated use `EventsField$outboundSchema` instead. */
    EventsField$.outboundSchema = EventsField$outboundSchema;
})(EventsField$ || (EventsField$ = {}));
