// @inkeep/inkeep-analytics/dist/esm/models/components/feedbackfield.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
/**
 * Available fields for Feedback
 */
export const FeedbackField = {
    Id: "id",
    Type: "type",
    MessageId: "messageId",
    CreatedAt: "createdAt",
    OrganizationId: "organizationId",
    Reasons: "reasons",
    Properties: "properties",
    UserProperties: "userProperties",
    ConversationId: "conversationId",
    ProjectId: "projectId",
    IntegrationId: "integrationId",
    UpdatedAt: "updatedAt",
    Status: "status",
};
/** @internal */
export const FeedbackField$inboundSchema = z.nativeEnum(FeedbackField);
/** @internal */
export const FeedbackField$outboundSchema = FeedbackField$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FeedbackField$;
(function (FeedbackField$) {
    /** @deprecated use `FeedbackField$inboundSchema` instead. */
    FeedbackField$.inboundSchema = FeedbackField$inboundSchema;
    /** @deprecated use `FeedbackField$outboundSchema` instead. */
    FeedbackField$.outboundSchema = FeedbackField$outboundSchema;
})(FeedbackField$ || (FeedbackField$ = {}));
