// @inkeep/inkeep-analytics/dist/esm/models/components/feedbackfilterand.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { FeedbackFilter$inboundSchema, FeedbackFilter$outboundSchema, } from "./feedbackfilter.js";
/** @internal */
export const FeedbackFilterANDAND$inboundSchema = z.lazy(() => FeedbackFilter$inboundSchema);
/** @internal */
export const FeedbackFilterANDAND$outboundSchema = z.lazy(() => FeedbackFilter$outboundSchema);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FeedbackFilterANDAND$;
(function (FeedbackFilterANDAND$) {
    /** @deprecated use `FeedbackFilterANDAND$inboundSchema` instead. */
    FeedbackFilterANDAND$.inboundSchema = FeedbackFilterANDAND$inboundSchema;
    /** @deprecated use `FeedbackFilterANDAND$outboundSchema` instead. */
    FeedbackFilterANDAND$.outboundSchema = FeedbackFilterANDAND$outboundSchema;
})(FeedbackFilterANDAND$ || (FeedbackFilterANDAND$ = {}));
export function feedbackFilterANDANDToJSON(feedbackFilterANDAND) {
    return JSON.stringify(FeedbackFilterANDAND$outboundSchema.parse(feedbackFilterANDAND));
}
export function feedbackFilterANDANDFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FeedbackFilterANDAND$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackFilterANDAND' from JSON`);
}
/** @internal */
export const FeedbackFilterAND$inboundSchema = z.object({
    and: z.array(z.lazy(() => FeedbackFilter$inboundSchema)),
});
/** @internal */
export const FeedbackFilterAND$outboundSchema = z.object({
    and: z.array(z.lazy(() => FeedbackFilter$outboundSchema)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FeedbackFilterAND$;
(function (FeedbackFilterAND$) {
    /** @deprecated use `FeedbackFilterAND$inboundSchema` instead. */
    FeedbackFilterAND$.inboundSchema = FeedbackFilterAND$inboundSchema;
    /** @deprecated use `FeedbackFilterAND$outboundSchema` instead. */
    FeedbackFilterAND$.outboundSchema = FeedbackFilterAND$outboundSchema;
})(FeedbackFilterAND$ || (FeedbackFilterAND$ = {}));
export function feedbackFilterANDToJSON(feedbackFilterAND) {
    return JSON.stringify(FeedbackFilterAND$outboundSchema.parse(feedbackFilterAND));
}
export function feedbackFilterANDFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FeedbackFilterAND$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FeedbackFilterAND' from JSON`);
}
