// @inkeep/inkeep-analytics/dist/esm/models/components/filteroperator.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
/**
 * Available operators for filtering data
 */
export const FilterOperator = {
    Eq: "eq",
    Neq: "neq",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
    In: "in",
    NotIn: "notIn",
    ArrayContains: "arrayContains",
    JsonContains: "jsonContains",
    JsonContainedBy: "jsonContainedBy",
    JsonHasKey: "jsonHasKey",
    JsonHasAnyKey: "jsonHasAnyKey",
    JsonHasAllKeys: "jsonHasAllKeys",
    IsNull: "isNull",
    IsNotNull: "isNotNull",
};
/** @internal */
export const FilterOperator$inboundSchema = z.nativeEnum(FilterOperator);
/** @internal */
export const FilterOperator$outboundSchema = FilterOperator$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var FilterOperator$;
(function (FilterOperator$) {
    /** @deprecated use `FilterOperator$inboundSchema` instead. */
    FilterOperator$.inboundSchema = FilterOperator$inboundSchema;
    /** @deprecated use `FilterOperator$outboundSchema` instead. */
    FilterOperator$.outboundSchema = FilterOperator$outboundSchema;
})(FilterOperator$ || (FilterOperator$ = {}));
