// @inkeep/inkeep-analytics/dist/esm/models/components/openaiimageurlobject.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
export const Detail = {
    Auto: "auto",
    Low: "low",
    High: "high",
};
/** @internal */
export const Detail$inboundSchema = z
    .nativeEnum(Detail);
/** @internal */
export const Detail$outboundSchema = Detail$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Detail$;
(function (Detail$) {
    /** @deprecated use `Detail$inboundSchema` instead. */
    Detail$.inboundSchema = Detail$inboundSchema;
    /** @deprecated use `Detail$outboundSchema` instead. */
    Detail$.outboundSchema = Detail$outboundSchema;
})(Detail$ || (Detail$ = {}));
/** @internal */
export const OpenAIImageUrlObject$inboundSchema = z.object({
    url: z.string(),
    detail: Detail$inboundSchema.default("auto"),
});
/** @internal */
export const OpenAIImageUrlObject$outboundSchema = z.object({
    url: z.string(),
    detail: Detail$outboundSchema.default("auto"),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OpenAIImageUrlObject$;
(function (OpenAIImageUrlObject$) {
    /** @deprecated use `OpenAIImageUrlObject$inboundSchema` instead. */
    OpenAIImageUrlObject$.inboundSchema = OpenAIImageUrlObject$inboundSchema;
    /** @deprecated use `OpenAIImageUrlObject$outboundSchema` instead. */
    OpenAIImageUrlObject$.outboundSchema = OpenAIImageUrlObject$outboundSchema;
})(OpenAIImageUrlObject$ || (OpenAIImageUrlObject$ = {}));
export function openAIImageUrlObjectToJSON(openAIImageUrlObject) {
    return JSON.stringify(OpenAIImageUrlObject$outboundSchema.parse(openAIImageUrlObject));
}
export function openAIImageUrlObjectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenAIImageUrlObject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenAIImageUrlObject' from JSON`);
}
