// @inkeep/inkeep-analytics/dist/esm/models/components/queryeventsresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { QueryEventsResponseData$inboundSchema, QueryEventsResponseData$outboundSchema, } from "./queryeventsresponsedata.js";
/**
 * Status of the response
 */
export const QueryEventsResponseBodyStatus = {
    Ok: "ok",
};
/** @internal */
export const QueryEventsResponseBodyStatus$inboundSchema = z.nativeEnum(QueryEventsResponseBodyStatus);
/** @internal */
export const QueryEventsResponseBodyStatus$outboundSchema = QueryEventsResponseBodyStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var QueryEventsResponseBodyStatus$;
(function (QueryEventsResponseBodyStatus$) {
    /** @deprecated use `QueryEventsResponseBodyStatus$inboundSchema` instead. */
    QueryEventsResponseBodyStatus$.inboundSchema = QueryEventsResponseBodyStatus$inboundSchema;
    /** @deprecated use `QueryEventsResponseBodyStatus$outboundSchema` instead. */
    QueryEventsResponseBodyStatus$.outboundSchema = QueryEventsResponseBodyStatus$outboundSchema;
})(QueryEventsResponseBodyStatus$ || (QueryEventsResponseBodyStatus$ = {}));
/** @internal */
export const QueryEventsResponseBody$inboundSchema = z.object({
    status: QueryEventsResponseBodyStatus$inboundSchema,
    data: QueryEventsResponseData$inboundSchema,
});
/** @internal */
export const QueryEventsResponseBody$outboundSchema = z.object({
    status: QueryEventsResponseBodyStatus$outboundSchema,
    data: QueryEventsResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var QueryEventsResponseBody$;
(function (QueryEventsResponseBody$) {
    /** @deprecated use `QueryEventsResponseBody$inboundSchema` instead. */
    QueryEventsResponseBody$.inboundSchema = QueryEventsResponseBody$inboundSchema;
    /** @deprecated use `QueryEventsResponseBody$outboundSchema` instead. */
    QueryEventsResponseBody$.outboundSchema = QueryEventsResponseBody$outboundSchema;
})(QueryEventsResponseBody$ || (QueryEventsResponseBody$ = {}));
export function queryEventsResponseBodyToJSON(queryEventsResponseBody) {
    return JSON.stringify(QueryEventsResponseBody$outboundSchema.parse(queryEventsResponseBody));
}
export function queryEventsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => QueryEventsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryEventsResponseBody' from JSON`);
}
