// @inkeep/inkeep-analytics/dist/esm/models/components/queryfeedbackresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { QueryFeedbackResponseData$inboundSchema, QueryFeedbackResponseData$outboundSchema, } from "./queryfeedbackresponsedata.js";
/**
 * Status of the response
 */
export const QueryFeedbackResponseBodyStatus = {
    Ok: "ok",
};
/** @internal */
export const QueryFeedbackResponseBodyStatus$inboundSchema = z.nativeEnum(QueryFeedbackResponseBodyStatus);
/** @internal */
export const QueryFeedbackResponseBodyStatus$outboundSchema = QueryFeedbackResponseBodyStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var QueryFeedbackResponseBodyStatus$;
(function (QueryFeedbackResponseBodyStatus$) {
    /** @deprecated use `QueryFeedbackResponseBodyStatus$inboundSchema` instead. */
    QueryFeedbackResponseBodyStatus$.inboundSchema = QueryFeedbackResponseBodyStatus$inboundSchema;
    /** @deprecated use `QueryFeedbackResponseBodyStatus$outboundSchema` instead. */
    QueryFeedbackResponseBodyStatus$.outboundSchema = QueryFeedbackResponseBodyStatus$outboundSchema;
})(QueryFeedbackResponseBodyStatus$ || (QueryFeedbackResponseBodyStatus$ = {}));
/** @internal */
export const QueryFeedbackResponseBody$inboundSchema = z.object({
    status: QueryFeedbackResponseBodyStatus$inboundSchema,
    data: QueryFeedbackResponseData$inboundSchema,
});
/** @internal */
export const QueryFeedbackResponseBody$outboundSchema = z.object({
    status: QueryFeedbackResponseBodyStatus$outboundSchema,
    data: QueryFeedbackResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var QueryFeedbackResponseBody$;
(function (QueryFeedbackResponseBody$) {
    /** @deprecated use `QueryFeedbackResponseBody$inboundSchema` instead. */
    QueryFeedbackResponseBody$.inboundSchema = QueryFeedbackResponseBody$inboundSchema;
    /** @deprecated use `QueryFeedbackResponseBody$outboundSchema` instead. */
    QueryFeedbackResponseBody$.outboundSchema = QueryFeedbackResponseBody$outboundSchema;
})(QueryFeedbackResponseBody$ || (QueryFeedbackResponseBody$ = {}));
export function queryFeedbackResponseBodyToJSON(queryFeedbackResponseBody) {
    return JSON.stringify(QueryFeedbackResponseBody$outboundSchema.parse(queryFeedbackResponseBody));
}
export function queryFeedbackResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => QueryFeedbackResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryFeedbackResponseBody' from JSON`);
}
