// @inkeep/inkeep-analytics/dist/esm/models/operations/getallfeedback.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const GetAllFeedbackRequest$inboundSchema = z.object({
    limit: z.nullable(z.number().int().default(10)),
    offset: z.nullable(z.number().int().default(0)),
});
/** @internal */
export const GetAllFeedbackRequest$outboundSchema = z.object({
    limit: z.nullable(z.number().int().default(10)),
    offset: z.nullable(z.number().int().default(0)),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetAllFeedbackRequest$;
(function (GetAllFeedbackRequest$) {
    /** @deprecated use `GetAllFeedbackRequest$inboundSchema` instead. */
    GetAllFeedbackRequest$.inboundSchema = GetAllFeedbackRequest$inboundSchema;
    /** @deprecated use `GetAllFeedbackRequest$outboundSchema` instead. */
    GetAllFeedbackRequest$.outboundSchema = GetAllFeedbackRequest$outboundSchema;
})(GetAllFeedbackRequest$ || (GetAllFeedbackRequest$ = {}));
export function getAllFeedbackRequestToJSON(getAllFeedbackRequest) {
    return JSON.stringify(GetAllFeedbackRequest$outboundSchema.parse(getAllFeedbackRequest));
}
export function getAllFeedbackRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAllFeedbackRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAllFeedbackRequest' from JSON`);
}
