// @inkeep/inkeep-analytics/dist/esm/models/operations/queryconversations.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const QueryConversationsSecurity$inboundSchema = z.object({
    WebIntegrationKey: z.string().optional(),
    ApiIntegrationKey: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "WebIntegrationKey": "webIntegrationKey",
        "ApiIntegrationKey": "apiIntegrationKey",
    });
});
/** @internal */
export const QueryConversationsSecurity$outboundSchema = z.object({
    webIntegrationKey: z.string().optional(),
    apiIntegrationKey: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        webIntegrationKey: "WebIntegrationKey",
        apiIntegrationKey: "ApiIntegrationKey",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var QueryConversationsSecurity$;
(function (QueryConversationsSecurity$) {
    /** @deprecated use `QueryConversationsSecurity$inboundSchema` instead. */
    QueryConversationsSecurity$.inboundSchema = QueryConversationsSecurity$inboundSchema;
    /** @deprecated use `QueryConversationsSecurity$outboundSchema` instead. */
    QueryConversationsSecurity$.outboundSchema = QueryConversationsSecurity$outboundSchema;
})(QueryConversationsSecurity$ || (QueryConversationsSecurity$ = {}));
export function queryConversationsSecurityToJSON(queryConversationsSecurity) {
    return JSON.stringify(QueryConversationsSecurity$outboundSchema.parse(queryConversationsSecurity));
}
export function queryConversationsSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => QueryConversationsSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QueryConversationsSecurity' from JSON`);
}
