// @inkeep/inkeep-analytics/dist/esm/models/operations/querysemanticthreads.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const QuerySemanticThreadsSecurity$inboundSchema = z.object({
    WebIntegrationKey: z.string().optional(),
    ApiIntegrationKey: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "WebIntegrationKey": "webIntegrationKey",
        "ApiIntegrationKey": "apiIntegrationKey",
    });
});
/** @internal */
export const QuerySemanticThreadsSecurity$outboundSchema = z.object({
    webIntegrationKey: z.string().optional(),
    apiIntegrationKey: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        webIntegrationKey: "WebIntegrationKey",
        apiIntegrationKey: "ApiIntegrationKey",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var QuerySemanticThreadsSecurity$;
(function (QuerySemanticThreadsSecurity$) {
    /** @deprecated use `QuerySemanticThreadsSecurity$inboundSchema` instead. */
    QuerySemanticThreadsSecurity$.inboundSchema = QuerySemanticThreadsSecurity$inboundSchema;
    /** @deprecated use `QuerySemanticThreadsSecurity$outboundSchema` instead. */
    QuerySemanticThreadsSecurity$.outboundSchema = QuerySemanticThreadsSecurity$outboundSchema;
})(QuerySemanticThreadsSecurity$ || (QuerySemanticThreadsSecurity$ = {}));
export function querySemanticThreadsSecurityToJSON(querySemanticThreadsSecurity) {
    return JSON.stringify(QuerySemanticThreadsSecurity$outboundSchema.parse(querySemanticThreadsSecurity));
}
export function querySemanticThreadsSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => QuerySemanticThreadsSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'QuerySemanticThreadsSecurity' from JSON`);
}
